{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7932464-d5f9-4cbe-b97e-c221327716d7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# rename column\n",
    "indel12 = renameit(indel1, '#Uploaded_variation', 'Uploaded_variation')\n",
    "snp12 = renameit(snp1, '#Uploaded_variation', 'Uploaded_variation')\n",
    "indel63 = renameit(indel6, '#Uploaded_variation', 'Uploaded_variation')\n",
    "snp63 = renameit(snp6, '#Uploaded_variation', 'Uploaded_variation')\n",
    "\n",
    "# split Location  on : to make chromosome(str) and chrCoordinate(int) cols\n",
    "indel12[['chromosome', 'chrCoordinate']] = indel12['Location'].str.split(':', expand=True)\n",
    "snp12[['chromosome', 'chrCoordinate']] = snp12['Location'].str.split(':', expand=True)\n",
    "indel63[['chromosome', 'chrCoordinate']] = indel63['Location'].str.split(':', expand=True)\n",
    "snp63[['chromosome', 'chrCoordinate']] = snp63['Location'].str.split(':', expand=True)\n",
    "\n",
    "\n",
    "## note chrCoordinnate is the onnly integer column in the dataframe\n",
    "\n",
    "# if chrCoordinate column contains \"-\", \n",
    "# split the column on - to get start and end columns, change start and end columns to type int, else change chrCoordinate column values to type int\n",
    "# NOTE the indel df  start position is being used as chrCoordinate and the start could actually be larger than end position depending on strand \n",
    "indel12['chrCoordinate'] = indel12['chrCoordinate'].str.split('-', expand=False).str[0].astype(int)\n",
    "snp12['chrCoordinate'] = snp12['chrCoordinate'].astype(int)\n",
    "indel63['chrCoordinate'] = indel63['chrCoordinate'].str.split('-', expand=False).str[0].astype(int)\n",
    "snp63['chrCoordinate'] = snp63['chrCoordinate'].astype(int)\n",
    "\n",
    "## note cDNA position for inndels has a - character and is not an integer or float\n",
    "## note cds position for inndels has a - character and is not an integer or float\n",
    "## note prot position for inndels has a - character and is not an integer or float\n",
    "\n",
    "\n",
    "# \"-\" values to nan ::\n",
    "def replace_dash_with_nan(df, col_name):\n",
    "       df[col_name] = df[col_name].replace('-', np.nan)\n",
    "       return df\n",
    "\n",
    "\n",
    "# change cDNA_position \"-\" values to nan \n",
    "indel12 = replace_dash_with_nan(indel12, 'cDNA_position')\n",
    "snp12 = replace_dash_with_nan(snp12, 'cDNA_position')\n",
    "indel63 = replace_dash_with_nan(indel63, 'cDNA_position')\n",
    "snp63 = replace_dash_with_nan(snp63, 'cDNA_position')\n",
    "\n",
    "# change CDS_position column values equal to \"-\" to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'CDS_position')\n",
    "snp12 = replace_dash_with_nan(snp12, 'CDS_position')\n",
    "indel63 = replace_dash_with_nan(indel63, 'CDS_position')\n",
    "snp63 = replace_dash_with_nan(snp63, 'CDS_position')\n",
    "\n",
    "# change Protein_position column values equal to \"-\" to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'Protein_position')\n",
    "snp12 = replace_dash_with_nan(snp12, 'Protein_position')\n",
    "indel63 = replace_dash_with_nan(indel63, 'Protein_position')\n",
    "snp63 = replace_dash_with_nan(snp63, 'Protein_position')\n",
    "\n",
    "\n",
    "# if SIFT column contains a value, split the column on \"(\" to get SIFTprediction and SIFTscore, remove \")\" from SIFTscore and change SIFTscore to type float\n",
    "snp12[['SIFTprediction', 'SIFTscore']] = snp12['SIFT'].str.split('(', expand=True)\n",
    "snp12['SIFTscore'] = snp12['SIFTscore'].str.replace(')', '')\n",
    "snp12['SIFTscore'] = snp12['SIFTscore'].astype(float)\n",
    "\n",
    "snp63[['SIFTprediction', 'SIFTscore']] = snp63['SIFT'].str.split('(', expand=True)\n",
    "snp63['SIFTscore'] = snp63['SIFTscore'].str.replace(')', '')\n",
    "snp63['SIFTscore'] = snp63['SIFTscore'].astype(float)\n",
    "\n",
    "# if PolyPhen column contains a value, split the column on \"(\" to get PolyPhenprediction and PolyPhenscore, remove \")\" from PolyPhenscore and change PolyPhenscore to type float\n",
    "snp12[['PolyPhenprediction', 'PolyPhenscore']] = snp12['PolyPhen'].str.split('(', \n",
    "expand=True)\n",
    "snp12['PolyPhenscore'] = snp12['PolyPhenscore'].str.replace(')', '')\n",
    "snp12['PolyPhenscore'] = snp12['PolyPhenscore'].astype(float)\n",
    "\n",
    "snp63[['PolyPhenprediction', 'PolyPhenscore']] = snp63['PolyPhen'].str.split('(', expand=True)\n",
    "snp63['PolyPhenscore'] = snp63['PolyPhenscore'].str.replace(')', '')\n",
    "snp63['PolyPhenscore'] = snp63['PolyPhenscore'].astype(float)\n",
    "\n",
    "\n",
    "# convert distance, exon numberator and intron nnumerators to float\n",
    "\n",
    "# change DISTANCE column values equal to \"-\" to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'DISTANCE')\n",
    "snp12 = replace_dash_with_nan(snp12, 'DISTANCE')\n",
    "indel63 = replace_dash_with_nan(indel63, 'DISTANCE')\n",
    "snp63 = replace_dash_with_nan(snp63, 'DISTANCE')\n",
    "\n",
    "# duplicate EXON and save as new column EXONnumber\n",
    "indel12['EXONnumber'] = indel12['EXON']\n",
    "snp12['EXONnumber'] = snp12['EXON']\n",
    "indel63['EXONnumber'] = indel63['EXON']\n",
    "snp63['EXONnumber'] = snp63['EXON']\n",
    "\n",
    "# change EXONnumber and INTRONnumber columns values equal to \"-\" to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'EXONnumber')\n",
    "snp12 = replace_dash_with_nan(snp12, 'EXONnumber')\n",
    "indel63 = replace_dash_with_nan(indel63, 'EXONnumber')\n",
    "snp63 = replace_dash_with_nan(snp63, 'EXONnumber')\n",
    "\n",
    "# split the EXONnumber and INTRONnumber column values on \"/\", retain only the value before the \"/\"\n",
    "indel12['EXONnumber'] = indel12['EXONnumber'].str.split('/').str[0]\n",
    "snp12['EXONnumber'] = snp12['EXONnumber'].str.split('/').str[0]\n",
    "indel63['EXONnumber'] = indel63['EXONnumber'].str.split('/').str[0]\n",
    "snp63['EXONnumber'] = snp63['EXONnumber'].str.split('/').str[0]\n",
    "\n",
    "\n",
    "# duplicate INTRON column and save as new column INTRONnumber\n",
    "indel12['INTRONnumber'] = indel12['INTRON']\n",
    "snp12['INTRONnumber'] = snp12['INTRON']\n",
    "indel63['INTRONnumber'] = indel63['INTRON']\n",
    "snp63['INTRONnumber'] = snp63['INTRON']\n",
    "\n",
    "# change INTRONnumber column values equal to \"-\" to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'INTRONnumber')\n",
    "snp12 = replace_dash_with_nan(snp12, 'INTRONnumber')\n",
    "indel63 = replace_dash_with_nan(indel63, 'INTRONnumber')\n",
    "snp63 = replace_dash_with_nan(snp63, 'INTRONnumber')\n",
    "\n",
    "# split the INTRONnumber column values on \"/\", retain only the value before the \"/\"\n",
    "indel12['INTRONnumber'] = indel12['INTRONnumber'].str.split('/').str[0]\n",
    "snp12['INTRONnumber'] = snp12['INTRONnumber'].str.split('/').str[0]\n",
    "indel63['INTRONnumber'] = indel63['INTRONnumber'].str.split('/').str[0]\n",
    "snp63['INTRONnumber'] = snp63['INTRONnumber'].str.split('/').str[0]\n",
    "\n",
    "\n",
    "\n",
    "# FLOAT CONVERRSION for distance, exonnumber, intronnumber\n",
    "col_list = ['DISTANCE', 'INTRONnumber', 'EXONnumber']\n",
    "\n",
    "\n",
    "def convert_to_float(df, col_list):\n",
    "    # a function that takes a list of columns and dataframe and \n",
    "    # returns the dataframe with the columns converted to type float\n",
    "    for i in col_list:\n",
    "        df[i] = df[i].astype(float)\n",
    "    return(df)\n",
    "\n",
    "indel12 = convert_to_float(indel12, col_list)\n",
    "snp12 = convert_to_float(snp12, col_list)\n",
    "indel63 = convert_to_float(indel63, col_list)\n",
    "snp63 = convert_to_float(snp63, col_list)\n",
    "\n",
    "\n",
    "# if colname contains \"gnomAD\", change the \"-\" values to nan and change to float\n",
    "\n",
    "for i in indel12.columns:\n",
    "    if 'gnomAD' in i:\n",
    "        indel12[i] = indel12[i].replace('-', np.nan)  \n",
    "        indel12[i] = indel12[i].astype(float) \n",
    "for i in snp12.columns:\n",
    "    if 'gnomAD' in i:\n",
    "        snp12[i] = snp12[i].replace('-', np.nan)\n",
    "        snp12[i] = snp12[i].astype(float)\n",
    "for i in indel63.columns:\n",
    "       if 'gnomAD' in i:\n",
    "              indel63[i] = indel63[i].replace('-', np.nan)\n",
    "              indel63[i] = indel63[i].astype(float)\n",
    "for i in snp63.columns:\n",
    "       if 'gnomAD' in i:\n",
    "              snp63[i] = snp63[i].replace('-', np.nan)\n",
    "              snp63[i] = snp63[i].astype(float)\n",
    "\n",
    "# change the \"-\" values in column MAX_AF to nan\n",
    "indel12 = replace_dash_with_nan(indel12, 'MAX_AF')\n",
    "snp12 = replace_dash_with_nan(snp12, 'MAX_AF')\n",
    "indel63 = replace_dash_with_nan(indel63, 'MAX_AF')\n",
    "snp63 = replace_dash_with_nan(snp63, 'MAX_AF')\n",
    "\n",
    "# change MAX_AF to type float\n",
    "indel12['MAX_AF'] = indel12['MAX_AF'].astype(float)\n",
    "snp12['MAX_AF'] = snp12['MAX_AF'].astype(float)\n",
    "indel63['MAX_AF'] = indel63['MAX_AF'].astype(float)\n",
    "snp63['MAX_AF'] = snp63['MAX_AF'].astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb8becd7-8540-44d5-b765-eb5926a0b972",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(2950097, 83)\n",
      "Uploaded_variation        object\n",
      "Location                  object\n",
      "Allele                    object\n",
      "Gene                      object\n",
      "Feature                   object\n",
      "Feature_type              object\n",
      "Consequence               object\n",
      "cDNA_position             object\n",
      "CDS_position              object\n",
      "Protein_position          object\n",
      "Amino_acids               object\n",
      "Codons                    object\n",
      "Existing_variation        object\n",
      "IND                       object\n",
      "ZYG                       object\n",
      "REF_ALLELE                object\n",
      "IMPACT                    object\n",
      "DISTANCE                 float64\n",
      "STRAND                    object\n",
      "FLAGS                     object\n",
      "VARIANT_CLASS             object\n",
      "SYMBOL                    object\n",
      "SYMBOL_SOURCE             object\n",
      "HGNC_ID                   object\n",
      "BIOTYPE                   object\n",
      "CANONICAL                 object\n",
      "MANE_SELECT               object\n",
      "MANE_PLUS_CLINICAL        object\n",
      "TSL                       object\n",
      "APPRIS                    object\n",
      "CCDS                      object\n",
      "ENSP                      object\n",
      "SWISSPROT                 object\n",
      "TREMBL                    object\n",
      "UNIPARC                   object\n",
      "UNIPROT_ISOFORM           object\n",
      "REFSEQ_MATCH              object\n",
      "SOURCE                    object\n",
      "REFSEQ_OFFSET             object\n",
      "GIVEN_REF                 object\n",
      "USED_REF                  object\n",
      "BAM_EDIT                  object\n",
      "GENE_PHENO                object\n",
      "SIFT                      object\n",
      "PolyPhen                  object\n",
      "EXON                      object\n",
      "INTRON                    object\n",
      "DOMAINS                   object\n",
      "miRNA                     object\n",
      "HGVSc                     object\n",
      "HGVSp                     object\n",
      "HGVS_OFFSET               object\n",
      "gnomADg_AF               float64\n",
      "gnomADg_AFR_AF           float64\n",
      "gnomADg_AMI_AF           float64\n",
      "gnomADg_AMR_AF           float64\n",
      "gnomADg_ASJ_AF           float64\n",
      "gnomADg_EAS_AF           float64\n",
      "gnomADg_FIN_AF           float64\n",
      "gnomADg_MID_AF           float64\n",
      "gnomADg_NFE_AF           float64\n",
      "gnomADg_OTH_AF           float64\n",
      "gnomADg_SAS_AF           float64\n",
      "MAX_AF                   float64\n",
      "MAX_AF_POPS               object\n",
      "CLIN_SIG                  object\n",
      "SOMATIC                   object\n",
      "PHENO                     object\n",
      "PUBMED                    object\n",
      "MOTIF_NAME                object\n",
      "MOTIF_POS                 object\n",
      "HIGH_INF_POS              object\n",
      "MOTIF_SCORE_CHANGE        object\n",
      "TRANSCRIPTION_FACTORS     object\n",
      "chromosome                object\n",
      "chrCoordinate              int64\n",
      "SIFTprediction            object\n",
      "SIFTscore                float64\n",
      "PolyPhenprediction        object\n",
      "PolyPhenscore            float64\n",
      "EXONnumber               float64\n",
      "INTRONnumber             float64\n",
      "chr                       object\n",
      "dtype: object\n",
      "Uploaded_variation             0\n",
      "Location                       0\n",
      "Allele                         0\n",
      "Gene                           0\n",
      "Feature                        0\n",
      "Feature_type                   0\n",
      "Consequence                    0\n",
      "cDNA_position            2874084\n",
      "CDS_position             2930494\n",
      "Protein_position         2930494\n",
      "Amino_acids                    0\n",
      "Codons                         0\n",
      "Existing_variation             0\n",
      "IND                            0\n",
      "ZYG                            0\n",
      "REF_ALLELE                     0\n",
      "IMPACT                         0\n",
      "DISTANCE                 2592349\n",
      "STRAND                         0\n",
      "FLAGS                          0\n",
      "VARIANT_CLASS                  0\n",
      "SYMBOL                         0\n",
      "SYMBOL_SOURCE                  0\n",
      "HGNC_ID                        0\n",
      "BIOTYPE                        0\n",
      "CANONICAL                      0\n",
      "MANE_SELECT                    0\n",
      "MANE_PLUS_CLINICAL             0\n",
      "TSL                            0\n",
      "APPRIS                         0\n",
      "CCDS                           0\n",
      "ENSP                           0\n",
      "SWISSPROT                      0\n",
      "TREMBL                         0\n",
      "UNIPARC                        0\n",
      "UNIPROT_ISOFORM                0\n",
      "REFSEQ_MATCH                   0\n",
      "SOURCE                         0\n",
      "REFSEQ_OFFSET                  0\n",
      "GIVEN_REF                      0\n",
      "USED_REF                       0\n",
      "BAM_EDIT                       0\n",
      "GENE_PHENO                     0\n",
      "SIFT                           0\n",
      "PolyPhen                       0\n",
      "EXON                           0\n",
      "INTRON                         0\n",
      "DOMAINS                        0\n",
      "miRNA                          0\n",
      "HGVSc                          0\n",
      "HGVSp                          0\n",
      "HGVS_OFFSET                    0\n",
      "gnomADg_AF                 28916\n",
      "gnomADg_AFR_AF             28918\n",
      "gnomADg_AMI_AF             28976\n",
      "gnomADg_AMR_AF             28919\n",
      "gnomADg_ASJ_AF             28929\n",
      "gnomADg_EAS_AF             28922\n",
      "gnomADg_FIN_AF             28921\n",
      "gnomADg_MID_AF             29004\n",
      "gnomADg_NFE_AF             28916\n",
      "gnomADg_OTH_AF             28931\n",
      "gnomADg_SAS_AF             28919\n",
      "MAX_AF                     28764\n",
      "MAX_AF_POPS                    0\n",
      "CLIN_SIG                       0\n",
      "SOMATIC                        0\n",
      "PHENO                          0\n",
      "PUBMED                         0\n",
      "MOTIF_NAME                     0\n",
      "MOTIF_POS                      0\n",
      "HIGH_INF_POS                   0\n",
      "MOTIF_SCORE_CHANGE             0\n",
      "TRANSCRIPTION_FACTORS          0\n",
      "chromosome                     0\n",
      "chrCoordinate                  0\n",
      "SIFTprediction                 0\n",
      "SIFTscore                2941355\n",
      "PolyPhenprediction             0\n",
      "PolyPhenscore            2941653\n",
      "EXONnumber               2874084\n",
      "INTRONnumber             1368926\n",
      "chr                            0\n",
      "dtype: int64\n",
      "(3138661, 83)\n",
      "Uploaded_variation        object\n",
      "Location                  object\n",
      "Allele                    object\n",
      "Gene                      object\n",
      "Feature                   object\n",
      "Feature_type              object\n",
      "Consequence               object\n",
      "cDNA_position             object\n",
      "CDS_position              object\n",
      "Protein_position          object\n",
      "Amino_acids               object\n",
      "Codons                    object\n",
      "Existing_variation        object\n",
      "IND                       object\n",
      "ZYG                       object\n",
      "REF_ALLELE                object\n",
      "IMPACT                    object\n",
      "DISTANCE                 float64\n",
      "STRAND                    object\n",
      "FLAGS                     object\n",
      "VARIANT_CLASS             object\n",
      "SYMBOL                    object\n",
      "SYMBOL_SOURCE             object\n",
      "HGNC_ID                   object\n",
      "BIOTYPE                   object\n",
      "CANONICAL                 object\n",
      "MANE_SELECT               object\n",
      "MANE_PLUS_CLINICAL        object\n",
      "TSL                       object\n",
      "APPRIS                    object\n",
      "CCDS                      object\n",
      "ENSP                      object\n",
      "SWISSPROT                 object\n",
      "TREMBL                    object\n",
      "UNIPARC                   object\n",
      "UNIPROT_ISOFORM           object\n",
      "REFSEQ_MATCH              object\n",
      "SOURCE                    object\n",
      "REFSEQ_OFFSET             object\n",
      "GIVEN_REF                 object\n",
      "USED_REF                  object\n",
      "BAM_EDIT                  object\n",
      "GENE_PHENO                object\n",
      "SIFT                      object\n",
      "PolyPhen                  object\n",
      "EXON                      object\n",
      "INTRON                    object\n",
      "DOMAINS                   object\n",
      "miRNA                     object\n",
      "HGVSc                     object\n",
      "HGVSp                     object\n",
      "HGVS_OFFSET               object\n",
      "gnomADg_AF               float64\n",
      "gnomADg_AFR_AF           float64\n",
      "gnomADg_AMI_AF           float64\n",
      "gnomADg_AMR_AF           float64\n",
      "gnomADg_ASJ_AF           float64\n",
      "gnomADg_EAS_AF           float64\n",
      "gnomADg_FIN_AF           float64\n",
      "gnomADg_MID_AF           float64\n",
      "gnomADg_NFE_AF           float64\n",
      "gnomADg_OTH_AF           float64\n",
      "gnomADg_SAS_AF           float64\n",
      "MAX_AF                   float64\n",
      "MAX_AF_POPS               object\n",
      "CLIN_SIG                  object\n",
      "SOMATIC                   object\n",
      "PHENO                     object\n",
      "PUBMED                    object\n",
      "MOTIF_NAME                object\n",
      "MOTIF_POS                 object\n",
      "HIGH_INF_POS              object\n",
      "MOTIF_SCORE_CHANGE        object\n",
      "TRANSCRIPTION_FACTORS     object\n",
      "chromosome                object\n",
      "chrCoordinate              int64\n",
      "SIFTprediction            object\n",
      "SIFTscore                float64\n",
      "PolyPhenprediction        object\n",
      "PolyPhenscore            float64\n",
      "EXONnumber               float64\n",
      "INTRONnumber             float64\n",
      "chr                       object\n",
      "dtype: object\n",
      "Uploaded_variation             0\n",
      "Location                       0\n",
      "Allele                         0\n",
      "Gene                           0\n",
      "Feature                        0\n",
      "Feature_type                   0\n",
      "Consequence                    0\n",
      "cDNA_position            3061218\n",
      "CDS_position             3119552\n",
      "Protein_position         3119552\n",
      "Amino_acids                    0\n",
      "Codons                         0\n",
      "Existing_variation             0\n",
      "IND                            0\n",
      "ZYG                            0\n",
      "REF_ALLELE                     0\n",
      "IMPACT                         0\n",
      "DISTANCE                 2771494\n",
      "STRAND                         0\n",
      "FLAGS                          0\n",
      "VARIANT_CLASS                  0\n",
      "SYMBOL                         0\n",
      "SYMBOL_SOURCE                  0\n",
      "HGNC_ID                        0\n",
      "BIOTYPE                        0\n",
      "CANONICAL                      0\n",
      "MANE_SELECT                    0\n",
      "MANE_PLUS_CLINICAL             0\n",
      "TSL                            0\n",
      "APPRIS                         0\n",
      "CCDS                           0\n",
      "ENSP                           0\n",
      "SWISSPROT                      0\n",
      "TREMBL                         0\n",
      "UNIPARC                        0\n",
      "UNIPROT_ISOFORM                0\n",
      "REFSEQ_MATCH                   0\n",
      "SOURCE                         0\n",
      "REFSEQ_OFFSET                  0\n",
      "GIVEN_REF                      0\n",
      "USED_REF                       0\n",
      "BAM_EDIT                       0\n",
      "GENE_PHENO                     0\n",
      "SIFT                           0\n",
      "PolyPhen                       0\n",
      "EXON                           0\n",
      "INTRON                         0\n",
      "DOMAINS                        0\n",
      "miRNA                          0\n",
      "HGVSc                          0\n",
      "HGVSp                          0\n",
      "HGVS_OFFSET                    0\n",
      "gnomADg_AF                 20143\n",
      "gnomADg_AFR_AF             20144\n",
      "gnomADg_AMI_AF             20176\n",
      "gnomADg_AMR_AF             20144\n",
      "gnomADg_ASJ_AF             20145\n",
      "gnomADg_EAS_AF             20148\n",
      "gnomADg_FIN_AF             20145\n",
      "gnomADg_MID_AF             20212\n",
      "gnomADg_NFE_AF             20143\n",
      "gnomADg_OTH_AF             20149\n",
      "gnomADg_SAS_AF             20146\n",
      "MAX_AF                     20072\n",
      "MAX_AF_POPS                    0\n",
      "CLIN_SIG                       0\n",
      "SOMATIC                        0\n",
      "PHENO                          0\n",
      "PUBMED                         0\n",
      "MOTIF_NAME                     0\n",
      "MOTIF_POS                      0\n",
      "HIGH_INF_POS                   0\n",
      "MOTIF_SCORE_CHANGE             0\n",
      "TRANSCRIPTION_FACTORS          0\n",
      "chromosome                     0\n",
      "chrCoordinate                  0\n",
      "SIFTprediction                 0\n",
      "SIFTscore                3130032\n",
      "PolyPhenprediction             0\n",
      "PolyPhenscore            3130350\n",
      "EXONnumber               3061218\n",
      "INTRONnumber             1451661\n",
      "chr                            0\n",
      "dtype: int64\n",
      "0          1_000017407_G_A\n",
      "1          1_000061987_A_G\n",
      "2          1_000061989_G_C\n",
      "3          1_000102951_C_T\n",
      "4          1_000133129_G_A\n",
      "                ...       \n",
      "2950092    X_156024517_C_T\n",
      "2950093    X_156025116_G_A\n",
      "2950094    X_156025186_G_C\n",
      "2950095    X_156025429_T_C\n",
      "2950096    X_156029856_C_G\n",
      "Name: keyID38, Length: 2950097, dtype: object\n",
      "0          1_000013273_G_C\n",
      "1          1_000016298_C_T\n",
      "2          1_000017365_C_G\n",
      "3          1_000017375_A_G\n",
      "4          1_000017385_G_A\n",
      "                ...       \n",
      "3138656    X_156021847_A_G\n",
      "3138657    X_156021966_A_G\n",
      "3138658    X_156025041_C_T\n",
      "3138659    X_156025116_G_A\n",
      "3138660    X_156025429_T_C\n",
      "Name: keyID38, Length: 3138661, dtype: object\n"
     ]
    }
   ],
   "source": [
    "indel12['chr'] = indel12['chromosome'].str.strip('chr')\n",
    "snp12['chr'] = snp12['chromosome'].str.strip('chr')\n",
    "indel63['chr'] = indel63['chromosome'].str.strip('chr')\n",
    "snp63['chr'] = snp63['chromosome'].str.strip('chr')\n",
    "\n",
    "## QC\n",
    "checkColumnValue(indel12, 'chr')\n",
    "checkColumnValue(snp12, 'chr')\n",
    "\n",
    "##  note snp ref alt  would have - value for missing if there were any missing but there are no mising values\n",
    "checkColumnValue(snp63, 'Allele')\n",
    "checkColumnValue(snp63, 'REF_ALLELE')\n",
    "checkColumnValue(snp12, 'Allele')\n",
    "checkColumnValue(snp12, 'REF_ALLELE')\n",
    "\n",
    "checkColumnValue(indel12, 'Allele')\n",
    "checkColumnValue(indel12, 'REF_ALLELE') \n",
    "checkColumnValue(indel63, 'Allele')\n",
    "checkColumnValue(indel63, 'REF_ALLELE')\n",
    "\n",
    "describeMe(snp12)\n",
    "describeMe(snp63)\n",
    "\n",
    "# confirmed on missing values for chrCoordinate in snp dfs\n",
    "\n",
    "# duplicate  chr Corrdinate column and turnn from int  back into string\n",
    "snp12['chrCoordinate-str'] = snp12['chrCoordinate'].astype(str)\n",
    "snp63['chrCoordinate-str'] = snp63['chrCoordinate'].astype(str)\n",
    "\n",
    "# format the chr-str columns to have 9 characters filled with leading 0s\n",
    "snp12['chrCoordinate-str'] = snp12['chrCoordinate-str'].str.zfill(9)\n",
    "snp63['chrCoordinate-str'] = snp63['chrCoordinate-str'].str.zfill(9)\n",
    "\n",
    "# create keyID column for snp only\n",
    "snp12['keyID38'] = snp12['chr'].astype(str) + \"_\" +  snp12['chrCoordinate-str'] +\\\n",
    "      \"_\" + snp12['REF_ALLELE'] + \"_\" + snp12['Allele']\n",
    "\n",
    "snp63['keyID38'] = snp63['chr'].astype(str) + \"_\" +  snp63['chrCoordinate-str'] +\\\n",
    "        \"_\" + snp63['REF_ALLELE'] + \"_\" + snp63['Allele']\n",
    "\n",
    "# QC\n",
    "print(snp12['keyID38'])\n",
    "print(snp63['keyID38'])\n",
    "\n",
    "# drop 'chrCoordinate-str' column from snp12 and snp63\n",
    "snp12.drop(columns=['chrCoordinate-str'], inplace=True)\n",
    "snp63.drop(columns=['chrCoordinate-str'], inplace=True)\n",
    "\n",
    "# drop 'chr-str' column from snp12 and snp63\n",
    "#snp12.drop(columns=['chr-str'], inplace=True)\n",
    "#snp63.drop(columns=['chr-str'], inplace=True)\n",
    "\n",
    "# check # of non-coding snps using  Amino_acids column  \"-\"\n",
    "# checkColumnValue(snp12, 'Amino_acids')\n",
    "# checkColumnValue(snp63, 'Amino_acids')\n",
    "# checkColumnValue(indel12, 'Amino_acids')\n",
    "# checkColumnValue(indel63, 'Amino_acids')\n",
    "\n",
    "# Check if GIVEN_REF matches USED_REF column, create a new column GIVEN_REF_MATCH based on the comparison\n",
    "indel12['GIVEN_REF_MATCH'] = indel12['GIVEN_REF'] == indel12['USED_REF']\n",
    "snp12['GIVEN_REF_MATCH'] = snp12['GIVEN_REF'] == snp12['USED_REF']\n",
    "indel63['GIVEN_REF_MATCH'] = indel63['GIVEN_REF'] == indel63['USED_REF']\n",
    "snp63['GIVEN_REF_MATCH'] = snp63['GIVEN_REF'] == snp63['USED_REF']\n",
    "\n",
    "\n",
    "# QC check # of rows where GIVEN_REF matches USED_REF\n",
    "# checkColumnValue(indel12, 'GIVEN_REF_MATCH')\n",
    "# checkColumnValue(snp12, 'GIVEN_REF_MATCH')\n",
    "# checkColumnValue(indel63, 'GIVEN_REF_MATCH')\n",
    "# checkColumnValue(snp63, 'GIVEN_REF_MATCH')\n",
    "\n",
    "# save given ref false\n",
    "# indel12_false = indel12[indel12['GIVEN_REF_MATCH'] == False]\n",
    "# snp12_false = snp12[snp12['GIVEN_REF_MATCH'] == False]\n",
    "# indel63_false = indel63[indel63['GIVEN_REF_MATCH'] == False]\n",
    "# snp63_false = snp63[snp63['GIVEN_REF_MATCH'] == False]\n",
    "\n",
    "# TODO what genes are these False values associated with?\n",
    "# checkColumnValue(indel12_false, 'SYMBOL')\n",
    "# checkColumnValue(snp12_false, 'SYMBOL')\n",
    "# checkColumnValue(indel63_false, 'SYMBOL')\n",
    "# checkColumnValue(snp63_false, 'SYMBOL')\n",
    "\n",
    "\n",
    "\n",
    "# # save cleaned files\n",
    "indel12.to_csv('1_clean_vep_output-indel-cad12-2.csv', index=False)\n",
    "snp12.to_csv('1_clean_vep_output-snp-cad12-2.csv', index=False)\n",
    "indel63.to_csv('1_clean_vep_output-indel-cad63-2.csv', index=False)\n",
    "snp63.to_csv('1_clean_vep_output-snp-cad63-v2.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fd51c3c0-e54f-4621-a0d9-b492b3ccbe79",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
      "       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN_REF_MATCH'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(indel12.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba0770f4-b66d-470b-8f27-e7f1985253b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
      "       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'EXONnumber', 'INTRONnumber', 'chr', 'keyID38', 'GIVEN_REF_MATCH'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "print(snp12.columns)"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": 3
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
