{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "134b29bd-c4c4-4b4b-b12b-4165c7e877cb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "\n",
    "# 2_summarize_cleaned_vep_output.py\n",
    "\n",
    "# input: 1_clean_...py output\n",
    "# output: excel sheets of summary dataframes\n",
    "\n",
    "\"\"\"\n",
    "save_colValueCount('Feature_type', dfs, dfnames, path, True):\n",
    "# saved in 3 as colname-checkValueCount.xlsx #####\n",
    "    Feature_type\n",
    "    Consequence\n",
    "    ZYG\n",
    "    IMPACT\n",
    "    VARIANT_CLASS\n",
    "    SYMBOL\n",
    "    SYMBOL_SOURCE\n",
    "    BIOTYPE\n",
    "    SIFT\n",
    "    PolyPhen\n",
    "    CLIN_SIG\n",
    "    PHENO\n",
    "    SOMATIC\n",
    "\n",
    "# new in v3 not in v2 ##############################\n",
    "    chr\n",
    "    Allele\n",
    "    REF_ALLELE\n",
    "    Amino_acids\n",
    "    Gene\n",
    "    STRAND\n",
    "    FLAGS\n",
    "    SWISSPROT\n",
    "    GENE_PHENO\n",
    "    MAX_AF_POPS\n",
    "    HIGH_INF_POS\n",
    "    MOTIF_NAME\n",
    "\n",
    "\"\"\"\n",
    "\n",
    "# %%\n",
    "from all_funx import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "26827c7a-a37d-4b59-8f87-e5f85655b815",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "<ipython-input-18-e0a276bbd151>:10: DtypeWarning: Columns (8,9,78) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  indel12 = pd.read_csv('1_clean_vep_output-indel-cad12-v2.csv')\n",
      "<ipython-input-18-e0a276bbd151>:12: DtypeWarning: Columns (82) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  snp12 = pd.read_csv('1_clean_vep_output-snp-cad12-v2.csv')\n",
      "<ipython-input-18-e0a276bbd151>:14: DtypeWarning: Columns (8,9,78) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  indel63 = pd.read_csv('1_clean_vep_output-indel-cad63-v2.csv')\n",
      "<ipython-input-18-e0a276bbd151>:16: DtypeWarning: Columns (82) have mixed types. Specify dtype option on import or set low_memory=False.\n",
      "  snp63 = pd.read_csv('1_clean_vep_output-snp-cad63-v2.csv')\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "# import dfs from 1_clean_excel_summary.py\n",
    "path  = \"/Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/\"\n",
    "\n",
    "# indel12 = pd.read_csv('summarize-output-1-indel-cad12.csv')\n",
    "# snp12 = pd.read_csv('summarize-output-1-snp-cad12.csv')\n",
    "# indel63 = pd.read_csv('summarize-output-1-indel-cad63.csv')\n",
    "# snp63 = pd.read_csv('summarize-output-1-snp-cad63.csv')\n",
    "\n",
    "indel12 = pd.read_csv('1_clean_vep_output-indel-cad12-v2.csv')\n",
    "\n",
    "snp12 = pd.read_csv('1_clean_vep_output-snp-cad12-v2.csv')\n",
    "\n",
    "indel63 = pd.read_csv('1_clean_vep_output-indel-cad63-v2.csv')\n",
    "\n",
    "snp63 = pd.read_csv('1_clean_vep_output-snp-cad63-v2.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cc4d1a25-990b-4a60-9731-503085190eee",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "def save_colValueCount(colname, dfs, dfnames, PATH, returnListDfs):\n",
    "    #  funxtion to save value counts of a column in a multi-sheet excel\n",
    "    #  dfs: list of dataframes\n",
    "    #  dfnames: list of names of dataframes\n",
    "    #  colname: name of column to check value counts\n",
    "    #  PATH: path to save excel\n",
    "    # return: True or False to return list of dfs\n",
    "    print(\"\\n################ \", colname, \" ################\\n\")\n",
    "    output =  PATH + colname + '-checkValueCount' + '.xlsx'\n",
    "    return_results = []\n",
    "    with pd.ExcelWriter(output) as writer:\n",
    "        for x, df in enumerate(dfs):\n",
    "            dfname = dfnames[x]\n",
    "            colval = checkColumnValue(df, colname)\n",
    "            print()\n",
    "            print(dfname)\n",
    "            print(colval)\n",
    "            colval.to_excel(writer, sheet_name = dfname)\n",
    "            return_results.append(colval)\n",
    "    print()\n",
    "    print('saved multi-sheet excel as ',  output)\n",
    "    print('return list of dfs length: ', len(return_results))\n",
    "    if returnListDfs:\n",
    "        print('returning list of dfs..')\n",
    "        return return_results\n",
    "    else:\n",
    "        print('returning None..')\n",
    "        print()\n",
    "        return None"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "caf0cf12-963e-477e-9eb3-f5fea06ca80a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "dfs = [indel12, indel63, snp12, snp63]\n",
    "dfnames = [\"indel12\", \"indel63\", \"snp12\", \"snp63\"]\n",
    "\n",
    "indel12_false = indel12[indel12['GIVEN_REF_MATCH'] == False]\n",
    "snp12_false = snp12[snp12['GIVEN_REF_MATCH'] == False]\n",
    "indel63_false = indel63[indel63['GIVEN_REF_MATCH'] == False]\n",
    "snp63_false = snp63[snp63['GIVEN_REF_MATCH'] == False]\n",
    "\n",
    "dfs_false = [indel12_false, indel63_false, snp12_false, snp63_false]\n",
    "dfnames_false = [\"indel12_false\", \"indel63_false\", \"snp12_false\", \"snp63_false\"]\n",
    "\n",
    "snps = [snp12, snp63]\n",
    "snpnames = [\"snp12\", \"snp63\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0f48103-579b-4397-9453-540b888c3b28",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SYMBOL  ################\n",
      "\n",
      "\n",
      "indel12_false\n",
      "        Count  count\n",
      "0      PABPC3      2\n",
      "1       GP1BA      2\n",
      "2        DSPP      2\n",
      "3      TRIOBP      2\n",
      "4      MUC5AC      2\n",
      "5       OR4L1      1\n",
      "6       NIPA1      1\n",
      "7     TMEM121      1\n",
      "8        RIN3      1\n",
      "9      ACTRT2      1\n",
      "10      TDRD3      1\n",
      "11      MEF2A      1\n",
      "12      ORAI1      1\n",
      "13      ROBO3      1\n",
      "14      SRSF8      1\n",
      "15      FOLR3      1\n",
      "16      MESP2      1\n",
      "17      ZFPM1      1\n",
      "18       IST1      1\n",
      "19      OR4X1      1\n",
      "20   C19orf33      1\n",
      "21       DLL3      1\n",
      "22     ZNF284      1\n",
      "23    DEFB132      1\n",
      "24      NCOA3      1\n",
      "25      CLIC6      1\n",
      "26    COL18A1      1\n",
      "27       SRRD      1\n",
      "28   CDC42EP1      1\n",
      "29   APOBEC3H      1\n",
      "30  SUPT20HL1      1\n",
      "31   KRTAP5-7      1\n",
      "32     ZNF408      1\n",
      "33      GPSM2      1\n",
      "34       TET2      1\n",
      "35      NBPF8      1\n",
      "36       FMN2      1\n",
      "37       ALLC      1\n",
      "38      SCRN3      1\n",
      "39     SCYGR4      1\n",
      "40     SCYGR8      1\n",
      "41     SPATA3      1\n",
      "42     YEATS2      1\n",
      "43      TACC3      1\n",
      "44        HTT      1\n",
      "45     ADRA2C      1\n",
      "46       REST      1\n",
      "47     MAP3K1      1\n",
      "48      SMPD1      1\n",
      "49      RRP36      1\n",
      "50      TTBK1      1\n",
      "51      FNDC1      1\n",
      "52        TBP      1\n",
      "53        ZAN      1\n",
      "54       IRF5      1\n",
      "55     OR2A14      1\n",
      "56    DYNC2I1      1\n",
      "57       ODF1      1\n",
      "58    SMARCA2      1\n",
      "59     SCART1      1\n",
      "60     OR52D1      1\n",
      "61      GPR50      1\n",
      "\n",
      "indel63_false\n",
      "        Count  count\n",
      "0        DSPP      5\n",
      "1      SCYGR8      2\n",
      "2      TRIOBP      2\n",
      "3       GP1BA      2\n",
      "4      SCYGR2      2\n",
      "5        RIN3      1\n",
      "6       MESP2      1\n",
      "7      LRRC49      1\n",
      "8     SLC24A1      1\n",
      "9     GOLGA8S      1\n",
      "10    TMEM121      1\n",
      "11       INF2      1\n",
      "12      CLCA4      1\n",
      "13     PERCC1      1\n",
      "14      RBM25      1\n",
      "15      OR4L1      1\n",
      "16     FBRSL1      1\n",
      "17      ORAI1      1\n",
      "18      ASCL1      1\n",
      "19      MUC19      1\n",
      "20   C12orf60      1\n",
      "21      MEF2A      1\n",
      "22      ZFPM1      1\n",
      "23       IST1      1\n",
      "24    NPIPB15      1\n",
      "25      APLP2      1\n",
      "26      MPRIP      1\n",
      "27     TXNDC2      1\n",
      "28    CABLES1      1\n",
      "29     ZNF714      1\n",
      "30     ZNF738      1\n",
      "31     ZNF283      1\n",
      "32     ZNF284      1\n",
      "33    DEFB132      1\n",
      "34      NCOA3      1\n",
      "35       OGFR      1\n",
      "36    COL18A1      1\n",
      "37       SRRD      1\n",
      "38       ATN1      1\n",
      "39      FOLR3      1\n",
      "40      ROBO3      1\n",
      "41     PTPN18      1\n",
      "42   CDKN2AIP      1\n",
      "43      LRIT3      1\n",
      "44        HTT      1\n",
      "45     FGFRL1      1\n",
      "46     YEATS2      1\n",
      "47     SPATA3      1\n",
      "48    SULT1C3      1\n",
      "49       VCAN      1\n",
      "50       ALLC      1\n",
      "51       FMN2      1\n",
      "52      IGFN1      1\n",
      "53     AXDND1      1\n",
      "54      SPRR5      1\n",
      "55      NBPF8      1\n",
      "56     MAP3K1      1\n",
      "57    PCDHB10      1\n",
      "58      GPSM2      1\n",
      "59  NIPSNAP3A      1\n",
      "60      OR4X1      1\n",
      "61    PPFIBP2      1\n",
      "62     MUC5AC      1\n",
      "63      FANK1      1\n",
      "64    CACNA1B      1\n",
      "65      PTGS1      1\n",
      "66     TGFBR1      1\n",
      "67      TTBK1      1\n",
      "68    DYNC2I1      1\n",
      "69     OR2A14      1\n",
      "70      TCAF2      1\n",
      "71       IRF5      1\n",
      "72        TBP      1\n",
      "73      FNDC1      1\n",
      "74      GPR50      1\n",
      "\n",
      "snp12_false\n",
      "        Count  count\n",
      "0    GAS2L1P2      3\n",
      "1      FRG2EP      3\n",
      "2    RPL31P11      2\n",
      "3   LINC02606      1\n",
      "4     ZNF815P      1\n",
      "5  MIR124-1HG      1\n",
      "6   LINC00705      1\n",
      "7   LINC00908      1\n",
      "8    SNORA71D      1\n",
      "9     BMS1P20      1\n",
      "\n",
      "snp63_false\n",
      "         Count  count\n",
      "0     GAS2L1P2      3\n",
      "1       FRG2EP      3\n",
      "2      BMS1P20      3\n",
      "3     RPL31P11      2\n",
      "4     MLLT10P1      2\n",
      "5    LINC02606      1\n",
      "6      PDIA3P1      1\n",
      "7      ZNF815P      1\n",
      "8   MIR124-1HG      1\n",
      "9    LINC00705      1\n",
      "10   LINC00908      1\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SYMBOL-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('SYMBOL', dfs_false, dfnames_false, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "380d9171-9296-4d02-ab62-c4f910f97de6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  Feature_type  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "               Count   count\n",
      "0         Transcript  395760\n",
      "1                  -  131184\n",
      "2  RegulatoryFeature   18037\n",
      "3       MotifFeature    1123\n",
      "\n",
      "indel63\n",
      "               Count   count\n",
      "0         Transcript  402814\n",
      "1                  -  141105\n",
      "2  RegulatoryFeature   17376\n",
      "3       MotifFeature    1058\n",
      "\n",
      "snp12\n",
      "               Count    count\n",
      "0         Transcript  2014932\n",
      "1                  -   817842\n",
      "2  RegulatoryFeature   108756\n",
      "3       MotifFeature     8567\n",
      "\n",
      "snp63\n",
      "               Count    count\n",
      "0         Transcript  2131610\n",
      "1                  -   889161\n",
      "2  RegulatoryFeature   109385\n",
      "3       MotifFeature     8505\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Feature_type-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n",
      "\n",
      "snp12\n",
      "               Count    count\n",
      "0         Transcript  2014932\n",
      "1                  -   817842\n",
      "2  RegulatoryFeature   108756\n",
      "3       MotifFeature     8567\n",
      "\n",
      "snp63\n",
      "               Count    count\n",
      "0         Transcript  2131610\n",
      "1                  -   889161\n",
      "2  RegulatoryFeature   109385\n",
      "3       MotifFeature     8505\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Feature_type-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "# save value counts of Feature_type column\n",
    "save_colValueCount('Feature_type', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9ba30de0-3d69-4bae-8af5-2380a759c3f6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  Consequence  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                                                Count   count\n",
      "0                                      intron_variant  225856\n",
      "1                                  intergenic_variant  131184\n",
      "2        intron_variant,non_coding_transcript_variant   78610\n",
      "3                               upstream_gene_variant   41634\n",
      "4                             downstream_gene_variant   37520\n",
      "5                           regulatory_region_variant   18037\n",
      "6                                 3_prime_UTR_variant    5798\n",
      "7                  non_coding_transcript_exon_variant    2932\n",
      "8               intron_variant,NMD_transcript_variant    1280\n",
      "9                             TF_binding_site_variant    1103\n",
      "10                                5_prime_UTR_variant     665\n",
      "11  splice_polypyrimidine_tract_variant,intron_var...     323\n",
      "12  splice_polypyrimidine_tract_variant,splice_reg...     182\n",
      "13                                   inframe_deletion     172\n",
      "14                                  inframe_insertion     157\n",
      "15                                 frameshift_variant     144\n",
      "16  splice_polypyrimidine_tract_variant,intron_var...     130\n",
      "17  splice_polypyrimidine_tract_variant,splice_reg...     128\n",
      "18               splice_region_variant,intron_variant      39\n",
      "19         3_prime_UTR_variant,NMD_transcript_variant      31\n",
      "20         splice_donor_region_variant,intron_variant      30\n",
      "21              TFBS_ablation,TF_binding_site_variant      20\n",
      "22  splice_region_variant,non_coding_transcript_ex...      18\n",
      "23  splice_donor_region_variant,intron_variant,non...      16\n",
      "24       splice_donor_5th_base_variant,intron_variant      12\n",
      "25  splice_region_variant,intron_variant,non_codin...      10\n",
      "26  splice_acceptor_variant,non_coding_transcript_...       6\n",
      "27  splice_region_variant,non_coding_transcript_va...       6\n",
      "28                     stop_gained,frameshift_variant       6\n",
      "29           frameshift_variant,splice_region_variant       5\n",
      "30  splice_donor_5th_base_variant,intron_variant,n...       4\n",
      "31  splice_polypyrimidine_tract_variant,splice_reg...       3\n",
      "32     frameshift_variant,splice_donor_region_variant       3\n",
      "33  frameshift_variant,start_lost,start_retained_v...       3\n",
      "34  splice_donor_variant,splice_donor_5th_base_var...       3\n",
      "35  splice_acceptor_variant,splice_polypyrimidine_...       3\n",
      "36  splice_acceptor_variant,non_coding_transcript_...       2\n",
      "37                            coding_sequence_variant       2\n",
      "38  splice_donor_variant,splice_donor_5th_base_var...       2\n",
      "39  frameshift_variant,splice_region_variant,intro...       2\n",
      "40                               mature_miRNA_variant       2\n",
      "41                               splice_donor_variant       2\n",
      "42  splice_donor_variant,splice_donor_region_varia...       1\n",
      "43  splice_acceptor_variant,splice_polypyrimidine_...       1\n",
      "44        splice_acceptor_variant,5_prime_UTR_variant       1\n",
      "45  splice_donor_variant,splice_donor_region_varia...       1\n",
      "46            inframe_insertion,splice_region_variant       1\n",
      "47         splice_acceptor_variant,frameshift_variant       1\n",
      "48  splice_donor_variant,splice_donor_5th_base_var...       1\n",
      "49  splice_acceptor_variant,coding_sequence_varian...       1\n",
      "50                                        stop_gained       1\n",
      "51                      frameshift_variant,start_lost       1\n",
      "52                            splice_acceptor_variant       1\n",
      "53          splice_region_variant,5_prime_UTR_variant       1\n",
      "54          stop_retained_variant,3_prime_UTR_variant       1\n",
      "55  splice_acceptor_variant,non_coding_transcript_...       1\n",
      "56  splice_donor_variant,splice_donor_5th_base_var...       1\n",
      "57             inframe_deletion,splice_region_variant       1\n",
      "58  splice_donor_variant,non_coding_transcript_var...       1\n",
      "59  splice_acceptor_variant,5_prime_UTR_variant,in...       1\n",
      "60  start_lost,start_retained_variant,5_prime_UTR_...       1\n",
      "\n",
      "indel63\n",
      "                                                Count   count\n",
      "0                                      intron_variant  229123\n",
      "1                                  intergenic_variant  141105\n",
      "2        intron_variant,non_coding_transcript_variant   84172\n",
      "3                               upstream_gene_variant   40239\n",
      "4                             downstream_gene_variant   36914\n",
      "5                           regulatory_region_variant   17376\n",
      "6                                 3_prime_UTR_variant    6059\n",
      "7                  non_coding_transcript_exon_variant    2859\n",
      "8               intron_variant,NMD_transcript_variant    1221\n",
      "9                             TF_binding_site_variant    1034\n",
      "10                                5_prime_UTR_variant     706\n",
      "11  splice_polypyrimidine_tract_variant,intron_var...     395\n",
      "12  splice_region_variant,splice_polypyrimidine_tr...     201\n",
      "13                                   inframe_deletion     175\n",
      "14                                  inframe_insertion     150\n",
      "15  splice_polypyrimidine_tract_variant,intron_var...     130\n",
      "16  splice_region_variant,splice_polypyrimidine_tr...     123\n",
      "17                                 frameshift_variant     120\n",
      "18               splice_region_variant,intron_variant      43\n",
      "19         splice_donor_region_variant,intron_variant      30\n",
      "20         3_prime_UTR_variant,NMD_transcript_variant      25\n",
      "21              TFBS_ablation,TF_binding_site_variant      24\n",
      "22  splice_donor_region_variant,intron_variant,non...      16\n",
      "23  splice_region_variant,non_coding_transcript_ex...      16\n",
      "24       splice_donor_5th_base_variant,intron_variant      12\n",
      "25  splice_acceptor_variant,non_coding_transcript_...       7\n",
      "26  splice_region_variant,non_coding_transcript_va...       6\n",
      "27  splice_donor_variant,splice_donor_5th_base_var...       5\n",
      "28  splice_region_variant,intron_variant,non_codin...       5\n",
      "29                     stop_gained,frameshift_variant       5\n",
      "30           frameshift_variant,splice_region_variant       4\n",
      "31                               mature_miRNA_variant       4\n",
      "32     frameshift_variant,splice_donor_region_variant       3\n",
      "33  splice_donor_variant,non_coding_transcript_var...       3\n",
      "34                       frameshift_variant,stop_lost       3\n",
      "35  splice_donor_5th_base_variant,intron_variant,n...       3\n",
      "36  splice_region_variant,splice_polypyrimidine_tr...       3\n",
      "37  splice_acceptor_variant,splice_polypyrimidine_...       2\n",
      "38  splice_acceptor_variant,splice_polypyrimidine_...       2\n",
      "39  splice_donor_variant,non_coding_transcript_exo...       2\n",
      "40         splice_acceptor_variant,frameshift_variant       2\n",
      "41          splice_region_variant,5_prime_UTR_variant       2\n",
      "42                            coding_sequence_variant       2\n",
      "43  splice_acceptor_variant,non_coding_transcript_...       2\n",
      "44           frameshift_variant,stop_retained_variant       1\n",
      "45  splice_acceptor_variant,coding_sequence_varian...       1\n",
      "46            inframe_insertion,splice_region_variant       1\n",
      "47        splice_acceptor_variant,5_prime_UTR_variant       1\n",
      "48  splice_donor_variant,splice_donor_region_varia...       1\n",
      "49  splice_donor_variant,splice_donor_region_varia...       1\n",
      "50  frameshift_variant,splice_region_variant,intro...       1\n",
      "51  splice_donor_variant,splice_donor_5th_base_var...       1\n",
      "52                               splice_donor_variant       1\n",
      "53             inframe_deletion,splice_region_variant       1\n",
      "54                           protein_altering_variant       1\n",
      "55  splice_donor_variant,splice_donor_region_varia...       1\n",
      "56  splice_polypyrimidine_tract_variant,intron_var...       1\n",
      "57          stop_retained_variant,3_prime_UTR_variant       1\n",
      "58                            splice_acceptor_variant       1\n",
      "59                      frameshift_variant,start_lost       1\n",
      "60  splice_donor_variant,splice_donor_5th_base_var...       1\n",
      "61  splice_donor_variant,splice_donor_region_varia...       1\n",
      "62  frameshift_variant,start_lost,start_retained_v...       1\n",
      "63  start_lost,start_retained_variant,5_prime_UTR_...       1\n",
      "\n",
      "snp12\n",
      "                                                Count    count\n",
      "0                                      intron_variant  1098898\n",
      "1                                  intergenic_variant   817842\n",
      "2        intron_variant,non_coding_transcript_variant   471945\n",
      "3                               upstream_gene_variant   185565\n",
      "4                             downstream_gene_variant   172183\n",
      "5                           regulatory_region_variant   108756\n",
      "6                  non_coding_transcript_exon_variant    26508\n",
      "7                                 3_prime_UTR_variant    25531\n",
      "8                                  synonymous_variant    10240\n",
      "9                                    missense_variant     8848\n",
      "10                            TF_binding_site_variant     8567\n",
      "11              intron_variant,NMD_transcript_variant     6700\n",
      "12                                5_prime_UTR_variant     3911\n",
      "13  splice_polypyrimidine_tract_variant,intron_var...     1282\n",
      "14  splice_region_variant,splice_polypyrimidine_tr...      755\n",
      "15               splice_region_variant,intron_variant      341\n",
      "16  splice_polypyrimidine_tract_variant,intron_var...      332\n",
      "17         splice_donor_region_variant,intron_variant      265\n",
      "18  splice_region_variant,non_coding_transcript_ex...      234\n",
      "19           splice_region_variant,synonymous_variant      216\n",
      "20  splice_region_variant,splice_polypyrimidine_tr...      205\n",
      "21             missense_variant,splice_region_variant      184\n",
      "22         3_prime_UTR_variant,NMD_transcript_variant      173\n",
      "23  splice_donor_region_variant,intron_variant,non...      119\n",
      "24                                        stop_gained       76\n",
      "25  splice_donor_variant,non_coding_transcript_var...       75\n",
      "26  splice_region_variant,intron_variant,non_codin...       75\n",
      "27       splice_donor_5th_base_variant,intron_variant       58\n",
      "28  splice_donor_5th_base_variant,intron_variant,n...       45\n",
      "29  splice_acceptor_variant,non_coding_transcript_...       45\n",
      "30          splice_region_variant,5_prime_UTR_variant       40\n",
      "31                                          stop_lost       18\n",
      "32                            splice_acceptor_variant       13\n",
      "33                               splice_donor_variant       12\n",
      "34                                         start_lost        8\n",
      "35                              stop_retained_variant        7\n",
      "36                               mature_miRNA_variant        6\n",
      "37          splice_region_variant,3_prime_UTR_variant        4\n",
      "38                   start_lost,splice_region_variant        2\n",
      "39  splice_region_variant,splice_polypyrimidine_tr...        2\n",
      "40     splice_acceptor_variant,NMD_transcript_variant        2\n",
      "41         5_prime_UTR_variant,NMD_transcript_variant        2\n",
      "42            missense_variant,NMD_transcript_variant        2\n",
      "43  splice_region_variant,intron_variant,NMD_trans...        1\n",
      "44  splice_polypyrimidine_tract_variant,intron_var...        1\n",
      "45                  stop_gained,splice_region_variant        1\n",
      "46  splice_region_variant,synonymous_variant,NMD_t...        1\n",
      "47  splice_region_variant,3_prime_UTR_variant,NMD_...        1\n",
      "\n",
      "snp63\n",
      "                                                Count    count\n",
      "0                                      intron_variant  1165581\n",
      "1                                  intergenic_variant   889161\n",
      "2        intron_variant,non_coding_transcript_variant   511049\n",
      "3                               upstream_gene_variant   190251\n",
      "4                             downstream_gene_variant   176916\n",
      "5                           regulatory_region_variant   109385\n",
      "6                                 3_prime_UTR_variant    27160\n",
      "7                  non_coding_transcript_exon_variant    26802\n",
      "8                                  synonymous_variant     9941\n",
      "9                                    missense_variant     8700\n",
      "10                            TF_binding_site_variant     8505\n",
      "11              intron_variant,NMD_transcript_variant     6764\n",
      "12                                5_prime_UTR_variant     3940\n",
      "13  splice_polypyrimidine_tract_variant,intron_var...     1238\n",
      "14  splice_region_variant,splice_polypyrimidine_tr...      774\n",
      "15               splice_region_variant,intron_variant      351\n",
      "16  splice_polypyrimidine_tract_variant,intron_var...      316\n",
      "17         splice_donor_region_variant,intron_variant      267\n",
      "18  splice_region_variant,non_coding_transcript_ex...      235\n",
      "19  splice_region_variant,splice_polypyrimidine_tr...      213\n",
      "20           splice_region_variant,synonymous_variant      182\n",
      "21             missense_variant,splice_region_variant      178\n",
      "22         3_prime_UTR_variant,NMD_transcript_variant      135\n",
      "23  splice_donor_region_variant,intron_variant,non...      104\n",
      "24  splice_donor_variant,non_coding_transcript_var...       75\n",
      "25  splice_region_variant,intron_variant,non_codin...       73\n",
      "26                                        stop_gained       65\n",
      "27       splice_donor_5th_base_variant,intron_variant       61\n",
      "28          splice_region_variant,5_prime_UTR_variant       45\n",
      "29  splice_donor_5th_base_variant,intron_variant,n...       44\n",
      "30  splice_acceptor_variant,non_coding_transcript_...       42\n",
      "31                               splice_donor_variant       21\n",
      "32                            splice_acceptor_variant       18\n",
      "33                                          stop_lost       15\n",
      "34                                         start_lost       12\n",
      "35                               mature_miRNA_variant        9\n",
      "36                              stop_retained_variant        8\n",
      "37          splice_region_variant,3_prime_UTR_variant        5\n",
      "38  splice_region_variant,splice_polypyrimidine_tr...        4\n",
      "39            missense_variant,NMD_transcript_variant        3\n",
      "40                   start_lost,splice_region_variant        2\n",
      "41         5_prime_UTR_variant,NMD_transcript_variant        2\n",
      "42     splice_acceptor_variant,NMD_transcript_variant        2\n",
      "43  splice_polypyrimidine_tract_variant,intron_var...        2\n",
      "44                  stop_gained,splice_region_variant        1\n",
      "45  splice_region_variant,intron_variant,NMD_trans...        1\n",
      "46  splice_region_variant,synonymous_variant,NMD_t...        1\n",
      "47  splice_region_variant,3_prime_UTR_variant,NMD_...        1\n",
      "48  incomplete_terminal_codon_variant,coding_seque...        1\n",
      "\n",
      "snp12\n",
      "                                                Count    count\n",
      "0                                      intron_variant  1098898\n",
      "1                                  intergenic_variant   817842\n",
      "2        intron_variant,non_coding_transcript_variant   471945\n",
      "3                               upstream_gene_variant   185565\n",
      "4                             downstream_gene_variant   172183\n",
      "5                           regulatory_region_variant   108756\n",
      "6                  non_coding_transcript_exon_variant    26508\n",
      "7                                 3_prime_UTR_variant    25531\n",
      "8                                  synonymous_variant    10240\n",
      "9                                    missense_variant     8848\n",
      "10                            TF_binding_site_variant     8567\n",
      "11              intron_variant,NMD_transcript_variant     6700\n",
      "12                                5_prime_UTR_variant     3911\n",
      "13  splice_polypyrimidine_tract_variant,intron_var...     1282\n",
      "14  splice_region_variant,splice_polypyrimidine_tr...      755\n",
      "15               splice_region_variant,intron_variant      341\n",
      "16  splice_polypyrimidine_tract_variant,intron_var...      332\n",
      "17         splice_donor_region_variant,intron_variant      265\n",
      "18  splice_region_variant,non_coding_transcript_ex...      234\n",
      "19           splice_region_variant,synonymous_variant      216\n",
      "20  splice_region_variant,splice_polypyrimidine_tr...      205\n",
      "21             missense_variant,splice_region_variant      184\n",
      "22         3_prime_UTR_variant,NMD_transcript_variant      173\n",
      "23  splice_donor_region_variant,intron_variant,non...      119\n",
      "24                                        stop_gained       76\n",
      "25  splice_donor_variant,non_coding_transcript_var...       75\n",
      "26  splice_region_variant,intron_variant,non_codin...       75\n",
      "27       splice_donor_5th_base_variant,intron_variant       58\n",
      "28  splice_donor_5th_base_variant,intron_variant,n...       45\n",
      "29  splice_acceptor_variant,non_coding_transcript_...       45\n",
      "30          splice_region_variant,5_prime_UTR_variant       40\n",
      "31                                          stop_lost       18\n",
      "32                            splice_acceptor_variant       13\n",
      "33                               splice_donor_variant       12\n",
      "34                                         start_lost        8\n",
      "35                              stop_retained_variant        7\n",
      "36                               mature_miRNA_variant        6\n",
      "37          splice_region_variant,3_prime_UTR_variant        4\n",
      "38                   start_lost,splice_region_variant        2\n",
      "39  splice_region_variant,splice_polypyrimidine_tr...        2\n",
      "40     splice_acceptor_variant,NMD_transcript_variant        2\n",
      "41         5_prime_UTR_variant,NMD_transcript_variant        2\n",
      "42            missense_variant,NMD_transcript_variant        2\n",
      "43  splice_region_variant,intron_variant,NMD_trans...        1\n",
      "44  splice_polypyrimidine_tract_variant,intron_var...        1\n",
      "45                  stop_gained,splice_region_variant        1\n",
      "46  splice_region_variant,synonymous_variant,NMD_t...        1\n",
      "47  splice_region_variant,3_prime_UTR_variant,NMD_...        1\n",
      "\n",
      "snp63\n",
      "                                                Count    count\n",
      "0                                      intron_variant  1165581\n",
      "1                                  intergenic_variant   889161\n",
      "2        intron_variant,non_coding_transcript_variant   511049\n",
      "3                               upstream_gene_variant   190251\n",
      "4                             downstream_gene_variant   176916\n",
      "5                           regulatory_region_variant   109385\n",
      "6                                 3_prime_UTR_variant    27160\n",
      "7                  non_coding_transcript_exon_variant    26802\n",
      "8                                  synonymous_variant     9941\n",
      "9                                    missense_variant     8700\n",
      "10                            TF_binding_site_variant     8505\n",
      "11              intron_variant,NMD_transcript_variant     6764\n",
      "12                                5_prime_UTR_variant     3940\n",
      "13  splice_polypyrimidine_tract_variant,intron_var...     1238\n",
      "14  splice_region_variant,splice_polypyrimidine_tr...      774\n",
      "15               splice_region_variant,intron_variant      351\n",
      "16  splice_polypyrimidine_tract_variant,intron_var...      316\n",
      "17         splice_donor_region_variant,intron_variant      267\n",
      "18  splice_region_variant,non_coding_transcript_ex...      235\n",
      "19  splice_region_variant,splice_polypyrimidine_tr...      213\n",
      "20           splice_region_variant,synonymous_variant      182\n",
      "21             missense_variant,splice_region_variant      178\n",
      "22         3_prime_UTR_variant,NMD_transcript_variant      135\n",
      "23  splice_donor_region_variant,intron_variant,non...      104\n",
      "24  splice_donor_variant,non_coding_transcript_var...       75\n",
      "25  splice_region_variant,intron_variant,non_codin...       73\n",
      "26                                        stop_gained       65\n",
      "27       splice_donor_5th_base_variant,intron_variant       61\n",
      "28          splice_region_variant,5_prime_UTR_variant       45\n",
      "29  splice_donor_5th_base_variant,intron_variant,n...       44\n",
      "30  splice_acceptor_variant,non_coding_transcript_...       42\n",
      "31                               splice_donor_variant       21\n",
      "32                            splice_acceptor_variant       18\n",
      "33                                          stop_lost       15\n",
      "34                                         start_lost       12\n",
      "35                               mature_miRNA_variant        9\n",
      "36                              stop_retained_variant        8\n",
      "37          splice_region_variant,3_prime_UTR_variant        5\n",
      "38  splice_region_variant,splice_polypyrimidine_tr...        4\n",
      "39            missense_variant,NMD_transcript_variant        3\n",
      "40                   start_lost,splice_region_variant        2\n",
      "41         5_prime_UTR_variant,NMD_transcript_variant        2\n",
      "42     splice_acceptor_variant,NMD_transcript_variant        2\n",
      "43  splice_polypyrimidine_tract_variant,intron_var...        2\n",
      "44                  stop_gained,splice_region_variant        1\n",
      "45  splice_region_variant,intron_variant,NMD_trans...        1\n",
      "46  splice_region_variant,synonymous_variant,NMD_t...        1\n",
      "47  splice_region_variant,3_prime_UTR_variant,NMD_...        1\n",
      "48  incomplete_terminal_codon_variant,coding_seque...        1\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Consequence-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('Consequence', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "67d86807-d799-43f5-8e6d-373146cf00ae",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  ZYG  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "  Count   count\n",
      "0   HET  396869\n",
      "1   HOM  149235\n",
      "\n",
      "indel63\n",
      "  Count   count\n",
      "0   HET  399640\n",
      "1   HOM  162713\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0   HET  1854809\n",
      "1   HOM  1095288\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0   HET  1935683\n",
      "1   HOM  1202978\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/ZYG-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0   HET  1935683\n",
      "1   HOM  1202978\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/ZYG-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('ZYG', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "62d3a762-ae77-4217-9b55-13ea679b2f40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  IMPACT  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "      Count   count\n",
      "0  MODIFIER  544654\n",
      "1       LOW     903\n",
      "2  MODERATE     351\n",
      "3      HIGH     196\n",
      "\n",
      "indel63\n",
      "      Count   count\n",
      "0  MODIFIER  560839\n",
      "1       LOW     987\n",
      "2  MODERATE     352\n",
      "3      HIGH     175\n",
      "\n",
      "snp12\n",
      "      Count    count\n",
      "0  MODIFIER  2926587\n",
      "1       LOW    14224\n",
      "2  MODERATE     9034\n",
      "3      HIGH      252\n",
      "\n",
      "snp63\n",
      "      Count    count\n",
      "0  MODIFIER  3115660\n",
      "1       LOW    13867\n",
      "2  MODERATE     8881\n",
      "3      HIGH      253\n",
      "\n",
      "snp12\n",
      "      Count    count\n",
      "0  MODIFIER  2926587\n",
      "1       LOW    14224\n",
      "2  MODERATE     9034\n",
      "3      HIGH      252\n",
      "\n",
      "snp63\n",
      "      Count    count\n",
      "0  MODIFIER  3115660\n",
      "1       LOW    13867\n",
      "2  MODERATE     8881\n",
      "3      HIGH      253\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/IMPACT-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('IMPACT', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2c1e70fb-8b0f-439c-83fe-14d7d8a28897",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  VARIANT_CLASS  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                 Count   count\n",
      "0             deletion  268067\n",
      "1            insertion  252293\n",
      "2  sequence_alteration   25744\n",
      "\n",
      "indel63\n",
      "                 Count   count\n",
      "0             deletion  280360\n",
      "1            insertion  259456\n",
      "2  sequence_alteration   22537\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0   SNV  2950097\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0   SNV  3138661\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/VARIANT_CLASS-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n",
      "\n",
      "indel63\n",
      "                 Count   count\n",
      "0             deletion  280360\n",
      "1            insertion  259456\n",
      "2  sequence_alteration   22537\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0   SNV  2950097\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0   SNV  3138661\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/VARIANT_CLASS-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('VARIANT_CLASS', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4a822087-0f0c-471d-a540-11a4a9d7a6b9",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SYMBOL  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "           Count   count\n",
      "0              -  204615\n",
      "1         RBFOX1     705\n",
      "2          Y_RNA     565\n",
      "3          CSMD1     410\n",
      "4           FHIT     314\n",
      "...          ...     ...\n",
      "25438    HPRT1P2       1\n",
      "25439     ERVE-1       1\n",
      "25440  LINC02120       1\n",
      "25441     NOS2P1       1\n",
      "25442   MTND1P18       1\n",
      "\n",
      "[25443 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "           Count   count\n",
      "0              -  216369\n",
      "1         RBFOX1     705\n",
      "2          CSMD1     506\n",
      "3          Y_RNA     491\n",
      "4        CNTNAP2     389\n",
      "...          ...     ...\n",
      "25515   MANEA-DT       1\n",
      "25516      FOXB2       1\n",
      "25517  RN7SL472P       1\n",
      "25518    ALG1L7P       1\n",
      "25519     RPL4P1       1\n",
      "\n",
      "[25520 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "         Count    count\n",
      "0            -  1254369\n",
      "1       RBFOX1     5789\n",
      "2        CSMD1     5507\n",
      "3        CDH13     2755\n",
      "4         FHIT     2394\n",
      "...        ...      ...\n",
      "28315   H2AC14        1\n",
      "28316  YBX1P10        1\n",
      "28317   GPR89P        1\n",
      "28318  IGKV4-1        1\n",
      "28319  VTI1BP1        1\n",
      "\n",
      "[28320 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "          Count    count\n",
      "0             -  1351685\n",
      "1         CSMD1     6368\n",
      "2        RBFOX1     6163\n",
      "3         PTPRD     3059\n",
      "4          FHIT     2785\n",
      "...         ...      ...\n",
      "28222  ACTG1P23        1\n",
      "28223    CERNA2        1\n",
      "28224     NCF1C        1\n",
      "28225   CYP2G2P        1\n",
      "28226     ACBD4        1\n",
      "\n",
      "[28227 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SYMBOL-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('SYMBOL', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4b40d2bb-42c6-478d-b0e0-ab063bf0da5f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SYMBOL_SOURCE  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "        Count   count\n",
      "0        HGNC  328897\n",
      "1           -  204615\n",
      "2  EntrezGene   11728\n",
      "3        RFAM     864\n",
      "\n",
      "indel63\n",
      "        Count   count\n",
      "0        HGNC  332881\n",
      "1           -  216369\n",
      "2  EntrezGene   12342\n",
      "3        RFAM     761\n",
      "\n",
      "snp12\n",
      "        Count    count\n",
      "0        HGNC  1627418\n",
      "1           -  1254369\n",
      "2  EntrezGene    64487\n",
      "3        RFAM     3823\n",
      "\n",
      "snp63\n",
      "        Count    count\n",
      "0        HGNC  1713401\n",
      "1           -  1351685\n",
      "2  EntrezGene    69764\n",
      "3        RFAM     3811\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SYMBOL_SOURCE-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n",
      "\n",
      "snp63\n",
      "        Count    count\n",
      "0        HGNC  1713401\n",
      "1           -  1351685\n",
      "2  EntrezGene    69764\n",
      "3        RFAM     3811\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SYMBOL_SOURCE-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('SYMBOL_SOURCE', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f53ab0fa-70ab-442d-9960-8725ef52dc85",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  BIOTYPE  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                                 Count   count\n",
      "0                       protein_coding  275598\n",
      "1                                    -  132307\n",
      "2                               lncRNA   90084\n",
      "3                             enhancer   13112\n",
      "4                 processed_pseudogene   10122\n",
      "5       protein_coding_CDS_not_defined    4484\n",
      "6               unprocessed_pseudogene    3534\n",
      "7                    CTCF_binding_site    2515\n",
      "8                open_chromatin_region    2066\n",
      "9               transcribed_pseudogene    2059\n",
      "10                            misc_RNA    1716\n",
      "11             nonsense_mediated_decay    1471\n",
      "12                               snRNA    1369\n",
      "13                                 TEC     741\n",
      "14  transcribed_unprocessed_pseudogene     703\n",
      "15                processed_transcript     599\n",
      "16      transcribed_unitary_pseudogene     506\n",
      "17                               miRNA     494\n",
      "18                     retained_intron     364\n",
      "19                              snoRNA     296\n",
      "20                     rRNA_pseudogene     290\n",
      "21    transcribed_processed_pseudogene     288\n",
      "22                     TF_binding_site     282\n",
      "23                  unitary_pseudogene     275\n",
      "24                  protein_coding_LoF     266\n",
      "25                           TR_V_gene     129\n",
      "26                           IG_V_gene     110\n",
      "27                                rRNA      80\n",
      "28                     IG_V_pseudogene      75\n",
      "29                            promoter      62\n",
      "30                              scaRNA      24\n",
      "31                       antisense_RNA      18\n",
      "32                     TR_V_pseudogene      15\n",
      "33                           TR_J_gene      13\n",
      "34                           IG_C_gene       7\n",
      "35                           IG_D_gene       7\n",
      "36                           TR_C_gene       6\n",
      "37                            ribozyme       5\n",
      "38                           IG_J_gene       4\n",
      "39                           TR_D_gene       3\n",
      "40                                sRNA       2\n",
      "41                           vault_RNA       1\n",
      "42                      non_stop_decay       1\n",
      "43     translated_processed_pseudogene       1\n",
      "\n",
      "indel63\n",
      "                                 Count   count\n",
      "0                       protein_coding  277205\n",
      "1                                    -  142163\n",
      "2                               lncRNA   95375\n",
      "3                             enhancer   12532\n",
      "4                 processed_pseudogene   10319\n",
      "5       protein_coding_CDS_not_defined    4571\n",
      "6               unprocessed_pseudogene    3450\n",
      "7                    CTCF_binding_site    2394\n",
      "8                open_chromatin_region    2073\n",
      "9               transcribed_pseudogene    2013\n",
      "10                            misc_RNA    1592\n",
      "11             nonsense_mediated_decay    1413\n",
      "12                               snRNA    1367\n",
      "13  transcribed_unprocessed_pseudogene     742\n",
      "14                                 TEC     729\n",
      "15                processed_transcript     640\n",
      "16      transcribed_unitary_pseudogene     502\n",
      "17                               miRNA     461\n",
      "18                     retained_intron     393\n",
      "19                              snoRNA     320\n",
      "20                     rRNA_pseudogene     312\n",
      "21                     TF_binding_site     311\n",
      "22    transcribed_processed_pseudogene     304\n",
      "23                  unitary_pseudogene     279\n",
      "24                  protein_coding_LoF     271\n",
      "25                           TR_V_gene     146\n",
      "26                           IG_V_gene     130\n",
      "27                                rRNA      78\n",
      "28                     IG_V_pseudogene      67\n",
      "29                            promoter      66\n",
      "30                       antisense_RNA      27\n",
      "31                     TR_V_pseudogene      22\n",
      "32                              scaRNA      20\n",
      "33                           TR_J_gene      18\n",
      "34                           TR_C_gene      12\n",
      "35                           IG_C_gene       6\n",
      "36                            ribozyme       6\n",
      "37                     IG_C_pseudogene       5\n",
      "38                                sRNA       4\n",
      "39                           TR_D_gene       3\n",
      "40     translated_processed_pseudogene       3\n",
      "41                      non_stop_decay       3\n",
      "42                           IG_D_gene       3\n",
      "43                          pseudogene       1\n",
      "44                           vault_RNA       1\n",
      "45                           IG_J_gene       1\n",
      "\n",
      "snp12\n",
      "                                 Count    count\n",
      "0                       protein_coding  1316340\n",
      "1                                    -   826409\n",
      "2                               lncRNA   534521\n",
      "3                             enhancer    78159\n",
      "4                 processed_pseudogene    53409\n",
      "5       protein_coding_CDS_not_defined    25139\n",
      "6               unprocessed_pseudogene    23732\n",
      "7                open_chromatin_region    14086\n",
      "8                    CTCF_binding_site    14039\n",
      "9               transcribed_pseudogene    10037\n",
      "10                            misc_RNA     7752\n",
      "11             nonsense_mediated_decay     7627\n",
      "12                               snRNA     6508\n",
      "13  transcribed_unprocessed_pseudogene     3725\n",
      "14                                 TEC     3674\n",
      "15                processed_transcript     3461\n",
      "16      transcribed_unitary_pseudogene     2760\n",
      "17                               miRNA     2524\n",
      "18                     TF_binding_site     2178\n",
      "19                     retained_intron     2058\n",
      "20                  protein_coding_LoF     1749\n",
      "21                     rRNA_pseudogene     1598\n",
      "22                  unitary_pseudogene     1595\n",
      "23                              snoRNA     1512\n",
      "24    transcribed_processed_pseudogene     1367\n",
      "25                           IG_V_gene     1044\n",
      "26                           TR_V_gene     1017\n",
      "27                                rRNA      691\n",
      "28                     IG_V_pseudogene      417\n",
      "29                            promoter      294\n",
      "30                     TR_V_pseudogene      118\n",
      "31                              scaRNA      108\n",
      "32                           TR_J_gene      102\n",
      "33                           IG_C_gene       62\n",
      "34                           TR_C_gene       61\n",
      "35                       antisense_RNA       56\n",
      "36                           vault_RNA       28\n",
      "37                            ribozyme       28\n",
      "38                      non_stop_decay       26\n",
      "39                           IG_D_gene       26\n",
      "40                           IG_J_gene       24\n",
      "41                           TR_D_gene       12\n",
      "42                                sRNA        9\n",
      "43     translated_processed_pseudogene        8\n",
      "44   translated_unprocessed_pseudogene        3\n",
      "45                       IG_pseudogene        2\n",
      "46                          pseudogene        1\n",
      "47                     IG_C_pseudogene        1\n",
      "\n",
      "snp63\n",
      "                                 Count    count\n",
      "0                       protein_coding  1384827\n",
      "1                                    -   897666\n",
      "2                               lncRNA   575790\n",
      "3                             enhancer    78253\n",
      "4                 processed_pseudogene    56453\n",
      "5       protein_coding_CDS_not_defined    26105\n",
      "6               unprocessed_pseudogene    23932\n",
      "7                open_chromatin_region    14345\n",
      "8                    CTCF_binding_site    14298\n",
      "9               transcribed_pseudogene    10301\n",
      "10             nonsense_mediated_decay     7745\n",
      "11                            misc_RNA     7517\n",
      "12                               snRNA     6721\n",
      "13  transcribed_unprocessed_pseudogene     3999\n",
      "14                                 TEC     3842\n",
      "15                processed_transcript     3679\n",
      "16      transcribed_unitary_pseudogene     2899\n",
      "17                               miRNA     2604\n",
      "18                     TF_binding_site     2157\n",
      "19                     retained_intron     2137\n",
      "20                  protein_coding_LoF     2004\n",
      "21                              snoRNA     1806\n",
      "22                     rRNA_pseudogene     1749\n",
      "23    transcribed_processed_pseudogene     1567\n",
      "24                  unitary_pseudogene     1537\n",
      "25                           IG_V_gene     1258\n",
      "26                           TR_V_gene     1162\n",
      "27                                rRNA      729\n",
      "28                     IG_V_pseudogene      443\n",
      "29                            promoter      332\n",
      "30                           TR_J_gene      135\n",
      "31                     TR_V_pseudogene      131\n",
      "32                           TR_C_gene      123\n",
      "33                       antisense_RNA       75\n",
      "34                            ribozyme       68\n",
      "35                              scaRNA       62\n",
      "36                           IG_C_gene       51\n",
      "37                      non_stop_decay       29\n",
      "38                           vault_RNA       24\n",
      "39                           IG_D_gene       22\n",
      "40                           IG_J_gene       20\n",
      "41                     IG_C_pseudogene       17\n",
      "42                           TR_D_gene       16\n",
      "43                                sRNA       15\n",
      "44     translated_processed_pseudogene        9\n",
      "45   translated_unprocessed_pseudogene        7\n",
      "\n",
      "snp12\n",
      "                                 Count    count\n",
      "0                       protein_coding  1316340\n",
      "1                                    -   826409\n",
      "2                               lncRNA   534521\n",
      "3                             enhancer    78159\n",
      "4                 processed_pseudogene    53409\n",
      "5       protein_coding_CDS_not_defined    25139\n",
      "6               unprocessed_pseudogene    23732\n",
      "7                open_chromatin_region    14086\n",
      "8                    CTCF_binding_site    14039\n",
      "9               transcribed_pseudogene    10037\n",
      "10                            misc_RNA     7752\n",
      "11             nonsense_mediated_decay     7627\n",
      "12                               snRNA     6508\n",
      "13  transcribed_unprocessed_pseudogene     3725\n",
      "14                                 TEC     3674\n",
      "15                processed_transcript     3461\n",
      "16      transcribed_unitary_pseudogene     2760\n",
      "17                               miRNA     2524\n",
      "18                     TF_binding_site     2178\n",
      "19                     retained_intron     2058\n",
      "20                  protein_coding_LoF     1749\n",
      "21                     rRNA_pseudogene     1598\n",
      "22                  unitary_pseudogene     1595\n",
      "23                              snoRNA     1512\n",
      "24    transcribed_processed_pseudogene     1367\n",
      "25                           IG_V_gene     1044\n",
      "26                           TR_V_gene     1017\n",
      "27                                rRNA      691\n",
      "28                     IG_V_pseudogene      417\n",
      "29                            promoter      294\n",
      "30                     TR_V_pseudogene      118\n",
      "31                              scaRNA      108\n",
      "32                           TR_J_gene      102\n",
      "33                           IG_C_gene       62\n",
      "34                           TR_C_gene       61\n",
      "35                       antisense_RNA       56\n",
      "36                           vault_RNA       28\n",
      "37                            ribozyme       28\n",
      "38                      non_stop_decay       26\n",
      "39                           IG_D_gene       26\n",
      "40                           IG_J_gene       24\n",
      "41                           TR_D_gene       12\n",
      "42                                sRNA        9\n",
      "43     translated_processed_pseudogene        8\n",
      "44   translated_unprocessed_pseudogene        3\n",
      "45                       IG_pseudogene        2\n",
      "46                          pseudogene        1\n",
      "47                     IG_C_pseudogene        1\n",
      "\n",
      "snp63\n",
      "                                 Count    count\n",
      "0                       protein_coding  1384827\n",
      "1                                    -   897666\n",
      "2                               lncRNA   575790\n",
      "3                             enhancer    78253\n",
      "4                 processed_pseudogene    56453\n",
      "5       protein_coding_CDS_not_defined    26105\n",
      "6               unprocessed_pseudogene    23932\n",
      "7                open_chromatin_region    14345\n",
      "8                    CTCF_binding_site    14298\n",
      "9               transcribed_pseudogene    10301\n",
      "10             nonsense_mediated_decay     7745\n",
      "11                            misc_RNA     7517\n",
      "12                               snRNA     6721\n",
      "13  transcribed_unprocessed_pseudogene     3999\n",
      "14                                 TEC     3842\n",
      "15                processed_transcript     3679\n",
      "16      transcribed_unitary_pseudogene     2899\n",
      "17                               miRNA     2604\n",
      "18                     TF_binding_site     2157\n",
      "19                     retained_intron     2137\n",
      "20                  protein_coding_LoF     2004\n",
      "21                              snoRNA     1806\n",
      "22                     rRNA_pseudogene     1749\n",
      "23    transcribed_processed_pseudogene     1567\n",
      "24                  unitary_pseudogene     1537\n",
      "25                           IG_V_gene     1258\n",
      "26                           TR_V_gene     1162\n",
      "27                                rRNA      729\n",
      "28                     IG_V_pseudogene      443\n",
      "29                            promoter      332\n",
      "30                           TR_J_gene      135\n",
      "31                     TR_V_pseudogene      131\n",
      "32                           TR_C_gene      123\n",
      "33                       antisense_RNA       75\n",
      "34                            ribozyme       68\n",
      "35                              scaRNA       62\n",
      "36                           IG_C_gene       51\n",
      "37                      non_stop_decay       29\n",
      "38                           vault_RNA       24\n",
      "39                           IG_D_gene       22\n",
      "40                           IG_J_gene       20\n",
      "41                     IG_C_pseudogene       17\n",
      "42                           TR_D_gene       16\n",
      "43                                sRNA       15\n",
      "44     translated_processed_pseudogene        9\n",
      "45   translated_unprocessed_pseudogene        7\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/BIOTYPE-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('BIOTYPE', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3a16e659-fb4b-4730-9fc8-97dec50942f8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  GIVEN_REF_MATCH  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "   Count   count\n",
      "0   True  546037\n",
      "1  False      67\n",
      "\n",
      "indel63\n",
      "   Count   count\n",
      "0   True  562270\n",
      "1  False      83\n",
      "\n",
      "snp12\n",
      "   Count    count\n",
      "0   True  2950082\n",
      "1  False       15\n",
      "\n",
      "snp63\n",
      "   Count    count\n",
      "0   True  3138642\n",
      "1  False       19\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/GIVEN_REF_MATCH-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('GIVEN_REF_MATCH', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "870a4a41-04a1-42dc-ae34-eb0dd5775691",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  CLIN_SIG  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                                          Count   count\n",
      "0                                             -  544545\n",
      "1                                        benign    1258\n",
      "2                                 likely_benign      76\n",
      "3                          benign,likely_benign      75\n",
      "4                        uncertain_significance      57\n",
      "5                 uncertain_significance,benign      50\n",
      "6   uncertain_significance,benign,likely_benign      17\n",
      "7          uncertain_significance,likely_benign      15\n",
      "8                                    pathogenic       2\n",
      "9                 benign,uncertain_significance       1\n",
      "10                  benign,benign/likely_benign       1\n",
      "11                 likely_pathogenic,pathogenic       1\n",
      "12                          not_provided,benign       1\n",
      "13                                drug_response       1\n",
      "14                                  risk_factor       1\n",
      "15                                   protective       1\n",
      "16                            benign,pathogenic       1\n",
      "17            not_provided,benign,likely_benign       1\n",
      "\n",
      "indel63\n",
      "                                          Count   count\n",
      "0                                             -  560690\n",
      "1                                        benign    1353\n",
      "2                                 likely_benign      76\n",
      "3                          benign,likely_benign      70\n",
      "4                 uncertain_significance,benign      63\n",
      "5                        uncertain_significance      43\n",
      "6   uncertain_significance,benign,likely_benign      21\n",
      "7          uncertain_significance,likely_benign      20\n",
      "8                           not_provided,benign       2\n",
      "9             not_provided,benign,likely_benign       2\n",
      "10                                drug_response       1\n",
      "11                                  association       1\n",
      "12              likely_benign,likely_pathogenic       1\n",
      "13                  benign,benign/likely_benign       1\n",
      "14            uncertain_significance,pathogenic       1\n",
      "15                                   pathogenic       1\n",
      "16                                  risk_factor       1\n",
      "17                           benign,risk_factor       1\n",
      "18                benign,uncertain_significance       1\n",
      "19                         likely_benign,benign       1\n",
      "20                                   protective       1\n",
      "21                            benign,pathogenic       1\n",
      "22                  benign/likely_benign,benign       1\n",
      "\n",
      "snp12\n",
      "                                                 Count    count\n",
      "0                                                    -  2919233\n",
      "1                                               benign    29181\n",
      "2                                 benign,likely_benign      422\n",
      "3                                        likely_benign      405\n",
      "4                               uncertain_significance       93\n",
      "..                                                 ...      ...\n",
      "113      benign,likely_pathogenic,benign/likely_benign        1\n",
      "114           uncertain_significance,likely_pathogenic        1\n",
      "115  conflicting_interpretations_of_pathogenicity,r...        1\n",
      "116             protective,benign/likely_benign,benign        1\n",
      "117                                  protective,benign        1\n",
      "\n",
      "[118 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "                                                 Count    count\n",
      "0                                                    -  3106563\n",
      "1                                               benign    30298\n",
      "2                                        likely_benign      434\n",
      "3                                 benign,likely_benign      399\n",
      "4                          benign,benign/likely_benign      108\n",
      "..                                                 ...      ...\n",
      "128                          drug_response,risk_factor        1\n",
      "129  not_provided,risk_factor,association,uncertain...        1\n",
      "130  risk_factor,not_provided,conflicting_interpret...        1\n",
      "131  benign,protective,conflicting_interpretations_...        1\n",
      "132  benign/likely_benign,benign,uncertain_signific...        1\n",
      "\n",
      "[133 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "                                                 Count    count\n",
      "0                                                    -  3106563\n",
      "1                                               benign    30298\n",
      "2                                        likely_benign      434\n",
      "3                                 benign,likely_benign      399\n",
      "4                          benign,benign/likely_benign      108\n",
      "..                                                 ...      ...\n",
      "128                          drug_response,risk_factor        1\n",
      "129  not_provided,risk_factor,association,uncertain...        1\n",
      "130  risk_factor,not_provided,conflicting_interpret...        1\n",
      "131  benign,protective,conflicting_interpretations_...        1\n",
      "132  benign/likely_benign,benign,uncertain_signific...        1\n",
      "\n",
      "[133 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/CLIN_SIG-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('CLIN_SIG', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "624b8d6b-5a6b-4824-9637-461cc8ae8b72",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  PHENO  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "            Count   count\n",
      "0               -  536495\n",
      "1               1    5073\n",
      "2             0,1    4104\n",
      "3             1,1     183\n",
      "4           0,1,1     179\n",
      "5         0,1,1,1      31\n",
      "6           1,1,1      19\n",
      "7       0,1,1,1,1       8\n",
      "8     0,1,1,1,1,1       6\n",
      "9   0,1,1,1,1,1,1       2\n",
      "10        1,1,1,1       2\n",
      "11      1,1,1,1,1       1\n",
      "12          0,0,1       1\n",
      "\n",
      "indel63\n",
      "                Count   count\n",
      "0                   -  552207\n",
      "1                   1    5463\n",
      "2                 0,1    4260\n",
      "3                 1,1     185\n",
      "4               0,1,1     173\n",
      "5             0,1,1,1      35\n",
      "6               1,1,1      16\n",
      "7           0,1,1,1,1       7\n",
      "8         0,1,1,1,1,1       4\n",
      "9           1,1,1,1,1       1\n",
      "10    0,1,1,1,1,1,1,1       1\n",
      "11  0,1,1,1,1,1,1,1,1       1\n",
      "\n",
      "snp12\n",
      "        Count    count\n",
      "0           -  2743335\n",
      "1         0,1    98449\n",
      "2           1    88554\n",
      "3         1,1    16839\n",
      "4       0,1,1     1643\n",
      "5       1,1,1     1169\n",
      "6     1,1,1,1       63\n",
      "7     0,1,1,1       37\n",
      "8   1,1,1,1,1        4\n",
      "9       0,0,1        3\n",
      "10    0,0,0,1        1\n",
      "\n",
      "snp63\n",
      "        Count    count\n",
      "0           -  2920895\n",
      "1         0,1   102627\n",
      "2           1    95271\n",
      "3         1,1    16968\n",
      "4       0,1,1     1607\n",
      "5       1,1,1     1185\n",
      "6     1,1,1,1       61\n",
      "7     0,1,1,1       38\n",
      "8       0,0,1        4\n",
      "9   1,1,1,1,1        3\n",
      "10    0,0,0,1        2\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/PHENO-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('PHENO', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "48e531f1-f54d-4ba0-8102-5487455a5484",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SOMATIC  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "            Count   count\n",
      "0               -  541559\n",
      "1             0,1    4278\n",
      "2           0,1,1     195\n",
      "3         0,1,1,1      33\n",
      "4               1      16\n",
      "5       0,1,1,1,1       9\n",
      "6     0,1,1,1,1,1       5\n",
      "7           1,1,1       2\n",
      "8   0,1,1,1,1,1,1       2\n",
      "9           0,0,1       2\n",
      "10            1,1       2\n",
      "11    0,0,1,1,1,1       1\n",
      "\n",
      "indel63\n",
      "                Count   count\n",
      "0                   -  557662\n",
      "1                 0,1    4427\n",
      "2               0,1,1     184\n",
      "3             0,1,1,1      35\n",
      "4                   1      22\n",
      "5           0,1,1,1,1       8\n",
      "6         0,1,1,1,1,1       4\n",
      "7                 1,1       4\n",
      "8               1,1,1       4\n",
      "9     0,1,1,1,1,1,1,1       1\n",
      "10              0,0,1       1\n",
      "11  0,1,1,1,1,1,1,1,1       1\n",
      "\n",
      "snp12\n",
      "        Count    count\n",
      "0           -  2832865\n",
      "1         0,1   114260\n",
      "2       0,1,1     1734\n",
      "3       0,0,1     1078\n",
      "4     0,0,1,1       58\n",
      "5           1       54\n",
      "6     0,1,1,1       29\n",
      "7     0,0,0,1       14\n",
      "8   0,0,1,1,1        3\n",
      "9         1,1        1\n",
      "10  0,0,0,0,1        1\n",
      "\n",
      "snp63\n",
      "        Count    count\n",
      "0           -  3017204\n",
      "1         0,1   118533\n",
      "2       0,1,1     1686\n",
      "3       0,0,1     1104\n",
      "4     0,0,1,1       59\n",
      "5           1       29\n",
      "6     0,1,1,1       28\n",
      "7     0,0,0,1       14\n",
      "8   0,0,1,1,1        2\n",
      "9   0,0,0,1,1        1\n",
      "10        1,1        1\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SOMATIC-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SOMATIC-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('SOMATIC', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e8ddcdb-cb4a-4526-8901-8d876ce506c7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SIFTprediction  ################\n",
      "\n",
      "\n",
      "snp12\n",
      "                        Count    count\n",
      "0                           -  2941355\n",
      "1                   tolerated     5434\n",
      "2    tolerated_low_confidence     1392\n",
      "3                 deleterious     1342\n",
      "4  deleterious_low_confidence      574\n",
      "\n",
      "snp63\n",
      "                        Count    count\n",
      "0                           -  3130032\n",
      "1                   tolerated     5412\n",
      "2                 deleterious     1344\n",
      "3    tolerated_low_confidence     1337\n",
      "4  deleterious_low_confidence      536\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SIFTprediction-checkValueCount.xlsx\n",
      "return list of dfs length:  2\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "#################### snps only ##########################################\n",
    "save_colValueCount('SIFTprediction', snps, snpnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d289bbc2-c2bc-4679-8a68-23bb3440d4cd",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  PolyPhenprediction  ################\n",
      "\n",
      "\n",
      "snp12\n",
      "               Count    count\n",
      "0                  -  2941653\n",
      "1             benign     6635\n",
      "2  possibly_damaging      711\n",
      "3  probably_damaging      582\n",
      "4            unknown      516\n",
      "\n",
      "snp63\n",
      "               Count    count\n",
      "0                  -  3130350\n",
      "1             benign     6546\n",
      "2  possibly_damaging      706\n",
      "3  probably_damaging      541\n",
      "4            unknown      518\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/PolyPhenprediction-checkValueCount.xlsx\n",
      "return list of dfs length:  2\n",
      "returning None..\n",
      "\n",
      "\n",
      "snp12\n",
      "               Count    count\n",
      "0                  -  2941653\n",
      "1             benign     6635\n",
      "2  possibly_damaging      711\n",
      "3  probably_damaging      582\n",
      "4            unknown      516\n",
      "\n",
      "snp63\n",
      "               Count    count\n",
      "0                  -  3130350\n",
      "1             benign     6546\n",
      "2  possibly_damaging      706\n",
      "3  probably_damaging      541\n",
      "4            unknown      518\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/PolyPhenprediction-checkValueCount.xlsx\n",
      "return list of dfs length:  2\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "#################### snps only ##########################################\n",
    "save_colValueCount('PolyPhenprediction', snps, snpnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7b87560-bf1d-4d3e-bdba-b4c656a84be7",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  chr  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "   Count  count\n",
      "0      1  45828\n",
      "1      2  41391\n",
      "2      3  33279\n",
      "3      7  31710\n",
      "4      4  30190\n",
      "5      5  29549\n",
      "6      6  29232\n",
      "7     10  27981\n",
      "8     12  27178\n",
      "9     11  25469\n",
      "10     8  24193\n",
      "11     9  23189\n",
      "12    17  20968\n",
      "13    19  19948\n",
      "14    16  18862\n",
      "15     X  18069\n",
      "16    14  16698\n",
      "17    13  16650\n",
      "18    15  16472\n",
      "19    20  15059\n",
      "20    18  13677\n",
      "21    21   9218\n",
      "22    22   7547\n",
      "23    22   3747\n",
      "\n",
      "indel63\n",
      "   Count  count\n",
      "0      1  46235\n",
      "1      2  44419\n",
      "2      3  35096\n",
      "3      4  34013\n",
      "4      7  32380\n",
      "5      5  31504\n",
      "6      6  30914\n",
      "7     10  28636\n",
      "8     12  27849\n",
      "9     11  26622\n",
      "10     8  24935\n",
      "11     9  23758\n",
      "12    17  19892\n",
      "13     X  18968\n",
      "14    13  17938\n",
      "15    16  17843\n",
      "16    19  17545\n",
      "17    14  17534\n",
      "18    15  16850\n",
      "19    18  14860\n",
      "20    20  14550\n",
      "21    21   9143\n",
      "22    22   8156\n",
      "23    22   2713\n",
      "\n",
      "snp12\n",
      "   Count   count\n",
      "0      2  234816\n",
      "1      1  228637\n",
      "2      3  192471\n",
      "3      4  189574\n",
      "4      5  170213\n",
      "5      6  165699\n",
      "6      7  165227\n",
      "7     10  154313\n",
      "8      8  149455\n",
      "9     11  145243\n",
      "10    12  138207\n",
      "11     9  130819\n",
      "12    13  100593\n",
      "13    16   97923\n",
      "14    14   88488\n",
      "15    17   87243\n",
      "16     X   87088\n",
      "17    15   85702\n",
      "18    20   82456\n",
      "19    18   79319\n",
      "20    19   76304\n",
      "21    21   49231\n",
      "22    22   47075\n",
      "23    22    4001\n",
      "\n",
      "snp63\n",
      "   Count   count\n",
      "0      2  256792\n",
      "1      1  241099\n",
      "2      4  215012\n",
      "3      3  208119\n",
      "4      5  185738\n",
      "5      6  178431\n",
      "6      7  175616\n",
      "7     10  160083\n",
      "8      8  156345\n",
      "9     11  156217\n",
      "10    12  146669\n",
      "11     9  137964\n",
      "12    13  109996\n",
      "13    16   98712\n",
      "14    14   96855\n",
      "15     X   94969\n",
      "16    15   89931\n",
      "17    18   88024\n",
      "18    17   87800\n",
      "19    20   83194\n",
      "20    19   68847\n",
      "21    21   50903\n",
      "22    22   46885\n",
      "23    22    4460\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/chr-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "#### new check values ##############################\n",
    "save_colValueCount('chr', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1eef004-a2a3-499b-a641-329c34c3669f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  Allele  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                        Count   count\n",
      "0                           -  280983\n",
      "1                           T   57413\n",
      "2                           A   57065\n",
      "3                           G   13664\n",
      "4                           C   13588\n",
      "...                       ...     ...\n",
      "20473                TGAGGAGC       1\n",
      "20474  ATGATAGTATGGAGTTTCAGTA       1\n",
      "20475               AATTCAACT       1\n",
      "20476  CCTATACATACTATGTTTTTTT       1\n",
      "20477      CTCTTCGGAGGTGGCTTC       1\n",
      "\n",
      "[20478 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "                      Count   count\n",
      "0                         -  291559\n",
      "1                         T   56741\n",
      "2                         A   56723\n",
      "3                         C   13891\n",
      "4                         G   13782\n",
      "...                     ...     ...\n",
      "22018              GCCTCTTG       1\n",
      "22019               AGCTGGG       1\n",
      "22020  ATCTCGCCCACCCTCTGCAC       1\n",
      "22021           GTGTGTGTTGT       1\n",
      "22022               TCACTAC       1\n",
      "\n",
      "[22023 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "  Count   count\n",
      "0     C  741398\n",
      "1     G  740525\n",
      "2     T  734671\n",
      "3     A  733503\n",
      "\n",
      "snp63\n",
      "  Count   count\n",
      "0     C  786099\n",
      "1     G  784661\n",
      "2     A  783951\n",
      "3     T  783950\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Allele-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('Allele', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "224849c8-c67b-47d8-a15b-34380e7a7a58",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  REF_ALLELE  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                                                   Count   count\n",
      "0                                                      -  252293\n",
      "1                                                      T   58401\n",
      "2                                                      A   57993\n",
      "3                                                      G   13755\n",
      "4                                                      C   13727\n",
      "...                                                  ...     ...\n",
      "24084                                           TCTGGCTG       1\n",
      "24085                                      CTTCCCCTTCTTT       1\n",
      "24086                            CTCCCTCTCCCTCTCCCCACGGT       1\n",
      "24087  AAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGAAAGGAAGGAA...       1\n",
      "24088                                         CCCCCGCACC       1\n",
      "\n",
      "[24089 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "                               Count   count\n",
      "0                                  -  259456\n",
      "1                                  T   58829\n",
      "2                                  A   58573\n",
      "3                                  C   14146\n",
      "4                                  G   14024\n",
      "...                              ...     ...\n",
      "25550                          GTGAC       1\n",
      "25551         TTTTTTTTGTTTTTGTTTTTTG       1\n",
      "25552  AGCAGACCCCCACTGGCACCTATCCCATA       1\n",
      "25553                  AACATTAATGGAG       1\n",
      "25554                     CCCCCGCACC       1\n",
      "\n",
      "[25555 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "  Count   count\n",
      "0     G  773395\n",
      "1     C  772893\n",
      "2     A  702154\n",
      "3     T  701655\n",
      "\n",
      "snp63\n",
      "  Count   count\n",
      "0     G  817820\n",
      "1     C  816307\n",
      "2     T  752390\n",
      "3     A  752144\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/REF_ALLELE-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('REF_ALLELE', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9cc2bef7-7ec3-40a5-b338-8150f05c0c04",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  Amino_acids  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "              Count   count\n",
      "0                 -  545607\n",
      "1               -/X      23\n",
      "2               E/-      10\n",
      "3               S/X       7\n",
      "4               R/X       6\n",
      "..              ...     ...\n",
      "352            LG/L       1\n",
      "353  GFSYRLQGRMTY/X       1\n",
      "354            C/CX       1\n",
      "355             H/-       1\n",
      "356         PTTGH/P       1\n",
      "\n",
      "[357 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "              Count   count\n",
      "0                 -  561884\n",
      "1               -/X      14\n",
      "2               E/-      11\n",
      "3               G/X       8\n",
      "4               -/Q       7\n",
      "..              ...     ...\n",
      "332  -/DR*ALPSSLHPX       1\n",
      "333            PA/P       1\n",
      "334            GL/G       1\n",
      "335       ATTRRTT/A       1\n",
      "336         PTTGH/P       1\n",
      "\n",
      "[337 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "    Count    count\n",
      "0       -  2930494\n",
      "1       A     1225\n",
      "2       P     1128\n",
      "3       L     1087\n",
      "4       S     1074\n",
      "..    ...      ...\n",
      "185   */G        1\n",
      "186   */S        1\n",
      "187   */K        1\n",
      "188   K/*        1\n",
      "189   */L        1\n",
      "\n",
      "[190 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "    Count    count\n",
      "0       -  3119552\n",
      "1       A     1159\n",
      "2       L     1144\n",
      "3       P     1092\n",
      "4       S      988\n",
      "..    ...      ...\n",
      "182   L/*        1\n",
      "183   */G        1\n",
      "184   */S        1\n",
      "185   */L        1\n",
      "186     X        1\n",
      "\n",
      "[187 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Amino_acids-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('Amino_acids', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6826a36-54d7-4d3d-8188-6890ea0d508f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  Gene  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                 Count   count\n",
      "0                    -  150344\n",
      "1      ENSG00000078328     705\n",
      "2      ENSG00000183117     410\n",
      "3      ENSG00000189283     314\n",
      "4      ENSG00000130226     308\n",
      "...                ...     ...\n",
      "34525  ENSG00000238808       1\n",
      "34526  ENSG00000253479       1\n",
      "34527  ENSG00000212473       1\n",
      "34528  ENSG00000212989       1\n",
      "34529  ENSG00000290825       1\n",
      "\n",
      "[34530 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "                 Count   count\n",
      "0                    -  159539\n",
      "1      ENSG00000078328     705\n",
      "2      ENSG00000183117     506\n",
      "3      ENSG00000174469     389\n",
      "4      ENSG00000153707     384\n",
      "...                ...     ...\n",
      "34530  ENSG00000270987       1\n",
      "34531  ENSG00000265697       1\n",
      "34532  ENSG00000214428       1\n",
      "34533        100133941       1\n",
      "34534  ENSG00000196422       1\n",
      "\n",
      "[34535 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "                 Count   count\n",
      "0                    -  935165\n",
      "1      ENSG00000078328    5789\n",
      "2      ENSG00000183117    5507\n",
      "3      ENSG00000140945    2755\n",
      "4      ENSG00000189283    2394\n",
      "...                ...     ...\n",
      "39597  ENSG00000033100       1\n",
      "39598  ENSG00000243836       1\n",
      "39599  ENSG00000239951       1\n",
      "39600  ENSG00000160284       1\n",
      "39601  ENSG00000290825       1\n",
      "\n",
      "[39602 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "                 Count    count\n",
      "0                    -  1007051\n",
      "1      ENSG00000183117     6368\n",
      "2      ENSG00000078328     6163\n",
      "3      ENSG00000153707     3059\n",
      "4      ENSG00000189283     2785\n",
      "...                ...      ...\n",
      "39455           220980        1\n",
      "39456  ENSG00000165507        1\n",
      "39457  ENSG00000181123        1\n",
      "39458  ENSG00000228462        1\n",
      "39459  ENSG00000279091        1\n",
      "\n",
      "[39460 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/Gene-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('Gene', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d9b19d7d-a660-41ee-82ff-515b55ad4b12",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  STRAND  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "  Count   count\n",
      "0     1  203788\n",
      "1    -1  193095\n",
      "2     -  149221\n",
      "\n",
      "indel63\n",
      "  Count   count\n",
      "0     1  207307\n",
      "1    -1  196565\n",
      "2     -  158481\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0     1  1042335\n",
      "1    -1   981164\n",
      "2     -   926598\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0     1  1099330\n",
      "1    -1  1040785\n",
      "2     -   998546\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0     1  1042335\n",
      "1    -1   981164\n",
      "2     -   926598\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0     1  1099330\n",
      "1    -1  1040785\n",
      "2     -   998546\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/STRAND-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('STRAND', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8ef17424-fce0-4bd0-b547-f732d284c25c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  FLAGS  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                     Count   count\n",
      "0                        -  541278\n",
      "1               cds_end_NF    3603\n",
      "2             cds_start_NF    1131\n",
      "3  cds_start_NF,cds_end_NF      92\n",
      "\n",
      "indel63\n",
      "                     Count   count\n",
      "0                        -  557492\n",
      "1               cds_end_NF    3661\n",
      "2             cds_start_NF    1130\n",
      "3  cds_start_NF,cds_end_NF      70\n",
      "\n",
      "snp12\n",
      "                     Count    count\n",
      "0                        -  2924027\n",
      "1               cds_end_NF    20039\n",
      "2             cds_start_NF     5559\n",
      "3  cds_start_NF,cds_end_NF      472\n",
      "\n",
      "snp63\n",
      "                     Count    count\n",
      "0                        -  3110973\n",
      "1               cds_end_NF    21082\n",
      "2             cds_start_NF     6166\n",
      "3  cds_start_NF,cds_end_NF      440\n",
      "\n",
      "snp63\n",
      "                     Count    count\n",
      "0                        -  3110973\n",
      "1               cds_end_NF    21082\n",
      "2             cds_start_NF     6166\n",
      "3  cds_start_NF,cds_end_NF      440\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/FLAGS-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('FLAGS', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e38cae99-187e-438d-9e48-9464eb5c802d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  SWISSPROT  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "            Count   count\n",
      "0               -  296501\n",
      "1      Q9NWB1.180     536\n",
      "2      Q96PZ7.164     410\n",
      "3      P49789.193     314\n",
      "4      Q9NZC7.193     298\n",
      "...           ...     ...\n",
      "16252  Q8N4W6.136       1\n",
      "16253   B5MD39.81       1\n",
      "16254  Q9BZL1.150       1\n",
      "16255  P02461.233       1\n",
      "16256  P51668.207       1\n",
      "\n",
      "[16257 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "            Count   count\n",
      "0               -  311578\n",
      "1      Q9NWB1.180     509\n",
      "2      Q96PZ7.164     506\n",
      "3      Q9UHC6.206     389\n",
      "4      P23468.222     384\n",
      "...           ...     ...\n",
      "16278  Q96DR4.142       1\n",
      "16279  Q96DB2.177       1\n",
      "16280  P57059.202       1\n",
      "16281  Q01081.235       1\n",
      "16282  Q9Y3D3.185       1\n",
      "\n",
      "[16283 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "            Count    count\n",
      "0               -  1768363\n",
      "1      Q96PZ7.164     5507\n",
      "2      Q9NWB1.180     4280\n",
      "3      P55290.200     2755\n",
      "4      P49789.193     2394\n",
      "...           ...      ...\n",
      "17086  Q9HD47.138        1\n",
      "17087   A6NIN4.66        1\n",
      "17088   Q30KR1.87        1\n",
      "17089  P14672.233        1\n",
      "17090  A6NNM3.107        1\n",
      "\n",
      "[17091 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "               Count    count\n",
      "0                  -  1894116\n",
      "1         Q96PZ7.164     6368\n",
      "2         Q9NWB1.180     4313\n",
      "3         P23468.222     3059\n",
      "4         P49789.193     2785\n",
      "...              ...      ...\n",
      "17013     P62841.179        1\n",
      "17014  A0A494C191.15        1\n",
      "17015     P05546.222        1\n",
      "17016     Q8TCP9.132        1\n",
      "17017     P21917.202        1\n",
      "\n",
      "[17018 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/SWISSPROT-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('SWISSPROT', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e134163-fb6b-4492-b819-cf913312dc40",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  GENE_PHENO  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "  Count   count\n",
      "0     -  443495\n",
      "1     1  102609\n",
      "\n",
      "indel63\n",
      "  Count   count\n",
      "0     -  458149\n",
      "1     1  104204\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0     -  2451455\n",
      "1     1   498642\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0     -  2611595\n",
      "1     1   527066\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/GENE_PHENO-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('GENE_PHENO', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "06ab2b3b-3346-49fe-9673-d51ff3ea6f9b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  MAX_AF_POPS  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "                                                  Count  count\n",
      "0                                           gnomADg_EAS  89924\n",
      "1                                           gnomADg_AMI  74362\n",
      "2                                                   AFR  53960\n",
      "3                                           gnomADg_AFR  51047\n",
      "4                                           gnomADg_FIN  43296\n",
      "...                                                 ...    ...\n",
      "2256  EAS,SAS,gnomADg_AFR,gnomADg_AMI,gnomADg_ASJ,gn...      1\n",
      "2257  AFR,AMR,EAS,EUR,SAS,gnomADe_AMR,gnomADe_ASJ,gn...      1\n",
      "2258  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...      1\n",
      "2259  AFR,AMR,EAS,EUR,SAS,gnomADe_ASJ,gnomADe_FIN,gn...      1\n",
      "2260  gnomADe_NFE,gnomADg_AFR,gnomADg_AMI,gnomADg_AM...      1\n",
      "\n",
      "[2261 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "                                                  Count  count\n",
      "0                                           gnomADg_AMI  88058\n",
      "1                                           gnomADg_EAS  81031\n",
      "2                                                   AFR  56114\n",
      "3                                           gnomADg_FIN  54074\n",
      "4                                           gnomADg_AFR  48199\n",
      "...                                                 ...    ...\n",
      "2323  gnomADe_AFR,gnomADe_AMR,gnomADe_ASJ,gnomADe_EA...      1\n",
      "2324  AFR,AMR,EAS,EUR,SAS,gnomADe_ASJ,gnomADe_FIN,gn...      1\n",
      "2325  AFR,AMR,EAS,EUR,SAS,gnomADg_AMR,gnomADg_ASJ,gn...      1\n",
      "2326  AMR,gnomADg_AMI,gnomADg_ASJ,gnomADg_EAS,gnomAD...      1\n",
      "2327  gnomADe_NFE,gnomADg_AFR,gnomADg_AMI,gnomADg_AM...      1\n",
      "\n",
      "[2328 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "                                                  Count   count\n",
      "0                                                   AFR  541582\n",
      "1                                           gnomADg_AMI  396889\n",
      "2                                                   EAS  354999\n",
      "3                                           gnomADg_EAS  352016\n",
      "4                                                   SAS  232238\n",
      "...                                                 ...     ...\n",
      "3978  AFR,AMR,EAS,EUR,SAS,gnomADg_AFR,gnomADg_AMI,gn...       1\n",
      "3979  EAS,gnomADe_EAS,gnomADe_FIN,gnomADg_ASJ,gnomAD...       1\n",
      "3980  gnomADe_AMR,gnomADe_ASJ,gnomADe_EAS,gnomADe_FI...       1\n",
      "3981  AFR,EAS,gnomADg_AMI,gnomADg_ASJ,gnomADg_EAS,gn...       1\n",
      "3982  AFR,AMR,SAS,gnomADe_AMR,gnomADe_ASJ,gnomADe_FI...       1\n",
      "\n",
      "[3983 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "                                                  Count   count\n",
      "0                                                   AFR  553259\n",
      "1                                           gnomADg_AMI  493338\n",
      "2                                                   EAS  344025\n",
      "3                                           gnomADg_EAS  337481\n",
      "4                                           gnomADg_FIN  249303\n",
      "...                                                 ...     ...\n",
      "4008  AFR,AMR,EUR,SAS,gnomADg_AFR,gnomADg_AMI,gnomAD...       1\n",
      "4009  AFR,EAS,EUR,SAS,gnomADg_AFR,gnomADg_AMI,gnomAD...       1\n",
      "4010  AFR,AMR,EAS,SAS,gnomADg_AFR,gnomADg_AMI,gnomAD...       1\n",
      "4011  AFR,AMR,EAS,EUR,SAS,gnomADg_AMI,gnomADg_EAS,gn...       1\n",
      "4012  AFR,AMR,SAS,gnomADe_AMR,gnomADe_ASJ,gnomADe_FI...       1\n",
      "\n",
      "[4013 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/MAX_AF_POPS-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('MAX_AF_POPS', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "04b294a8-6491-4555-8a05-6eec5ed84a72",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  HIGH_INF_POS  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "  Count   count\n",
      "0     -  544981\n",
      "1     N    1121\n",
      "2     Y       2\n",
      "\n",
      "indel63\n",
      "  Count   count\n",
      "0     -  561295\n",
      "1     N    1058\n",
      "\n",
      "snp12\n",
      "  Count    count\n",
      "0     -  2941530\n",
      "1     N     5923\n",
      "2     Y     2644\n",
      "\n",
      "snp63\n",
      "  Count    count\n",
      "0     -  3130156\n",
      "1     N     5905\n",
      "2     Y     2600\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/HIGH_INF_POS-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('HIGH_INF_POS', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ee2e7708-882b-499c-91db-d3e1dfe7d405",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "################  MOTIF_NAME  ################\n",
      "\n",
      "\n",
      "indel12\n",
      "          Count   count\n",
      "0             -  544981\n",
      "1    ENSPFM0577      28\n",
      "2    ENSPFM0597      26\n",
      "3    ENSPFM0042      26\n",
      "4    ENSPFM0403      24\n",
      "..          ...     ...\n",
      "224  ENSPFM0124       1\n",
      "225  ENSPFM0304       1\n",
      "226  ENSPFM0527       1\n",
      "227  ENSPFM0113       1\n",
      "228  ENSPFM0027       1\n",
      "\n",
      "[229 rows x 2 columns]\n",
      "\n",
      "indel63\n",
      "          Count   count\n",
      "0             -  561295\n",
      "1    ENSPFM0042      29\n",
      "2    ENSPFM0573      24\n",
      "3    ENSPFM0577      23\n",
      "4    ENSPFM0597      23\n",
      "..          ...     ...\n",
      "222  ENSPFM0099       1\n",
      "223  ENSPFM0074       1\n",
      "224  ENSPFM0096       1\n",
      "225  ENSPFM0177       1\n",
      "226  ENSPFM0027       1\n",
      "\n",
      "[227 rows x 2 columns]\n",
      "\n",
      "snp12\n",
      "          Count    count\n",
      "0             -  2941530\n",
      "1    ENSPFM0042      284\n",
      "2    ENSPFM0598      252\n",
      "3    ENSPFM0597      232\n",
      "4    ENSPFM0521      217\n",
      "..          ...      ...\n",
      "273  ENSPFM0167        1\n",
      "274  ENSPFM0177        1\n",
      "275  ENSPFM0093        1\n",
      "276  ENSPFM0099        1\n",
      "277  ENSPFM0040        1\n",
      "\n",
      "[278 rows x 2 columns]\n",
      "\n",
      "snp63\n",
      "          Count    count\n",
      "0             -  3130156\n",
      "1    ENSPFM0042      314\n",
      "2    ENSPFM0598      235\n",
      "3    ENSPFM0597      231\n",
      "4    ENSPFM0521      220\n",
      "..          ...      ...\n",
      "277  ENSPFM0167        1\n",
      "278  ENSPFM0445        1\n",
      "279  ENSPFM0093        1\n",
      "280  ENSPFM0350        1\n",
      "281  ENSPFM0176        1\n",
      "\n",
      "[282 rows x 2 columns]\n",
      "\n",
      "saved multi-sheet excel as  /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/MOTIF_NAME-checkValueCount.xlsx\n",
      "return list of dfs length:  4\n",
      "returning None..\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_colValueCount('MOTIF_NAME', dfs, dfnames, path, False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1ee9265a-ce17-4797-800a-9c8bfca741f5",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "def save_crossTabIt_snp(i, dfname, PATH):\n",
    "    # snp func include SIFTprediction and PolyPhenprediction\n",
    "    # funx to save crosstab of columns in a sheet of an excel\n",
    "    # i: dataframe\n",
    "    # dfname: name of dataframe\n",
    "    # col types\n",
    "    typedf = i.dtypes\n",
    "    desdf = i.describe(include='number')\n",
    "    describee = pd.DataFrame(desdf).T\n",
    "    # columns with np.nan values from  script 1\n",
    "    nan_column_from_script1 =['cDNA_position', 'CDS_position', 'Protein_position', 'DISTANCE', 'EXONnumber', 'INTRONnumber', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF','MAX_AF', 'CLIN_SIG'] \n",
    "    df_with_nan = i[nan_column_from_script1].copy()\n",
    "    sumdf = df_with_nan.isna().sum()\n",
    "    # write objects to separate sheets of an excel \n",
    "    output =  PATH + dfname + '-crossTabIt' + '.xlsx'\n",
    "    unique_values = ['keyID38','Gene','SYMBOL', 'CCDS', 'Feature', 'ENSP','SWISSPROT']\n",
    "    # crosstab of columns\n",
    "    x1,s1 = save_crosstabit_sheet(i,\"CLIN_SIG\", \"ZYG\", True)\n",
    "    x2,s2 = save_crosstabit_sheet(i,\"IMPACT\", \"ZYG\", True)\n",
    "    x3,s3 = save_crosstabit_sheet(i,\"SIFTprediction\", \"ZYG\", True)\n",
    "    x4,s4 = save_crosstabit_sheet(i,\"PolyPhenprediction\", \"ZYG\", True)\n",
    "    x7,s7 = save_crosstabit_sheet(i,\"Consequence\", \"ZYG\", True)\n",
    "    x8,s8 = save_crosstabit_sheet(i,\"Consequence\", \"IMPACT\", True)\n",
    "    x9,s9 = save_crosstabit_sheet(i,\"Consequence\", \"SIFTprediction\", True)\n",
    "    x10,s10 = save_crosstabit_sheet(i,\"Consequence\", \"PolyPhenprediction\", True)\n",
    "    x6,s6 = save_crosstabit_sheet(i, \"SYMBOL\", \"IMPACT\", True)\n",
    "    x11,s11 = save_crosstabit_sheet(i,\"SYMBOL\", \"Consequence\", True)\n",
    "    x12,s12 = save_crosstabit_sheet(i,\"SYMBOL\", \"SIFTprediction\", True)\n",
    "    x13,s13 = save_crosstabit_sheet(i,\"SYMBOL\", \"PolyPhenprediction\", True)\n",
    "    with pd.ExcelWriter(output) as writer:\n",
    "        typedf.to_excel(writer, sheet_name='datatypes')\n",
    "        describee.to_excel(writer, sheet_name='number_stats')\n",
    "        sumdf.to_excel(writer, sheet_name='nan_value_counts')\n",
    "        # get unique values for list of df columns\n",
    "        for col in unique_values:\n",
    "            allCol =  []\n",
    "            uniqCol = []\n",
    "            # allCol.append(col)\n",
    "            # uniqCol.append(col)\n",
    "            lengthAll = len(i[col])\n",
    "            lengthUniq = len(i[col].unique())\n",
    "            sheetname = col + '-unique'\n",
    "            colname1 = 'total-' + col\n",
    "            colname2 = 'unique-' + col\n",
    "            allCol.append(colname1)\n",
    "            uniqCol.append(colname2)\n",
    "            allCol.append(lengthAll)\n",
    "            uniqCol.append(lengthUniq)\n",
    "            uniqueDF = pd.DataFrame([allCol, uniqCol])\n",
    "            # columns=[colname1, colname2\n",
    "            print(uniqueDF)\n",
    "            uniqueDF.to_excel(writer, sheet_name=sheetname)\n",
    "        x1.to_excel(writer, sheet_name=s1)\n",
    "        x2.to_excel(writer, sheet_name=s2)\n",
    "        x3.to_excel(writer, sheet_name=s3)\n",
    "        x4.to_excel(writer, sheet_name=s4)\n",
    "        x7.to_excel(writer, sheet_name=s7)\n",
    "        x8.to_excel(writer, sheet_name=s8)\n",
    "        x9.to_excel(writer, sheet_name=s9)\n",
    "        x10.to_excel(writer, sheet_name=s10)\n",
    "        x6.to_excel(writer, sheet_name=s6)\n",
    "        x11.to_excel(writer, sheet_name=s11)\n",
    "        x12.to_excel(writer, sheet_name=s12)\n",
    "        x13.to_excel(writer, sheet_name=s13)\n",
    "    print('saved as :', output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7d67bd0c-2a05-4c3a-b5e9-1c21a7fcae35",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                0        1\n",
      "0   total-keyID38  2950097\n",
      "1  unique-keyID38  2950097\n",
      "             0        1\n",
      "0   total-Gene  2950097\n",
      "1  unique-Gene    39602\n",
      "               0        1\n",
      "0   total-SYMBOL  2950097\n",
      "1  unique-SYMBOL    28320\n",
      "             0        1\n",
      "0   total-CCDS  2950097\n",
      "1  unique-CCDS    17285\n",
      "                0        1\n",
      "0   total-Feature  2950097\n",
      "1  unique-Feature    98895\n",
      "             0        1\n",
      "0   total-ENSP  2950097\n",
      "1  unique-ENSP    20685\n",
      "                  0        1\n",
      "0   total-SWISSPROT  2950097\n",
      "1  unique-SWISSPROT    17091\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mariapalafox/anaconda3/envs/py39/lib/python3.9/site-packages/openpyxl/workbook/child.py:99: UserWarning: Title is more than 31 characters. Some applications may not be able to read the file\n",
      "  warnings.warn(\"Title is more than 31 characters. Some applications may not be able to read the file\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "saved as : /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/snp12-crossTabIt.xlsx\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_crossTabIt_snp(snp12, 'snp12', path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ed2389f4-bfaf-4867-bec9-4cd0e3d896bb",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                0        1\n",
      "0   total-keyID38  3138661\n",
      "1  unique-keyID38  3138661\n",
      "             0        1\n",
      "0   total-Gene  3138661\n",
      "1  unique-Gene    39460\n",
      "               0        1\n",
      "0   total-SYMBOL  3138661\n",
      "1  unique-SYMBOL    28227\n",
      "             0        1\n",
      "0   total-CCDS  3138661\n",
      "1  unique-CCDS    17200\n",
      "                0        1\n",
      "0   total-Feature  3138661\n",
      "1  unique-Feature    98869\n",
      "             0        1\n",
      "0   total-ENSP  3138661\n",
      "1  unique-ENSP    20596\n",
      "                  0        1\n",
      "0   total-SWISSPROT  3138661\n",
      "1  unique-SWISSPROT    17018\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/mariapalafox/anaconda3/envs/py39/lib/python3.9/site-packages/openpyxl/workbook/child.py:99: UserWarning: Title is more than 31 characters. Some applications may not be able to read the file\n",
      "  warnings.warn(\"Title is more than 31 characters. Some applications may not be able to read the file\")\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "saved as : /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/snp63-crossTabIt.xlsx\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_crossTabIt_snp(snp63, 'snp63', path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71c7c971-6fd0-406a-b67d-b1533b08b713",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "def save_crossTabIt_indel(i, dfname, PATH):\n",
    "    # indel func does not include SIFTprediction and PolyPhenprediction\n",
    "    # funx to save crosstab of columns in a sheet of an excel\n",
    "    # i: dataframe\n",
    "    # dfname: name of dataframe\n",
    "    # col types\n",
    "    typedf = i.dtypes\n",
    "    desdf = i.describe(include='number')\n",
    "    describee = pd.DataFrame(desdf).T\n",
    "    # columns with np.nan values from  script 1\n",
    "    nan_column_from_script1 =['cDNA_position', 'CDS_position', 'Protein_position', 'DISTANCE', 'EXONnumber', 'INTRONnumber', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF','MAX_AF', 'CLIN_SIG'] \n",
    "    df_with_nan = i[nan_column_from_script1].copy()\n",
    "    sumdf = df_with_nan.isna().sum()\n",
    "    # write objects to separate sheets of an excel \n",
    "    output =  PATH + dfname + '-crossTabIt' + '.xlsx'\n",
    "    unique_values = ['Gene','SYMBOL', 'CCDS', 'Feature', 'ENSP','SWISSPROT']\n",
    "    # crosstab of columns\n",
    "    x1,s1 = save_crosstabit_sheet(i,\"CLIN_SIG\", \"ZYG\", True)\n",
    "    x2,s2 = save_crosstabit_sheet(i,\"IMPACT\", \"ZYG\", True)\n",
    "    x7,s7 = save_crosstabit_sheet(i,\"Consequence\", \"ZYG\", True)\n",
    "    x8,s8 = save_crosstabit_sheet(i,\"Consequence\", \"IMPACT\", True)\n",
    "    x6,s6 = save_crosstabit_sheet(i, \"SYMBOL\", \"IMPACT\", True)\n",
    "    x11,s11 = save_crosstabit_sheet(i,\"SYMBOL\", \"Consequence\", True)\n",
    "    with pd.ExcelWriter(output) as writer:\n",
    "        typedf.to_excel(writer, sheet_name='datatypes')\n",
    "        describee.to_excel(writer, sheet_name='number_stats')\n",
    "        sumdf.to_excel(writer, sheet_name='nan_value_counts')\n",
    "        # get unique values for list of df columns\n",
    "        for col in unique_values:\n",
    "            allCol =  []\n",
    "            uniqCol = []\n",
    "            # allCol.append(col)\n",
    "            # uniqCol.append(col)\n",
    "            lengthAll = len(i[col])\n",
    "            lengthUniq = len(i[col].unique())\n",
    "            sheetname = col + '-unique'\n",
    "            colname1 = 'total-' + col\n",
    "            colname2 = 'unique-' + col\n",
    "            allCol.append(colname1)\n",
    "            uniqCol.append(colname2)\n",
    "            allCol.append(lengthAll)\n",
    "            uniqCol.append(lengthUniq)\n",
    "            uniqueDF = pd.DataFrame([allCol, uniqCol])\n",
    "            # columns=[colname1, colname2\n",
    "            print(uniqueDF)\n",
    "            uniqueDF.to_excel(writer, sheet_name=sheetname)\n",
    "        x1.to_excel(writer, sheet_name=s1)\n",
    "        x2.to_excel(writer, sheet_name=s2)\n",
    "        x7.to_excel(writer, sheet_name=s7)\n",
    "        x8.to_excel(writer, sheet_name=s8)\n",
    "        x6.to_excel(writer, sheet_name=s6)\n",
    "        x11.to_excel(writer, sheet_name=s11)\n",
    "    print('saved as :', output)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "234ee478-20ea-4ceb-ada8-2ddbb5f80f8b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             0       1\n",
      "0   total-Gene  546104\n",
      "1  unique-Gene   34530\n",
      "               0       1\n",
      "0   total-SYMBOL  546104\n",
      "1  unique-SYMBOL   25443\n",
      "             0       1\n",
      "0   total-CCDS  546104\n",
      "1  unique-CCDS   16404\n",
      "                0       1\n",
      "0   total-Feature  546104\n",
      "1  unique-Feature   51924\n",
      "             0       1\n",
      "0   total-ENSP  546104\n",
      "1  unique-ENSP   19159\n",
      "                  0       1\n",
      "0   total-SWISSPROT  546104\n",
      "1  unique-SWISSPROT   16257\n",
      "saved as : /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/indel12-crossTabIt.xlsx\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_crossTabIt_indel(indel12, 'indel12', path)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c98a9954-73ad-4643-9cd2-6dfb786cfa99",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "             0       1\n",
      "0   total-Gene  562353\n",
      "1  unique-Gene   34535\n",
      "               0       1\n",
      "0   total-SYMBOL  562353\n",
      "1  unique-SYMBOL   25520\n",
      "             0       1\n",
      "0   total-CCDS  562353\n",
      "1  unique-CCDS   16395\n",
      "                0       1\n",
      "0   total-Feature  562353\n",
      "1  unique-Feature   51385\n",
      "             0       1\n",
      "0   total-ENSP  562353\n",
      "1  unique-ENSP   19150\n",
      "                  0       1\n",
      "0   total-SWISSPROT  562353\n",
      "1  unique-SWISSPROT   16283\n",
      "saved as : /Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/2-summarize-v3/indel63-crossTabIt.xlsx\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "save_crossTabIt_indel(indel63, 'indel63', path)"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": 3
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
