{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>3253032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>152860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>54530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Count    count\n",
       "0      Common (>=10%)  3253032\n",
       "1  Less common (<10%)   152860\n",
       "2          Rare (<1%)    54530\n",
       "3                        35779"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'MAF')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sum(cad12.MAF.isna())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature.type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA.position</th>\n",
       "      <th>CDS.position</th>\n",
       "      <th>Protein.position</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing.variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>REF.ALLELE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL.SOURCE</th>\n",
       "      <th>HGNC.ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE.SELECT</th>\n",
       "      <th>MANE.PLUS.CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT.ISOFORM</th>\n",
       "      <th>REFSEQ.MATCH</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>REFSEQ.OFFSET</th>\n",
       "      <th>GIVEN.REF</th>\n",
       "      <th>USED.REF</th>\n",
       "      <th>BAM.EDIT</th>\n",
       "      <th>GENE.PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVS.OFFSET</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>chr</th>\n",
       "      <th>GIVEN.REF.MATCH</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "      <th>keyID38</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>var.exclusive.btw.lines</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>gene.exclusive.btw.lines</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>Database.py</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr1_181584_GG/-</td>\n",
       "      <td>chr1:181584-181585</td>\n",
       "      <td>-</td>\n",
       "      <td>102725121</td>\n",
       "      <td>NR_148357.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>GG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>803.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>DDX11L17</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:55080</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>-</td>\n",
       "      <td>GG</td>\n",
       "      <td>GG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>181584</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>chr1_267778_-/AAAAAAAA</td>\n",
       "      <td>chr1:267777-267778</td>\n",
       "      <td>AAAAAAAA</td>\n",
       "      <td>ENSG00000286448</td>\n",
       "      <td>ENST00000669836</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000669836.1:n.203-336_203-329dup</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>267777</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>chr1_909143_-/GGTCGGGGTCAGGCCCCCGGGCGCACCTTTGC...</td>\n",
       "      <td>chr1:909142-909143</td>\n",
       "      <td>GGTCGGGGTCAGGCCCCCGGGCGCACCTTTGCTGGTATATGCGGG</td>\n",
       "      <td>ENSG00000230699</td>\n",
       "      <td>ENST00000448179</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HOM</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>2292.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>909142</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>chr1_1245213_CACTCTTCTCCCCTCCTCCCCCACAC/-</td>\n",
       "      <td>chr1:1245213-1245238</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000184163</td>\n",
       "      <td>ENST00000330388</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>CACTCTTCTCCCCTCCTCCCCCACAC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:32308</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001014980.3</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS30554.1</td>\n",
       "      <td>ENSP00000329137</td>\n",
       "      <td>Q5T7M4.111</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI00001D7E7A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>CACTCTTCTCCCCTCCTCCCCCACAC</td>\n",
       "      <td>CACTCTTCTCCCCTCCTCCCCCACAC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000330388.2:c.178-741_178-716del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1245213</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>chr1_1245259_CCTACCCTCTCCTCCCCCCAT/-</td>\n",
       "      <td>chr1:1245259-1245279</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000184163</td>\n",
       "      <td>ENST00000330388</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>CCTACCCTCTCCTCCCCCCAT</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>C1QTNF12</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:32308</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001014980.3</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS30554.1</td>\n",
       "      <td>ENSP00000329137</td>\n",
       "      <td>Q5T7M4.111</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI00001D7E7A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>CCTACCCTCTCCTCCCCCCAT</td>\n",
       "      <td>CCTACCCTCTCCTCCCCCCAT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000330388.2:c.178-782_178-762del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>1245259</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949580</th>\n",
       "      <td>chrX_155533381_A/G</td>\n",
       "      <td>chrX:155533381</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000185973</td>\n",
       "      <td>ENST00000334398</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>TMLHE</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:18308</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_018196.4</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS14768.1</td>\n",
       "      <td>ENSP00000335261</td>\n",
       "      <td>Q9NVH6.173</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000013707B</td>\n",
       "      <td>Q9NVH6-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2/7</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000334398.8:c.182-8749T&gt;C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155533381</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_155533381_A_G</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949667</th>\n",
       "      <td>chrX_155638619_C/T</td>\n",
       "      <td>chrX:155638619</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000168939</td>\n",
       "      <td>ENST00000695325</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11271</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001304990.2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS14769.4</td>\n",
       "      <td>ENSP00000511806</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI0000135E98</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000695325.1:c.-282+25972C&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155638619</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_155638619_C_T</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949688</th>\n",
       "      <td>chrX_155666592_C/T</td>\n",
       "      <td>chrX:155666592</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000168939</td>\n",
       "      <td>ENST00000695325</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>SPRY3</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11271</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001304990.2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS14769.4</td>\n",
       "      <td>ENSP00000511806</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI0000135E98</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000695325.1:c.-282+53945C&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155666592</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_155666592_C_T</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949871</th>\n",
       "      <td>chrX_155870253_T/A</td>\n",
       "      <td>chrX:155870253</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>T</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155870253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_155870253_T_A</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2949876</th>\n",
       "      <td>chrX_155873456_C/T</td>\n",
       "      <td>chrX:155873456</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000237040</td>\n",
       "      <td>ENST00000422618</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>2180.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>DPH3P2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:38719</td>\n",
       "      <td>processed_pseudogene</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>155873456</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_155873456_C_T</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td></td>\n",
       "      <td>ClinVar</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>35779 rows Ã— 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Uploaded.variation              Location                                         Allele             Gene          Feature Feature.type                                   Consequence cDNA.position CDS.position Protein.position Amino.acids Codons Existing.variation      IND  ZYG                  REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS VARIANT.CLASS    SYMBOL SYMBOL.SOURCE     HGNC.ID                 BIOTYPE CANONICAL     MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS         CCDS             ENSP   SWISSPROT TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET                   GIVEN.REF                    USED.REF BAM.EDIT GENE.PHENO SIFT PolyPhen EXON INTRON DOMAINS miRNA                                   HGVSc HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF  MAX.AF MAX.AF.POPS CLIN.SIG SOMATIC PHENO PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines MAF MAF2  PATHO    VUS  BENIGN ClinVar Database.py\n",
       "6                                         chr1_181584_GG/-    chr1:181584-181585                                              -        102725121      NR_148357.1   Transcript                         upstream_gene_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                          GG  MODIFIER     803.0      1     -      deletion  DDX11L17    EntrezGene  HGNC:55080  transcribed_pseudogene       YES               -                  -   -      -            -                -           -      -              -               -            -   RefSeq             -                          GG                          GG        -          -    -        -    -      -       -     -                                       -     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chr1         181584         NaN           NaN   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False           False  False   False             ClinVar\n",
       "9                                   chr1_267778_-/AAAAAAAA    chr1:267777-267778                                       AAAAAAAA  ENSG00000286448  ENST00000669836   Transcript  intron_variant,non_coding_transcript_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           -  MODIFIER       NaN      1     -     insertion         -             -           -                  lncRNA       YES               -                  -   -      -            -                -           -      -              -               -            -  Ensembl             -                           -                           -        -          -    -        -    -    1/1       -     -  ENST00000669836.1:n.203-336_203-329dup     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chr1         267777         NaN           1.0   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False           False  False   False             ClinVar\n",
       "59       chr1_909143_-/GGTCGGGGTCAGGCCCCCGGGCGCACCTTTGC...    chr1:909142-909143  GGTCGGGGTCAGGCCCCCGGGCGCACCTTTGCTGGTATATGCGGG  ENSG00000230699  ENST00000448179   Transcript                         upstream_gene_variant           NaN          NaN              NaN           -      -                  -  P732912  HOM                           -  MODIFIER    2292.0      1     -     insertion         -             -           -                  lncRNA       YES               -                  -   2      -            -                -           -      -              -               -            -  Ensembl             -                           -                           -        -          -    -        -    -      -       -     -                                       -     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chr1         909142         NaN           NaN   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False           False  False   False             ClinVar\n",
       "205              chr1_1245213_CACTCTTCTCCCCTCCTCCCCCACAC/-  chr1:1245213-1245238                                              -  ENSG00000184163  ENST00000330388   Transcript                                intron_variant           NaN          NaN              NaN           -      -                  -  P732912  HET  CACTCTTCTCCCCTCCTCCCCCACAC  MODIFIER       NaN     -1     -      deletion  C1QTNF12          HGNC  HGNC:32308          protein_coding       YES  NM_001014980.3                  -   1     P1  CCDS30554.1  ENSP00000329137  Q5T7M4.111      -  UPI00001D7E7A               -            -  Ensembl             -  CACTCTTCTCCCCTCCTCCCCCACAC  CACTCTTCTCCCCTCCTCCCCCACAC        -          -    -        -    -    1/7       -     -  ENST00000330388.2:c.178-741_178-716del     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chr1        1245213         NaN           1.0   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False           False  False   False             ClinVar\n",
       "206                   chr1_1245259_CCTACCCTCTCCTCCCCCCAT/-  chr1:1245259-1245279                                              -  ENSG00000184163  ENST00000330388   Transcript                                intron_variant           NaN          NaN              NaN           -      -                  -  P732912  HET       CCTACCCTCTCCTCCCCCCAT  MODIFIER       NaN     -1     -      deletion  C1QTNF12          HGNC  HGNC:32308          protein_coding       YES  NM_001014980.3                  -   1     P1  CCDS30554.1  ENSP00000329137  Q5T7M4.111      -  UPI00001D7E7A               -            -  Ensembl             -       CCTACCCTCTCCTCCCCCCAT       CCTACCCTCTCCTCCCCCCAT        -          -    -        -    -    1/7       -     -  ENST00000330388.2:c.178-782_178-762del     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chr1        1245259         NaN           1.0   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False           False  False   False             ClinVar\n",
       "...                                                    ...                   ...                                            ...              ...              ...          ...                                           ...           ...          ...              ...         ...    ...                ...      ...  ...                         ...       ...       ...    ...   ...           ...       ...           ...         ...                     ...       ...             ...                ...  ..    ...          ...              ...         ...    ...            ...             ...          ...      ...           ...                         ...                         ...      ...        ...  ...      ...  ...    ...     ...   ...                                     ...   ...         ...         ...             ...             ...             ...             ...             ...             ...             ...             ...             ...             ...     ...         ...      ...     ...   ...    ...        ...       ...          ...                ...                   ...        ...            ...         ...           ...  ..              ...            ...        ...                ...            ...              ...                ...                      ...                 ...                       ...  ..  ...    ...    ...     ...     ...         ...\n",
       "2949580                                 chrX_155533381_A/G        chrX:155533381                                              G  ENSG00000185973  ENST00000334398   Transcript                                intron_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           A  MODIFIER       NaN     -1     -           SNV     TMLHE          HGNC  HGNC:18308          protein_coding       YES     NM_018196.4                  -   1     P1  CCDS14768.1  ENSP00000335261  Q9NVH6.173      -  UPI000013707B        Q9NVH6-1            -  Ensembl             -                           A                           A        -          1    -        -    -    2/7       -     -         ENST00000334398.8:c.182-8749T>C     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chrX      155533381         NaN           2.0   X             True              -        NaN                  -            NaN  X_155533381_A_G              False                     True                True                     False           False  False   False             ClinVar\n",
       "2949667                                 chrX_155638619_C/T        chrX:155638619                                              T  ENSG00000168939  ENST00000695325   Transcript                                intron_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           C  MODIFIER       NaN      1     -           SNV     SPRY3          HGNC  HGNC:11271          protein_coding       YES  NM_001304990.2                  -   -     P1  CCDS14769.4  ENSP00000511806           -      -  UPI0000135E98               -            -  Ensembl             -                           C                           C        -          -    -        -    -    1/2       -     -       ENST00000695325.1:c.-282+25972C>T     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chrX      155638619         NaN           1.0   X             True              -        NaN                  -            NaN  X_155638619_C_T              False                     True                True                     False           False  False   False             ClinVar\n",
       "2949688                                 chrX_155666592_C/T        chrX:155666592                                              T  ENSG00000168939  ENST00000695325   Transcript                                intron_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           C  MODIFIER       NaN      1     -           SNV     SPRY3          HGNC  HGNC:11271          protein_coding       YES  NM_001304990.2                  -   -     P1  CCDS14769.4  ENSP00000511806           -      -  UPI0000135E98               -            -  Ensembl             -                           C                           C        -          -    -        -    -    1/2       -     -       ENST00000695325.1:c.-282+53945C>T     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chrX      155666592         NaN           1.0   X             True              -        NaN                  -            NaN  X_155666592_C_T              False                     True                True                     False           False  False   False             ClinVar\n",
       "2949871                                 chrX_155870253_T/A        chrX:155870253                                              A                -                -            -                            intergenic_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           T  MODIFIER       NaN      -     -           SNV         -             -           -                       -         -               -                  -   -      -            -                -           -      -              -               -            -        -             -                           -                           -        -          -    -        -    -      -       -     -                                       -     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chrX      155870253         NaN           NaN   X             True              -        NaN                  -            NaN  X_155870253_T_A              False                     True                True                     False           False  False   False             ClinVar\n",
       "2949876                                 chrX_155873456_C/T        chrX:155873456                                              T  ENSG00000237040  ENST00000422618   Transcript                         upstream_gene_variant           NaN          NaN              NaN           -      -                  -  P732912  HET                           C  MODIFIER    2180.0      1     -           SNV    DPH3P2          HGNC  HGNC:38719    processed_pseudogene       YES               -                  -   -      -            -                -           -      -              -               -            -  Ensembl             -                           C                           C        -          -    -        -    -      -       -     -                                       -     -           -         NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN             NaN     NaN           -        -       -     -      -          -         -            -                  -                     -       chrX      155873456         NaN           NaN   X             True              -        NaN                  -            NaN  X_155873456_C_T              False                     True                True                     False           False  False   False             ClinVar\n",
       "\n",
       "[35779 rows x 96 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12[cad12['MAF'] == '']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "35633283-b212-4a93-a098-b42ab69d2e22",
   "metadata": {},
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "module 'numpy' has no attribute 'na'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[43], line 3\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[39m# %%\u001b[39;00m\n\u001b[1;32m      2\u001b[0m \u001b[39m# add MAF column to  dataframes, if MAX.AF is < 0.01 label Rare, if its less than 0.1 label less common, if its greater than or equal  to 0.1 label common, else label empty string\u001b[39;00m\n\u001b[0;32m----> 3\u001b[0m cad12[\u001b[39m'\u001b[39m\u001b[39mMAF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m np\u001b[39m.\u001b[39mselect([cad12[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m<\u001b[39m \u001b[39m0.01\u001b[39m, (cad12[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m>\u001b[39m\u001b[39m=\u001b[39m \u001b[39m0.01\u001b[39m) \u001b[39m&\u001b[39m (cad12[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m<\u001b[39m \u001b[39m0.1\u001b[39m), cad12[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m>\u001b[39m\u001b[39m=\u001b[39m \u001b[39m0.1\u001b[39m], [\u001b[39m'\u001b[39m\u001b[39mRare (<1\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mLess common (<10\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mCommon (>=10\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m], default \u001b[39m=\u001b[39m np\u001b[39m.\u001b[39;49mna)\n\u001b[1;32m      5\u001b[0m cad63[\u001b[39m'\u001b[39m\u001b[39mMAF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m=\u001b[39m np\u001b[39m.\u001b[39mselect([cad63[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m<\u001b[39m \u001b[39m0.01\u001b[39m, (cad63[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m>\u001b[39m\u001b[39m=\u001b[39m \u001b[39m0.01\u001b[39m) \u001b[39m&\u001b[39m (cad63[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m<\u001b[39m \u001b[39m0.1\u001b[39m), cad63[\u001b[39m'\u001b[39m\u001b[39mMAX.AF\u001b[39m\u001b[39m'\u001b[39m] \u001b[39m>\u001b[39m\u001b[39m=\u001b[39m \u001b[39m0.1\u001b[39m], [\u001b[39m'\u001b[39m\u001b[39mRare (<1\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mLess common (<10\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m, \u001b[39m'\u001b[39m\u001b[39mCommon (>=10\u001b[39m\u001b[39m%\u001b[39m\u001b[39m)\u001b[39m\u001b[39m'\u001b[39m], default \u001b[39m=\u001b[39m np\u001b[39m.\u001b[39mna)\n",
      "File \u001b[0;32m~/anaconda3/envs/py39/lib/python3.9/site-packages/numpy/__init__.py:333\u001b[0m, in \u001b[0;36m__getattr__\u001b[0;34m(attr)\u001b[0m\n\u001b[1;32m    330\u001b[0m     \u001b[39m\"\u001b[39m\u001b[39mRemoved in NumPy 1.25.0\u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    331\u001b[0m     \u001b[39mraise\u001b[39;00m \u001b[39mRuntimeError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mTester was removed in NumPy 1.25.\u001b[39m\u001b[39m\"\u001b[39m)\n\u001b[0;32m--> 333\u001b[0m \u001b[39mraise\u001b[39;00m \u001b[39mAttributeError\u001b[39;00m(\u001b[39m\"\u001b[39m\u001b[39mmodule \u001b[39m\u001b[39m{!r}\u001b[39;00m\u001b[39m has no attribute \u001b[39m\u001b[39m\"\u001b[39m\n\u001b[1;32m    334\u001b[0m                      \u001b[39m\"\u001b[39m\u001b[39m{!r}\u001b[39;00m\u001b[39m\"\u001b[39m\u001b[39m.\u001b[39mformat(\u001b[39m__name__\u001b[39m, attr))\n",
      "\u001b[0;31mAttributeError\u001b[0m: module 'numpy' has no attribute 'na'"
     ]
    }
   ],
   "source": [
    "# add MAF column to  dataframes, if MAX.AF is < 0.01 label Rare, if its less than 0.1 label less common, if its greater than or equal  to 0.1 label common, else label empty string\n",
    "cad12['MAF'] = np.select([cad12['MAX.AF'] < 0.01, (cad12['MAX.AF'] >= 0.01) & (cad12['MAX.AF'] < 0.1), cad12['MAX.AF'] >= 0.1], ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], default = np.na)\n",
    "\n",
    "cad63['MAF'] = np.select([cad63['MAX.AF'] < 0.01, (cad63['MAX.AF'] >= 0.01) & (cad63['MAX.AF'] < 0.1), cad63['MAX.AF'] >= 0.1], ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], default = np.na)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "73575a8d-8bfc-487e-831b-0f4f4eb7f98c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add MAF column to  dataframes, if MAX.AF is < 0.01 label Rare, if its less than 0.1 label less common, if its greater than or equal  to 0.1 label common, else label empty string\n",
    "cad12['MAF'] = np.select([cad12['MAX.AF'] < 0.01, (cad12['MAX.AF'] >= 0.01) & (cad12['MAX.AF'] < 0.1), cad12['MAX.AF'] >= 0.1], ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], default = np.nan)\n",
    "\n",
    "cad63['MAF'] = np.select([cad63['MAX.AF'] < 0.01, (cad63['MAX.AF'] >= 0.01) & (cad63['MAX.AF'] < 0.1), cad63['MAX.AF'] >= 0.1], ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>3253032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>152860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>54530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nan</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Count    count\n",
       "0      Common (>=10%)  3253032\n",
       "1  Less common (<10%)   152860\n",
       "2          Rare (<1%)    54530\n",
       "3                 nan    35779"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'MAF')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>3462469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>157624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>54790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nan</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Count    count\n",
       "0      Common (>=10%)  3462469\n",
       "1  Less common (<10%)   157624\n",
       "2          Rare (<1%)    54790\n",
       "3                 nan    26131"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'MAF')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "caf6668d-9dfe-4786-badd-285c47b60ae7",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the R code into python\n",
    "cad12['MAF2'] = np.select([cad12['MAX.AF'] < 0.05, \n",
    "                           cad12['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'],\n",
    "                            default=np.nan)\n",
    "\n",
    "cad63['MAF2'] = np.select([cad63['MAX.AF'] < 0.05, \n",
    "                           cad63['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'], \n",
    "                           default=np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>3542705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>132178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nan</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Count    count\n",
       "0  Common (>=5%)  3542705\n",
       "1     Rare (<5%)   132178\n",
       "2            nan    26131"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'MAF2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>3326323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>134099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nan</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Count    count\n",
       "0  Common (>=5%)  3326323\n",
       "1     Rare (<5%)   134099\n",
       "2            nan    35779"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'MAF2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0          3464042\n",
       "1  BENIGN    31769\n",
       "2     VUS      302\n",
       "3   PATHO       88"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "id": "bba3896c-bba8-4786-be11-72d599f621e6",
   "metadata": {},
   "outputs": [],
   "source": [
    "# functionn that checks CLIN.SIG column for rows containing 'pathogenic' or 'likely pathogenic' and creates new column with 'PATHO' label. the new column should also  contain 'VUS' and 'BENIGN' labels if the row contain 'benign' for 'BENIGN' and 'uncertain significance' for 'VUS'.\n",
    "\n",
    "def label_clin_sig(df):\n",
    "    df['PATHO'] = df['CLIN.SIG'].str.contains('pathogenic|likely pathogenic', case=False)\n",
    "    df['VUS'] = df['CLIN.SIG'].str.contains('uncertain_significance', case=False)\n",
    "    df['BENIGN'] = df['CLIN.SIG'].str.contains('benign|likely_benign', case=False)\n",
    "    return df\n",
    "\n",
    "\n",
    "# new function creates one column with new labels based on rule heirarchy; if pathogenic or likely pathogenic then labeled PATHO else if uncertain_significance then VUS else if benign or likely benign then BENIGN\n",
    "def label_clin_sig2(df):\n",
    "    newcol = []\n",
    "    for row in df.itertuples():\n",
    "        if row.PATHO:\n",
    "            newcol.append('PATHO')\n",
    "        elif row.VUS:\n",
    "            newcol.append('VUS')\n",
    "        elif row.BENIGN:\n",
    "            newcol.append('BENIGN')\n",
    "        else:\n",
    "            newcol.append(np.nan) # append nan value instead of '' emptry string\n",
    "    df['ClinVar'] = newcol\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "fb2ba1dc-7e56-48fe-8f03-ce93bb669d06",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         Uploaded.variation        Location Allele             Gene          Feature Feature.type              Consequence cDNA.position CDS.position Protein.position Amino.acids Codons Existing.variation      IND  ZYG REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS VARIANT.CLASS  SYMBOL SYMBOL.SOURCE     HGNC.ID         BIOTYPE CANONICAL MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS CCDS             ENSP SWISSPROT        TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET GIVEN.REF USED.REF BAM.EDIT GENE.PHENO SIFT PolyPhen EXON INTRON DOMAINS miRNA                            HGVSc HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF   MAX.AF                                        MAX.AF.POPS CLIN.SIG SOMATIC PHENO PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines                 MAF           MAF2  PATHO    VUS  BENIGN ClinVar       Database.py\n",
      "2950092  chrX_156024517_C/T  chrX:156024517      T  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -      -         rs55796665  P732912  HOM          C  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         C        C        -          -    -        -    -    8/9       -     -  ENST00000359512.8:c.1234+245C>T     -           -     1.00000        0.999900         1.00000        1.000000        1.000000        1.000000         1.00000         1.00000         1.00000         1.00000         1.00000  1.00000  gnomADg_AMI,gnomADg_AMR,gnomADg_ASJ,gnomADg_EA...        -       -     -      -          -         -            -                  -                     -       chrX      156024517         NaN           8.0   X             True              -        NaN                  -            NaN  X_156024517_C_T              False                     True                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "2950093  chrX_156025116_G/A  chrX:156025116      A  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -      -          rs3953915  P732912  HET          G  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         G        G        -          -    -        -    -    9/9       -     -   ENST00000359512.8:c.1303+16G>A     -           -     0.31330        0.188500         0.35090        0.302100        0.318000        0.037370         0.39250         0.28900         0.39760         0.31000         0.28810  0.40540                                        gnomADe_NFE        -       -     -      -          -         -            -                  -                     -       chrX      156025116         NaN           9.0   X             True              -        NaN                  -            NaN  X_156025116_G_A               True                    False                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "2950094  chrX_156025186_G/C  chrX:156025186      C  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -      -        rs144106392  P732912  HET          G  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         G        G        -          -    -        -    -    9/9       -     -   ENST00000359512.8:c.1304-55G>C     -           -     0.01565        0.002506         0.01429        0.004662        0.007501        0.000192         0.06131         0.01274         0.01901         0.01204         0.04165  0.06131                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chrX      156025186         NaN           9.0   X             True              -        NaN                  -            NaN  X_156025186_G_C              False                     True                True                     False  Less common (<10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "2950095  chrX_156025429_T/C  chrX:156025429      C  ENSG00000182484  ENST00000359512   Transcript  downstream_gene_variant           NaN          NaN              NaN           -      -        rs189478274  P732912  HET          T  MODIFIER      55.0      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         T        T        -          -    -        -    -      -       -     -                                -     -           -     0.11990        0.030570         0.21480        0.066790        0.177400        0.032690         0.32920         0.17320         0.14960         0.11890         0.23620  0.32920                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chrX      156025429         NaN           NaN   X             True              -        NaN                  -            NaN  X_156025429_T_C               True                    False                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "2950096  chrX_156029856_C/G  chrX:156029856      G  ENSG00000182484  ENST00000359512   Transcript  downstream_gene_variant           NaN          NaN              NaN           -      -         rs12353930  P732912  HET          C  MODIFIER    4482.0      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         C        C        -          -    -        -    -      -       -     -                                -     -           -     0.17900        0.081900         0.22300        0.204200        0.166700        0.368700         0.26540         0.18330         0.19210         0.16030         0.25960  0.36870                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -       chrX      156029856         NaN           NaN   X             True              -        NaN                  -            NaN  X_156029856_C_G              False                     True                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "         Uploaded.variation        Location Allele             Gene          Feature Feature.type              Consequence cDNA.position CDS.position Protein.position Amino.acids   Codons Existing.variation      IND  ZYG REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS VARIANT.CLASS  SYMBOL SYMBOL.SOURCE     HGNC.ID         BIOTYPE CANONICAL MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS CCDS             ENSP SWISSPROT        TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET GIVEN.REF USED.REF BAM.EDIT GENE.PHENO             SIFT                  PolyPhen  EXON INTRON                                            DOMAINS miRNA                           HGVSc                          HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF  MAX.AF  MAX.AF.POPS CLIN.SIG SOMATIC PHENO PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines             MAF           MAF2  PATHO    VUS  BENIGN ClinVar       Database.py\n",
      "3138656  chrX_156021847_A/G  chrX:156021847      G  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -        -         rs62608353  P931263  HET          A  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         A        A        -          -                -                         -     -    2/9                                                  -     -   ENST00000359512.8:c.300+55A>G                              -           -     0.68030         0.59450          0.7048         0.71410          0.6540         0.57790         0.71200          0.6835          0.7335          0.6839         0.61720  0.7335  gnomADg_NFE        -       -     -      -          -         -            -                  -                     -       chrX      156021847         NaN           2.0   X             True              -        NaN                  -            NaN  X_156021847_A_G               True                    False                True                     False  Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "3138657  chrX_156021966_A/G  chrX:156021966      G  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -        -        rs199556443  P931263  HET          A  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         A        A        -          -                -                         -     -    2/9                                                  -     -   ENST00000359512.8:c.301-33A>G                              -           -     0.48200         0.47300          0.4903         0.49350          0.4836         0.50320         0.47680          0.4811          0.4821          0.4796         0.49310  0.5032  gnomADg_EAS        -       -     -      -          -         -            -                  -                     -       chrX      156021966         NaN           2.0   X             True              -        NaN                  -            NaN  X_156021966_A_G              False                     True                True                     False  Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "3138658  chrX_156025041_C/T  chrX:156025041      T  ENSG00000182484  ENST00000359512   Transcript         missense_variant        1244.0       1244.0            415.0         T/M  aCg/aTg          rs4612551  P931263  HET          C  MODERATE       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         C        C        -          -  tolerated(0.06)  probably_damaging(0.991)  9/10      -  PANTHER:PTHR23331:SF2,PANTHER:PTHR23331,MobiDB...     -     ENST00000359512.8:c.1244C>T  ENSP00000504557.1:p.Thr415Met           -     0.09199         0.02293          0.1469         0.17370          0.1225         0.07087         0.07164          0.1216          0.1229          0.1306         0.02694  0.2394  gnomADe_AMR        -       -     -      -          -         -            -                  -                     -       chrX      156025041         9.0           NaN   X             True      tolerated       0.06  probably_damaging          0.991  X_156025041_C_T              False                     True                True                     False  Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "3138659  chrX_156025116_G/A  chrX:156025116      A  ENSG00000182484  ENST00000359512   Transcript           intron_variant           NaN          NaN              NaN           -        -          rs3953915  P931263  HET          G  MODIFIER       NaN      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         G        G        -          -                -                         -     -    9/9                                                  -     -  ENST00000359512.8:c.1303+16G>A                              -           -     0.31330         0.18850          0.3509         0.30210          0.3180         0.03737         0.39250          0.2890          0.3976          0.3100         0.28810  0.4054  gnomADe_NFE        -       -     -      -          -         -            -                  -                     -       chrX      156025116         NaN           9.0   X             True              -        NaN                  -            NaN  X_156025116_G_A               True                    False                True                     False  Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n",
      "3138660  chrX_156025429_T/C  chrX:156025429      C  ENSG00000182484  ENST00000359512   Transcript  downstream_gene_variant           NaN          NaN              NaN           -        -        rs189478274  P931263  HET          T  MODIFIER      55.0      1     -           SNV  WASH6P          HGNC  HGNC:31685  protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         T        T        -          -                -                         -     -      -                                                  -     -                               -                              -           -     0.11990         0.03057          0.2148         0.06679          0.1774         0.03269         0.32920          0.1732          0.1496          0.1189         0.23620  0.3292  gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chrX      156025429         NaN           NaN   X             True              -        NaN                  -            NaN  X_156025429_T_C               True                    False                True                     False  Common (>=10%)  Common (>=5%)  False  False   False     NaN  gnomAD & ClinVar\n"
     ]
    }
   ],
   "source": [
    "cad12 = label_clin_sig(cad12)\n",
    "cad12 = label_clin_sig2(cad12)\n",
    "print(cad12.tail())\n",
    "\n",
    "cad63 = label_clin_sig(cad63)\n",
    "cad63 = label_clin_sig2(cad63)\n",
    "print(cad63.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "id": "e7d847fc-4948-48b8-a51b-978a894f7a1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>31769</td>\n",
       "      <td>3464042</td>\n",
       "      <td>88</td>\n",
       "      <td>302</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>31769</td>\n",
       "      <td>3464042</td>\n",
       "      <td>88</td>\n",
       "      <td>302</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P732912   31769  3464042     88  302  3496201\n",
       "Total     31769  3464042     88  302  3496201"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho12 = cad12[cad12['ClinVar'] == 'PATHO']\n",
    "crosstabit(cad12, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "7c0a0c7c-87cc-4f81-bbd3-c22b6a9e6c2e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>33038</td>\n",
       "      <td>3667535</td>\n",
       "      <td>111</td>\n",
       "      <td>330</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>33038</td>\n",
       "      <td>3667535</td>\n",
       "      <td>111</td>\n",
       "      <td>330</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P931263   33038  3667535    111  330  3701014\n",
       "Total     33038  3667535    111  330  3701014"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho63 = cad63[cad63['ClinVar'] == 'PATHO']\n",
    "crosstabit(cad63, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31769\n",
       "2     VUS      302\n",
       "3   PATHO       88"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33038\n",
       "2     VUS      330\n",
       "3   PATHO      111"
      ]
     },
     "execution_count": 56,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          True\n",
       "1          True\n",
       "2          True\n",
       "3          True\n",
       "4          True\n",
       "           ... \n",
       "2950092    True\n",
       "2950093    True\n",
       "2950094    True\n",
       "2950095    True\n",
       "2950096    True\n",
       "Name: ClinVar, Length: 3496201, dtype: bool"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12.ClinVar.isna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0          False\n",
       "1          False\n",
       "2          False\n",
       "3          False\n",
       "4          False\n",
       "           ...  \n",
       "2950092    False\n",
       "2950093    False\n",
       "2950094    False\n",
       "2950095    False\n",
       "2950096    False\n",
       "Name: MAF, Length: 3496201, dtype: bool"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12.MAF.isna()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "c86e6dbb-b020-4473-80c3-c309f7c03567",
   "metadata": {},
   "outputs": [],
   "source": [
    "# works with default label functions set to np.nan instead of empty string\n",
    "qc_clinvar = ['PATHO', 'VUS', 'BENIGN']\n",
    "qc_gnomad = ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)']\n",
    "\n",
    "# create a new column 'Database' in both dataframes, if MAX.AF is not NA and ClinVar is NA label 'gnomAD', if ClinVar is not NA and MAX.AF is NA label 'ClinVar', if both are not NA label 'gnomAD & ClinVar', else label 'No DB source'\n",
    "\n",
    "cad12['Database.py'] = np.select([cad12['ClinVar'].isin(qc_clinvar) & cad12['MAX.AF'].isna(), \n",
    "                                  cad12['MAX.AF'].isin(qc_gnomad) & cad12['ClinVar'].isna(), \n",
    "                                  cad12['MAX.AF'].isin(qc_gnomad) & cad12['ClinVar'].isin(qc_clinvar), \n",
    "                                  cad12['MAX.AF'].isna() & cad12['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "\n",
    "cad63['Database.py'] = np.select([cad63['ClinVar'].isin(qc_clinvar) & cad63['MAX.AF'].isna(), \n",
    "                                  cad63['MAX.AF'].isin(qc_gnomad) & cad63['ClinVar'].isna(), \n",
    "                                  cad63['MAX.AF'].isin(qc_gnomad) & cad63['ClinVar'].isin(qc_clinvar), \n",
    "                                  cad63['MAX.AF'].isna() & cad63['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "id": "38f59883-3db2-45c8-bc0d-2c609b01c4e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>nan</td>\n",
       "      <td>3460422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>35778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ClinVar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Count    count\n",
       "0           nan  3460422\n",
       "1  No DB source    35778\n",
       "2       ClinVar        1"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'Database.py')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "id": "d745e33a-af7d-4a08-a30e-312b46d2ac4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>nan</td>\n",
       "      <td>3674883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "          Count    count\n",
       "0           nan  3674883\n",
       "1  No DB source    26131"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'Database.py')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "id": "f215506e-8159-49da-bcf2-f935db244706",
   "metadata": {},
   "outputs": [],
   "source": [
    "# works with default label functions set to np.nan instead of empty string\n",
    "qc_clinvar = ['PATHO', 'VUS', 'BENIGN']\n",
    "qc_gnomad = ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)']\n",
    "\n",
    "# create a new column 'Database' in both dataframes, if MAX.AF is not NA and ClinVar is NA label 'gnomAD', if ClinVar is not NA and MAX.AF is NA label 'ClinVar', if both are not NA label 'gnomAD & ClinVar', else label 'No DB source'\n",
    "\n",
    "\n",
    "cad12['Database.py'] = np.select([cad12['ClinVar'].notna() & cad12['MAX.AF'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].notna(), \n",
    "                                  cad12['MAX.AF'].isna() & cad12['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "\n",
    "cad63['Database.py'] = np.select([cad63['ClinVar'].notna() & cad63['MAX.AF'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].notna(), \n",
    "                                  cad63['MAX.AF'].isna() & cad63['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "id": "113135c0-31db-44e0-9593-1535d71c4869",
   "metadata": {},
   "outputs": [],
   "source": [
    "# works with default label functions set to np.nan instead of empty string\n",
    "qc_clinvar = ['PATHO', 'VUS', 'BENIGN']\n",
    "qc_gnomad = ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)']\n",
    "\n",
    "# create a new column 'Database' in both dataframes, if MAX.AF is not NA and ClinVar is NA label 'gnomAD', if ClinVar is not NA and MAX.AF is NA label 'ClinVar', if both are not NA label 'gnomAD & ClinVar', else label 'No DB source'\n",
    "\n",
    "\n",
    "cad12['Database.py'] = np.select([cad12['ClinVar'].notna() & cad12['MAX.AF'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].notna(), \n",
    "                                  cad12['MAX.AF'].isna() & cad12['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "\n",
    "cad63['Database.py'] = np.select([cad63['ClinVar'].notna() & cad63['MAX.AF'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].notna(), \n",
    "                                  cad63['MAX.AF'].isna() & cad63['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3428264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>35778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>32158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ClinVar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3428264\n",
       "1      No DB source    35778\n",
       "2  gnomAD & ClinVar    32158\n",
       "3           ClinVar        1"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'Database.py')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3641404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>33479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3641404\n",
       "1  gnomAD & ClinVar    33479\n",
       "2      No DB source    26131"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'Database.py')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature.type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA.position</th>\n",
       "      <th>CDS.position</th>\n",
       "      <th>Protein.position</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing.variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>REF.ALLELE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL.SOURCE</th>\n",
       "      <th>HGNC.ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE.SELECT</th>\n",
       "      <th>MANE.PLUS.CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT.ISOFORM</th>\n",
       "      <th>REFSEQ.MATCH</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>REFSEQ.OFFSET</th>\n",
       "      <th>GIVEN.REF</th>\n",
       "      <th>USED.REF</th>\n",
       "      <th>BAM.EDIT</th>\n",
       "      <th>GENE.PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVS.OFFSET</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>chr</th>\n",
       "      <th>GIVEN.REF.MATCH</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "      <th>keyID38</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>var.exclusive.btw.lines</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>gene.exclusive.btw.lines</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>Database.py</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_14398_TGT/-</td>\n",
       "      <td>chr1:14398-14400</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>1646-1648</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs756427959</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>TGT</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>TGT</td>\n",
       "      <td>TGT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000456328.2:n.1648_1650del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01817</td>\n",
       "      <td>0.002934</td>\n",
       "      <td>0.01613</td>\n",
       "      <td>0.024030</td>\n",
       "      <td>0.034360</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.025360</td>\n",
       "      <td>0.036360</td>\n",
       "      <td>0.030010</td>\n",
       "      <td>0.016860</td>\n",
       "      <td>0.019240</td>\n",
       "      <td>0.03636</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>14398</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_20317_A/-</td>\n",
       "      <td>chr1:20317</td>\n",
       "      <td>-</td>\n",
       "      <td>653635</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1320311790</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:38034</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>OK</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024540.1:n.205-1951del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.27440</td>\n",
       "      <td>0.107700</td>\n",
       "      <td>0.28910</td>\n",
       "      <td>0.307300</td>\n",
       "      <td>0.302200</td>\n",
       "      <td>0.347000</td>\n",
       "      <td>0.369400</td>\n",
       "      <td>0.314200</td>\n",
       "      <td>0.346100</td>\n",
       "      <td>0.286500</td>\n",
       "      <td>0.272700</td>\n",
       "      <td>0.36940</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>20317</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_104161_-/ACAC/ACACACAC</td>\n",
       "      <td>chr1:104160-104161</td>\n",
       "      <td>ACAC</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENST00000477740</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs372078516</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000477740.5:n.480+8536_480+8539dup</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.32460</td>\n",
       "      <td>0.226700</td>\n",
       "      <td>0.29700</td>\n",
       "      <td>0.342900</td>\n",
       "      <td>0.362700</td>\n",
       "      <td>0.466100</td>\n",
       "      <td>0.283200</td>\n",
       "      <td>0.449200</td>\n",
       "      <td>0.361800</td>\n",
       "      <td>0.333700</td>\n",
       "      <td>0.420600</td>\n",
       "      <td>0.46610</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>104160</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_133707_AGA/-</td>\n",
       "      <td>chr1:133707-133709</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENST00000477740</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1354754438</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>AGA</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>4490.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>AGA</td>\n",
       "      <td>AGA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.00245</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.001503</td>\n",
       "      <td>0.010940</td>\n",
       "      <td>0.010740</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.003401</td>\n",
       "      <td>0.002703</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.008426</td>\n",
       "      <td>0.01094</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>133707</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...</td>\n",
       "      <td>chr1:136603-136604</td>\n",
       "      <td>GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGTGTGAGGCAAGGGCTCG</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1557463817</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>3186.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02026</td>\n",
       "      <td>0.022410</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.021840</td>\n",
       "      <td>0.023380</td>\n",
       "      <td>0.012480</td>\n",
       "      <td>0.017270</td>\n",
       "      <td>0.035710</td>\n",
       "      <td>0.020320</td>\n",
       "      <td>0.008929</td>\n",
       "      <td>0.013200</td>\n",
       "      <td>0.03571</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>136603</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950092</th>\n",
       "      <td>chrX_156024517_C/T</td>\n",
       "      <td>chrX:156024517</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs55796665</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HOM</td>\n",
       "      <td>C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>8/9</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000359512.8:c.1234+245C&gt;T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>0.999900</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.00000</td>\n",
       "      <td>gnomADg_AMI,gnomADg_AMR,gnomADg_ASJ,gnomADg_EA...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156024517</td>\n",
       "      <td>NaN</td>\n",
       "      <td>8.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_156024517_C_T</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950093</th>\n",
       "      <td>chrX_156025116_G/A</td>\n",
       "      <td>chrX:156025116</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs3953915</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>G</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>9/9</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000359512.8:c.1303+16G&gt;A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.31330</td>\n",
       "      <td>0.188500</td>\n",
       "      <td>0.35090</td>\n",
       "      <td>0.302100</td>\n",
       "      <td>0.318000</td>\n",
       "      <td>0.037370</td>\n",
       "      <td>0.392500</td>\n",
       "      <td>0.289000</td>\n",
       "      <td>0.397600</td>\n",
       "      <td>0.310000</td>\n",
       "      <td>0.288100</td>\n",
       "      <td>0.40540</td>\n",
       "      <td>gnomADe_NFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156025116</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_156025116_G_A</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950094</th>\n",
       "      <td>chrX_156025186_G/C</td>\n",
       "      <td>chrX:156025186</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs144106392</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>G</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>9/9</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000359512.8:c.1304-55G&gt;C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01565</td>\n",
       "      <td>0.002506</td>\n",
       "      <td>0.01429</td>\n",
       "      <td>0.004662</td>\n",
       "      <td>0.007501</td>\n",
       "      <td>0.000192</td>\n",
       "      <td>0.061310</td>\n",
       "      <td>0.012740</td>\n",
       "      <td>0.019010</td>\n",
       "      <td>0.012040</td>\n",
       "      <td>0.041650</td>\n",
       "      <td>0.06131</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156025186</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9.0</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_156025186_G_C</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950095</th>\n",
       "      <td>chrX_156025429_T/C</td>\n",
       "      <td>chrX:156025429</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs189478274</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>T</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>55.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.11990</td>\n",
       "      <td>0.030570</td>\n",
       "      <td>0.21480</td>\n",
       "      <td>0.066790</td>\n",
       "      <td>0.177400</td>\n",
       "      <td>0.032690</td>\n",
       "      <td>0.329200</td>\n",
       "      <td>0.173200</td>\n",
       "      <td>0.149600</td>\n",
       "      <td>0.118900</td>\n",
       "      <td>0.236200</td>\n",
       "      <td>0.32920</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156025429</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_156025429_T_C</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2950096</th>\n",
       "      <td>chrX_156029856_C/G</td>\n",
       "      <td>chrX:156029856</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs12353930</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>C</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>4482.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.17900</td>\n",
       "      <td>0.081900</td>\n",
       "      <td>0.22300</td>\n",
       "      <td>0.204200</td>\n",
       "      <td>0.166700</td>\n",
       "      <td>0.368700</td>\n",
       "      <td>0.265400</td>\n",
       "      <td>0.183300</td>\n",
       "      <td>0.192100</td>\n",
       "      <td>0.160300</td>\n",
       "      <td>0.259600</td>\n",
       "      <td>0.36870</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chrX</td>\n",
       "      <td>156029856</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>X_156029856_C_G</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>gnomAD</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3496201 rows Ã— 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        Uploaded.variation            Location                                            Allele             Gene          Feature Feature.type                                   Consequence cDNA.position CDS.position Protein.position Amino.acids Codons Existing.variation      IND  ZYG REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS VARIANT.CLASS   SYMBOL SYMBOL.SOURCE     HGNC.ID                 BIOTYPE CANONICAL MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS CCDS             ENSP SWISSPROT        TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET GIVEN.REF USED.REF BAM.EDIT GENE.PHENO SIFT PolyPhen EXON INTRON DOMAINS miRNA                                     HGVSc HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF   MAX.AF                                        MAX.AF.POPS CLIN.SIG SOMATIC PHENO PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines                 MAF           MAF2  PATHO    VUS  BENIGN ClinVar Database.py\n",
       "0                                         chr1_14398_TGT/-    chr1:14398-14400                                                 -  ENSG00000290825  ENST00000456328   Transcript            non_coding_transcript_exon_variant     1646-1648          NaN              NaN           -      -        rs756427959  P732912  HET        TGT  MODIFIER       NaN      1     -      deletion  DDX11L2    EntrezGene           -                  lncRNA       YES           -                  -   1      -    -                -         -             -              -               -            -  Ensembl             -       TGT      TGT        -          -    -        -  3/3      -       -     -          ENST00000456328.2:n.1648_1650del     -           -     0.01817        0.002934         0.01613        0.024030        0.034360        0.002087        0.025360        0.036360        0.030010        0.016860        0.019240  0.03636                                        gnomADg_MID        -       -     -      -          -         -            -                  -                     -       chr1          14398         3.0           NaN   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Less common (<10%)     Rare (<5%)  False  False   False     NaN      gnomAD\n",
       "1                                           chr1_20317_A/-          chr1:20317                                                 -           653635      NR_024540.1   Transcript  intron_variant,non_coding_transcript_variant           NaN          NaN              NaN           -      -       rs1320311790  P732912  HET          A  MODIFIER       NaN     -1     -      deletion   WASH7P    EntrezGene  HGNC:38034  transcribed_pseudogene       YES           -                  -   -      -    -                -         -             -              -               -            -   RefSeq             -         A        A       OK          -    -        -    -   2/10       -     -                 NR_024540.1:n.205-1951del     -           -     0.27440        0.107700         0.28910        0.307300        0.302200        0.347000        0.369400        0.314200        0.346100        0.286500        0.272700  0.36940                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chr1          20317         NaN           2.0   1             True            NaN        NaN                NaN            NaN              NaN              False                     True               False                      True      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "2                              chr1_104161_-/ACAC/ACACACAC  chr1:104160-104161                                              ACAC  ENSG00000238009  ENST00000477740   Transcript  intron_variant,non_coding_transcript_variant           NaN          NaN              NaN           -      -        rs372078516  P732912  HET          -  MODIFIER       NaN     -1     -     insertion        -             -           -                  lncRNA       YES           -                  -   5      -    -                -         -             -              -               -            -  Ensembl             -         -        -        -          -    -        -    -    3/3       -     -  ENST00000477740.5:n.480+8536_480+8539dup     -           -     0.32460        0.226700         0.29700        0.342900        0.362700        0.466100        0.283200        0.449200        0.361800        0.333700        0.420600  0.46610                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -       chr1         104160         NaN           3.0   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "3                                        chr1_133707_AGA/-  chr1:133707-133709                                                 -  ENSG00000238009  ENST00000477740   Transcript                         upstream_gene_variant           NaN          NaN              NaN           -      -       rs1354754438  P732912  HET        AGA  MODIFIER    4490.0     -1     -      deletion        -             -           -                  lncRNA       YES           -                  -   5      -    -                -         -             -              -               -            -  Ensembl             -       AGA      AGA        -          -    -        -    -      -       -     -                                         -     -           -     0.00245        0.000491         0.00000        0.001503        0.010940        0.010740        0.000099        0.003401        0.002703        0.003115        0.008426  0.01094                                        gnomADg_ASJ        -       -     -      -          -         -            -                  -                     -       chr1         133707         NaN           NaN   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Less common (<10%)     Rare (<5%)  False  False   False     NaN      gnomAD\n",
       "4        chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...  chr1:136603-136604  GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGTGTGAGGCAAGGGCTCG  ENSG00000239906  ENST00000493797   Transcript                       downstream_gene_variant           NaN          NaN              NaN           -      -       rs1557463817  P732912  HET          -  MODIFIER    3186.0     -1     -     insertion        -             -           -                  lncRNA       YES           -                  -   2      -    -                -         -             -              -               -            -  Ensembl             -         -        -        -          -    -        -    -      -       -     -                                         -     -           -     0.02026        0.022410         0.00000        0.021840        0.023380        0.012480        0.017270        0.035710        0.020320        0.008929        0.013200  0.03571                                        gnomADg_MID        -       -     -      -          -         -            -                  -                     -       chr1         136603         NaN           NaN   1             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Less common (<10%)     Rare (<5%)  False  False   False     NaN      gnomAD\n",
       "...                                                    ...                 ...                                               ...              ...              ...          ...                                           ...           ...          ...              ...         ...    ...                ...      ...  ...        ...       ...       ...    ...   ...           ...      ...           ...         ...                     ...       ...         ...                ...  ..    ...  ...              ...       ...           ...            ...             ...          ...      ...           ...       ...      ...      ...        ...  ...      ...  ...    ...     ...   ...                                       ...   ...         ...         ...             ...             ...             ...             ...             ...             ...             ...             ...             ...             ...      ...                                                ...      ...     ...   ...    ...        ...       ...          ...                ...                   ...        ...            ...         ...           ...  ..              ...            ...        ...                ...            ...              ...                ...                      ...                 ...                       ...                 ...            ...    ...    ...     ...     ...         ...\n",
       "2950092                                 chrX_156024517_C/T      chrX:156024517                                                 T  ENSG00000182484  ENST00000359512   Transcript                                intron_variant           NaN          NaN              NaN           -      -         rs55796665  P732912  HOM          C  MODIFIER       NaN      1     -           SNV   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         C        C        -          -    -        -    -    8/9       -     -           ENST00000359512.8:c.1234+245C>T     -           -     1.00000        0.999900         1.00000        1.000000        1.000000        1.000000        1.000000        1.000000        1.000000        1.000000        1.000000  1.00000  gnomADg_AMI,gnomADg_AMR,gnomADg_ASJ,gnomADg_EA...        -       -     -      -          -         -            -                  -                     -       chrX      156024517         NaN           8.0   X             True              -        NaN                  -            NaN  X_156024517_C_T              False                     True                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "2950093                                 chrX_156025116_G/A      chrX:156025116                                                 A  ENSG00000182484  ENST00000359512   Transcript                                intron_variant           NaN          NaN              NaN           -      -          rs3953915  P732912  HET          G  MODIFIER       NaN      1     -           SNV   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         G        G        -          -    -        -    -    9/9       -     -            ENST00000359512.8:c.1303+16G>A     -           -     0.31330        0.188500         0.35090        0.302100        0.318000        0.037370        0.392500        0.289000        0.397600        0.310000        0.288100  0.40540                                        gnomADe_NFE        -       -     -      -          -         -            -                  -                     -       chrX      156025116         NaN           9.0   X             True              -        NaN                  -            NaN  X_156025116_G_A               True                    False                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "2950094                                 chrX_156025186_G/C      chrX:156025186                                                 C  ENSG00000182484  ENST00000359512   Transcript                                intron_variant           NaN          NaN              NaN           -      -        rs144106392  P732912  HET          G  MODIFIER       NaN      1     -           SNV   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         G        G        -          -    -        -    -    9/9       -     -            ENST00000359512.8:c.1304-55G>C     -           -     0.01565        0.002506         0.01429        0.004662        0.007501        0.000192        0.061310        0.012740        0.019010        0.012040        0.041650  0.06131                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chrX      156025186         NaN           9.0   X             True              -        NaN                  -            NaN  X_156025186_G_C              False                     True                True                     False  Less common (<10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "2950095                                 chrX_156025429_T/C      chrX:156025429                                                 C  ENSG00000182484  ENST00000359512   Transcript                       downstream_gene_variant           NaN          NaN              NaN           -      -        rs189478274  P732912  HET          T  MODIFIER      55.0      1     -           SNV   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         T        T        -          -    -        -    -      -       -     -                                         -     -           -     0.11990        0.030570         0.21480        0.066790        0.177400        0.032690        0.329200        0.173200        0.149600        0.118900        0.236200  0.32920                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -       chrX      156025429         NaN           NaN   X             True              -        NaN                  -            NaN  X_156025429_T_C               True                    False                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "2950096                                 chrX_156029856_C/G      chrX:156029856                                                 G  ENSG00000182484  ENST00000359512   Transcript                       downstream_gene_variant           NaN          NaN              NaN           -      -         rs12353930  P732912  HET          C  MODIFIER    4482.0      1     -           SNV   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         C        C        -          -    -        -    -      -       -     -                                         -     -           -     0.17900        0.081900         0.22300        0.204200        0.166700        0.368700        0.265400        0.183300        0.192100        0.160300        0.259600  0.36870                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -       chrX      156029856         NaN           NaN   X             True              -        NaN                  -            NaN  X_156029856_C_G              False                     True                True                     False      Common (>=10%)  Common (>=5%)  False  False   False     NaN      gnomAD\n",
       "\n",
       "[3496201 rows x 96 columns]"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>3542705</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>132178</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nan</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Count    count\n",
       "0  Common (>=5%)  3542705\n",
       "1     Rare (<5%)   132178\n",
       "2            nan    26131"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'MAF2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>3326323</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>134099</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>nan</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "           Count    count\n",
       "0  Common (>=5%)  3326323\n",
       "1     Rare (<5%)   134099\n",
       "2            nan    35779"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'MAF2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>3253032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>152860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>54530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nan</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Count    count\n",
       "0      Common (>=10%)  3253032\n",
       "1  Less common (<10%)   152860\n",
       "2          Rare (<1%)    54530\n",
       "3                 nan    35779"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'MAF')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>3462469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>157624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>54790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>nan</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Count    count\n",
       "0      Common (>=10%)  3462469\n",
       "1  Less common (<10%)   157624\n",
       "2          Rare (<1%)    54790\n",
       "3                 nan    26131"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'MAF')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33038\n",
       "2     VUS      330\n",
       "3   PATHO      111"
      ]
     },
     "execution_count": 72,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31769\n",
       "2     VUS      302\n",
       "3   PATHO       88"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho12"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py39",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
