{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1b39c79-b27c-4069-a620-f42108c99475",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !/usr/bin/env python3\n",
    "# -*- coding: utf-8 -*-\n",
    "\n",
    "# %%\n",
    "from all_funx import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70ae61df-cd64-4af0-8fe3-72f0b45ac3d8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "# import dfs from 1_clean_excel_summary.py\n",
    "path  = \"/Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-summarize/\"\n",
    "\n",
    "keepme = [\"Uploaded_variation\",\n",
    "\"Feature_type\",\n",
    "\"Consequence\",\n",
    "#\"Protein_position\",\n",
    "\"Amino_acids\",\n",
    "\"Existing_variation\",\n",
    "\"IND\",\n",
    "\"ZYG\",\n",
    "\"IMPACT\",\n",
    "#\"DISTANCE\",\n",
    "\"VARIANT_CLASS\",\n",
    "\"SYMBOL\",\n",
    "\"BIOTYPE\",\n",
    "\"SWISSPROT\",\n",
    "\"GENE_PHENO\",\n",
    "\"EXON\",\n",
    "\"INTRON\",\n",
    "\"DOMAINS\",\n",
    "\"miRNA\",\n",
    "\"gnomADg_AF\",\n",
    "\"gnomADg_AFR_AF\",\n",
    "\"gnomADg_AMI_AF\",\n",
    "\"gnomADg_AMR_AF\",\n",
    "\"gnomADg_ASJ_AF\",\n",
    "\"gnomADg_EAS_AF\",\n",
    "\"gnomADg_FIN_AF\",\n",
    "\"gnomADg_MID_AF\",\n",
    "\"gnomADg_NFE_AF\",\n",
    "\"gnomADg_OTH_AF\",\n",
    "\"gnomADg_SAS_AF\",\n",
    "\"MAX_AF\",\n",
    "\"MAX_AF_POPS\",\n",
    "\"CLIN_SIG\",\n",
    "\"PUBMED\",\n",
    "\"MOTIF_NAME\",\n",
    "\"HIGH_INF_POS\",\n",
    "\"MOTIF_SCORE_CHANGE\",\n",
    "\"TRANSCRIPTION_FACTORS\",\n",
    "\"EXONnumber\",\n",
    "\"INTRONnumber\"]\n",
    "\n",
    "keepme2 = keepme + [\"SIFTprediction\",\"SIFTscore\", \"PolyPhenprediction\", \"PolyPhenscore\"]\n",
    "\n",
    "indel12 = pd.read_csv('1_clean_vep_output-indel-cad12-v2.csv', usecols = keepme )\n",
    "snp12 = pd.read_csv('1_clean_vep_output-snp-cad12-v2.csv', usecols = keepme2 )\n",
    "indel63 = pd.read_csv('1_clean_vep_output-indel-cad63-v2.csv', usecols = keepme )\n",
    "snp63 = pd.read_csv('1_clean_vep_output-snp-cad63-v2.csv', usecols = keepme2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cee18a68-79c1-483b-a60f-37b6780acea9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "cad12 = pd.concat([indel12, snp12], axis=0)\n",
    "cad63 = pd.concat([indel63, snp63], axis=0)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "72ab1149-8525-4216-8533-d919abb32869",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_14398_TGT/-</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs756427959</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>deletion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01817</td>\n",
       "      <td>0.002934</td>\n",
       "      <td>0.01613</td>\n",
       "      <td>0.024030</td>\n",
       "      <td>0.03436</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.025360</td>\n",
       "      <td>0.036360</td>\n",
       "      <td>0.030010</td>\n",
       "      <td>0.016860</td>\n",
       "      <td>0.019240</td>\n",
       "      <td>0.03636</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_20317_A/-</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1320311790</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.27440</td>\n",
       "      <td>0.107700</td>\n",
       "      <td>0.28910</td>\n",
       "      <td>0.307300</td>\n",
       "      <td>0.30220</td>\n",
       "      <td>0.347000</td>\n",
       "      <td>0.369400</td>\n",
       "      <td>0.314200</td>\n",
       "      <td>0.346100</td>\n",
       "      <td>0.286500</td>\n",
       "      <td>0.272700</td>\n",
       "      <td>0.36940</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_104161_-/ACAC/ACACACAC</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs372078516</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.32460</td>\n",
       "      <td>0.226700</td>\n",
       "      <td>0.29700</td>\n",
       "      <td>0.342900</td>\n",
       "      <td>0.36270</td>\n",
       "      <td>0.466100</td>\n",
       "      <td>0.283200</td>\n",
       "      <td>0.449200</td>\n",
       "      <td>0.361800</td>\n",
       "      <td>0.333700</td>\n",
       "      <td>0.420600</td>\n",
       "      <td>0.46610</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_133707_AGA/-</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1354754438</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>deletion</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.00245</td>\n",
       "      <td>0.000491</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.001503</td>\n",
       "      <td>0.01094</td>\n",
       "      <td>0.010740</td>\n",
       "      <td>0.000099</td>\n",
       "      <td>0.003401</td>\n",
       "      <td>0.002703</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.008426</td>\n",
       "      <td>0.01094</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1557463817</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02026</td>\n",
       "      <td>0.022410</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.021840</td>\n",
       "      <td>0.02338</td>\n",
       "      <td>0.012480</td>\n",
       "      <td>0.017270</td>\n",
       "      <td>0.035710</td>\n",
       "      <td>0.020320</td>\n",
       "      <td>0.008929</td>\n",
       "      <td>0.013200</td>\n",
       "      <td>0.03571</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Uploaded_variation Feature_type                                   Consequence Amino_acids Existing_variation      IND  ZYG    IMPACT VARIANT_CLASS   SYMBOL                 BIOTYPE SWISSPROT GENE_PHENO EXON INTRON DOMAINS miRNA  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF   MAX_AF  MAX_AF_POPS CLIN_SIG PUBMED MOTIF_NAME HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  EXONnumber  INTRONnumber SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore\n",
       "0                                   chr1_14398_TGT/-   Transcript            non_coding_transcript_exon_variant           -        rs756427959  P732912  HET  MODIFIER      deletion  DDX11L2                  lncRNA         -          -  3/3      -       -     -     0.01817        0.002934         0.01613        0.024030         0.03436        0.002087        0.025360        0.036360        0.030010        0.016860        0.019240  0.03636  gnomADg_MID        -      -          -            -                  -                     -         3.0           NaN            NaN        NaN                NaN            NaN\n",
       "1                                     chr1_20317_A/-   Transcript  intron_variant,non_coding_transcript_variant           -       rs1320311790  P732912  HET  MODIFIER      deletion   WASH7P  transcribed_pseudogene         -          -    -   2/10       -     -     0.27440        0.107700         0.28910        0.307300         0.30220        0.347000        0.369400        0.314200        0.346100        0.286500        0.272700  0.36940  gnomADg_FIN        -      -          -            -                  -                     -         NaN           2.0            NaN        NaN                NaN            NaN\n",
       "2                        chr1_104161_-/ACAC/ACACACAC   Transcript  intron_variant,non_coding_transcript_variant           -        rs372078516  P732912  HET  MODIFIER     insertion        -                  lncRNA         -          -    -    3/3       -     -     0.32460        0.226700         0.29700        0.342900         0.36270        0.466100        0.283200        0.449200        0.361800        0.333700        0.420600  0.46610  gnomADg_EAS        -      -          -            -                  -                     -         NaN           3.0            NaN        NaN                NaN            NaN\n",
       "3                                  chr1_133707_AGA/-   Transcript                         upstream_gene_variant           -       rs1354754438  P732912  HET  MODIFIER      deletion        -                  lncRNA         -          -    -      -       -     -     0.00245        0.000491         0.00000        0.001503         0.01094        0.010740        0.000099        0.003401        0.002703        0.003115        0.008426  0.01094  gnomADg_ASJ        -      -          -            -                  -                     -         NaN           NaN            NaN        NaN                NaN            NaN\n",
       "4  chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...   Transcript                       downstream_gene_variant           -       rs1557463817  P732912  HET  MODIFIER     insertion        -                  lncRNA         -          -    -      -       -     -     0.02026        0.022410         0.00000        0.021840         0.02338        0.012480        0.017270        0.035710        0.020320        0.008929        0.013200  0.03571  gnomADg_MID        -      -          -            -                  -                     -         NaN           NaN            NaN        NaN                NaN            NaN"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "cad12.head()    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c140278-b515-4084-8ea3-83a284c5fa4c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_13418_-/GAGA</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs777038595</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>insertion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.12360</td>\n",
       "      <td>0.03366</td>\n",
       "      <td>0.22650</td>\n",
       "      <td>0.15070</td>\n",
       "      <td>0.1875</td>\n",
       "      <td>0.07761</td>\n",
       "      <td>0.1684</td>\n",
       "      <td>0.13060</td>\n",
       "      <td>0.17460</td>\n",
       "      <td>0.12620</td>\n",
       "      <td>0.17310</td>\n",
       "      <td>0.2265</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_15904_-/C</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs557514207</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HOM</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>insertion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.55450</td>\n",
       "      <td>0.20840</td>\n",
       "      <td>0.80110</td>\n",
       "      <td>0.60690</td>\n",
       "      <td>0.7156</td>\n",
       "      <td>0.98550</td>\n",
       "      <td>0.7871</td>\n",
       "      <td>0.57970</td>\n",
       "      <td>0.82570</td>\n",
       "      <td>0.50640</td>\n",
       "      <td>0.74940</td>\n",
       "      <td>0.9855</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>35533190</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_19191_C/-</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1346737557</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>deletion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.06882</td>\n",
       "      <td>0.02787</td>\n",
       "      <td>0.08072</td>\n",
       "      <td>0.08207</td>\n",
       "      <td>0.1102</td>\n",
       "      <td>0.01799</td>\n",
       "      <td>0.1046</td>\n",
       "      <td>0.03571</td>\n",
       "      <td>0.09681</td>\n",
       "      <td>0.09414</td>\n",
       "      <td>0.04594</td>\n",
       "      <td>0.1102</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_61872_-/T</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs368183979</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>insertion</td>\n",
       "      <td>OR4F5</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.26970</td>\n",
       "      <td>0.15320</td>\n",
       "      <td>0.34200</td>\n",
       "      <td>0.24270</td>\n",
       "      <td>0.3713</td>\n",
       "      <td>0.03265</td>\n",
       "      <td>0.2996</td>\n",
       "      <td>0.27160</td>\n",
       "      <td>0.37930</td>\n",
       "      <td>0.26700</td>\n",
       "      <td>0.26690</td>\n",
       "      <td>0.3793</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_105775_T/-</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1236596781</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>deletion</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.26320</td>\n",
       "      <td>0.20450</td>\n",
       "      <td>0.33600</td>\n",
       "      <td>0.29930</td>\n",
       "      <td>0.3046</td>\n",
       "      <td>0.03975</td>\n",
       "      <td>0.3309</td>\n",
       "      <td>0.21430</td>\n",
       "      <td>0.30110</td>\n",
       "      <td>0.25420</td>\n",
       "      <td>0.21660</td>\n",
       "      <td>0.3360</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Uploaded_variation Feature_type                                   Consequence Amino_acids Existing_variation      IND  ZYG    IMPACT VARIANT_CLASS   SYMBOL         BIOTYPE SWISSPROT GENE_PHENO EXON INTRON DOMAINS miRNA  gnomADg_AF  gnomADg_AFR_AF  gnomADg_AMI_AF  gnomADg_AMR_AF  gnomADg_ASJ_AF  gnomADg_EAS_AF  gnomADg_FIN_AF  gnomADg_MID_AF  gnomADg_NFE_AF  gnomADg_OTH_AF  gnomADg_SAS_AF  MAX_AF  MAX_AF_POPS CLIN_SIG    PUBMED MOTIF_NAME HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS  EXONnumber  INTRONnumber SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore\n",
       "0  chr1_13418_-/GAGA   Transcript            non_coding_transcript_exon_variant           -        rs777038595  P931263  HET  MODIFIER     insertion  DDX11L2          lncRNA         -          -  3/3      -       -     -     0.12360         0.03366         0.22650         0.15070          0.1875         0.07761          0.1684         0.13060         0.17460         0.12620         0.17310  0.2265  gnomADg_AMI        -         -          -            -                  -                     -         3.0           NaN            NaN        NaN                NaN            NaN\n",
       "1     chr1_15904_-/C   Transcript                       downstream_gene_variant           -        rs557514207  P931263  HOM  MODIFIER     insertion  DDX11L2          lncRNA         -          -    -      -       -     -     0.55450         0.20840         0.80110         0.60690          0.7156         0.98550          0.7871         0.57970         0.82570         0.50640         0.74940  0.9855  gnomADg_EAS        -  35533190          -            -                  -                     -         NaN           NaN            NaN        NaN                NaN            NaN\n",
       "2     chr1_19191_C/-   Transcript                       downstream_gene_variant           -       rs1346737557  P931263  HET  MODIFIER      deletion  DDX11L2          lncRNA         -          -    -      -       -     -     0.06882         0.02787         0.08072         0.08207          0.1102         0.01799          0.1046         0.03571         0.09681         0.09414         0.04594  0.1102  gnomADg_ASJ        -         -          -            -                  -                     -         NaN           NaN            NaN        NaN                NaN            NaN\n",
       "3     chr1_61872_-/T   Transcript                         upstream_gene_variant           -        rs368183979  P931263  HET  MODIFIER     insertion    OR4F5  protein_coding         -          -    -      -       -     -     0.26970         0.15320         0.34200         0.24270          0.3713         0.03265          0.2996         0.27160         0.37930         0.26700         0.26690  0.3793  gnomADg_NFE        -         -          -            -                  -                     -         NaN           NaN            NaN        NaN                NaN            NaN\n",
       "4    chr1_105775_T/-   Transcript  intron_variant,non_coding_transcript_variant           -       rs1236596781  P931263  HET  MODIFIER      deletion        -          lncRNA         -          -    -    3/3       -     -     0.26320         0.20450         0.33600         0.29930          0.3046         0.03975          0.3309         0.21430         0.30110         0.25420         0.21660  0.3360  gnomADg_AMI        -         -          -            -                  -                     -         NaN           3.0            NaN        NaN                NaN            NaN"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "cad63.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8a2b59f1-f2e0-447e-aa33-b0da2e84e066",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1525780 1730593 1970421\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "# create a list of unique 'Uploaded_variation' in both dataframes\n",
    "cad12_variations = cad12['Uploaded_variation'].unique()\n",
    "cad63_variations = cad63['Uploaded_variation'].unique()\n",
    "\n",
    "# create a separate list for variations that are exclusively in cad12\n",
    "cad12_exclusive_variations = list(set(cad12_variations) - set(cad63_variations))\n",
    "\n",
    "# create a separate list for variations that are exclusively in cad63\n",
    "cad63_exclusive_variations = list(set(cad63_variations) - set(cad12_variations))\n",
    "\n",
    "# create a separate list for variations that are in both cad12 and cad63\n",
    "common_variations = list(set(cad12_variations) & set(cad63_variations))\n",
    "\n",
    "print( len(cad12_exclusive_variations), len(cad63_exclusive_variations), len(common_variations))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "215dc443-1be2-44a3-9be3-72d288b5fd6f",
   "metadata": {},
   "outputs": [],
   "source": [
    "# %%\n",
    "\n",
    "# Optimize the code by avoiding unnecessary function calls and repeated operations\n",
    "cad12['var_in_both_lines'] = cad12['Uploaded_variation'].isin(common_variations)\n",
    "cad12['var_exclusive_btw_lines'] = cad12['Uploaded_variation'].isin(cad12_exclusive_variations)\n",
    "cad63['var_in_both_lines'] = cad63['Uploaded_variation'].isin(common_variations)\n",
    "cad63['var_exclusive_btw_lines'] = cad63['Uploaded_variation'].isin(cad63_exclusive_variations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c243360f-45e8-4ef7-9aa5-32beb8340a95",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "809 743 28137\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "# create a list of unique 'SYMBOL' in both dataframes\n",
    "cad12_genes = cad12['SYMBOL'].unique()\n",
    "cad63_genes = cad63['SYMBOL'].unique()\n",
    "\n",
    "# create a separate list for SYMBOL that are exclusively in cad12\n",
    "cad12_exclusive = list(set(cad12_genes) - set(cad63_genes))\n",
    "\n",
    "# create a separate list for SYMBOL that are exclusively in cad63\n",
    "cad63_exclusive = list(set(cad63_genes) - set(cad12_genes))\n",
    "\n",
    "# create a separate list for SYMBOL that are in both cad12 and cad63\n",
    "common_genes = list(set(cad12_genes) & set(cad63_genes))\n",
    "\n",
    "cad12['gene_in_both_lines'] = cad12['SYMBOL'].isin(common_genes)\n",
    "cad12['gene_exclusive_btw_lines'] = cad12['SYMBOL'].isin(cad12_exclusive)\n",
    "cad63['gene_in_both_lines'] = cad63['SYMBOL'].isin(common_genes)\n",
    "cad63['gene_exclusive_btw_lines'] = cad63['SYMBOL'].isin(cad63_exclusive)\n",
    "\n",
    "print( len(cad12_exclusive), len(cad63_exclusive), len(common_genes))\n",
    "\n",
    "\n",
    "save_describe(cad12, '4-cad12')\n",
    "save_describe(cad63, '4-cad63')\n",
    "\n",
    "cad12.to_csv('4-combined-cad12.csv', index=False)\n",
    "cad63.to_csv('4-combined-cad63.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "00513d17-cd37-44f3-bcc9-40d223eeea13",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Feature_type', 'Consequence', 'Amino_acids', 'Existing_variation', 'IND', 'ZYG', 'IMPACT', 'VARIANT_CLASS', 'SYMBOL', 'BIOTYPE', 'SWISSPROT', 'GENE_PHENO', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'PUBMED', 'MOTIF_NAME', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS', 'EXONnumber', 'INTRONnumber', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "print(cad12.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "85db9371-b1bf-46b2-ae49-7006305d1de6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Feature_type', 'Consequence', 'Amino_acids', 'Existing_variation', 'IND', 'ZYG', 'IMPACT', 'VARIANT_CLASS', 'SYMBOL', 'BIOTYPE', 'SWISSPROT', 'GENE_PHENO', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'PUBMED', 'MOTIF_NAME', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE', 'TRANSCRIPTION_FACTORS', 'EXONnumber', 'INTRONnumber', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'], dtype='object')\n"
     ]
    }
   ],
   "source": [
    "# %%\n",
    "\n",
    "print(cad63.columns)"
   ]
  }
 ],
 "metadata": {
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": 3
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
