{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Format and Label VEP annotated WGS results for CAD12 and CAD63\n",
    "**last update : july 9 2024 3pm**\n",
    "\n",
    "### code version info \n",
    "1.  clinvar col with labels is not star reviewed, v1overestimates  number of pathogenic variants because it adds label if the phrase occurs. for high confidendence pathogenic variants, check the clin.sig column to confirm absence of 'benign' or 'uncertain significance' phrases (DONE)\n",
    "2. added Database column with python instead of  R to keep all label functions together (DONE)\n",
    "3. note: keyID38 is only for SNVs\n",
    "4. note: var.in.both.lines has T/F values\n",
    "5. note: gene.in.both.lines has T/F values\n",
    "6. note: MAF is based on MAX.AF column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "from all_funx import *      # conda env  py39 (Python 3.9.-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "216ec7fd-050d-4a84-97d7-018da0cf6acf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(546104, 80)\n",
      "(2950097, 85)\n",
      "(562353, 80)\n",
      "(3138661, 85)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# import dfs from 1_clean_excel_summary.py\n",
    "\n",
    "path  = \"/Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-interpret/\"\n",
    "\n",
    "\n",
    "# import cad 12 wgs results for snp indel\n",
    "\n",
    "indel12 = pd.read_csv('1_clean_vep_output-indel-cad12-v2.csv', low_memory=False)\n",
    "snp12 = pd.read_csv('1_clean_vep_output-snp-cad12-v2.csv', low_memory=False)\n",
    "\n",
    "print(indel12.shape)    # (546104, 80)\n",
    "print(snp12.shape)      # (2950097, 85)\n",
    "\n",
    "\n",
    "# import cad 63 wgs results for snp indel\n",
    "\n",
    "indel63 = pd.read_csv('1_clean_vep_output-indel-cad63-v2.csv', low_memory=False)\n",
    "snp63 = pd.read_csv('1_clean_vep_output-snp-cad63-v2.csv', low_memory=False)\n",
    "\n",
    "print(indel63.shape)        # (562353, 80)\n",
    "print(snp63.shape)          # (3138661, 85)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. CAD63 has 16,249 more INDELs and 188,564 more SNVs than CAD12 :\n",
    "\n",
    "* cad12 :\n",
    "    * INDEL = 546,104\n",
    "    * SNV = 2,950,097\n",
    "\n",
    "* cad63 :\n",
    "    * INDEL = 562,353\n",
    "    * SNV = 3,138,661\n",
    "\n",
    "* difference INDEL : 562353-546104 = 16249\n",
    "\n",
    "* difference SNV : 3138661-2950097 = 188564\n",
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "978cedfb-f919-4b78-bbe7-d9c4a91e8218",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 85)\n",
      "(3701014, 85)\n"
     ]
    }
   ],
   "source": [
    "# make cad12 and cad63 complete dfs\n",
    "\n",
    "cad12 = pd.concat([indel12, snp12], axis=0)\n",
    "cad63 = pd.concat([indel63, snp63], axis=0)\n",
    "\n",
    "print(cad12.shape)\n",
    "print(cad63.shape)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "---\n",
    "\n",
    "# [1]  label  Exclusive VARIANT and Exclusive SYMBOL with variant btw CAD12 vs CAD63\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7a387389-81e3-45a5-a199-8832a3d192e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1525780 1730593 1970421\n"
     ]
    }
   ],
   "source": [
    "# create list of unique 'Uploaded_variation'\n",
    "\n",
    "cad12_variations = cad12['Uploaded_variation'].unique()\n",
    "cad63_variations = cad63['Uploaded_variation'].unique()\n",
    "\n",
    "# create  list for  exclusively in cad12\n",
    "\n",
    "cad12_exclusive_variations = list(set(cad12_variations) - set(cad63_variations))\n",
    "\n",
    "\n",
    "# create  list for  exclusively in cad63\n",
    "\n",
    "cad63_exclusive_variations = list(set(cad63_variations) - set(cad12_variations))\n",
    "\n",
    "# create list for in both lines\n",
    "\n",
    "common_variations = list(set(cad12_variations) & set(cad63_variations))\n",
    "\n",
    "\n",
    "print( len(cad12_exclusive_variations), len(cad63_exclusive_variations), len(common_variations)) "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result.Exclusive and Shared VARIANT :\n",
    "\n",
    "###  CAD12 minus CAD63 =    1,525,780\n",
    "###  CAD63 minus CAD12 =    1,730,593\n",
    "## Shared =                 1,970,421\n",
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5c17f2ee-ba76-4812-83c4-dc185fed2e7f",
   "metadata": {},
   "outputs": [],
   "source": [
    "#  dfs for CAD12 and CAD63\n",
    "\n",
    "# # TODO drop redundant column labeling function calls\n",
    "\n",
    "cad12['var_in_both_lines'] = cad12['Uploaded_variation'].isin(common_variations)\n",
    "cad12['var_exclusive_btw_lines'] = cad12['Uploaded_variation'].isin(cad12_exclusive_variations)\n",
    "\n",
    "cad63['var_in_both_lines'] = cad63['Uploaded_variation'].isin(common_variations)\n",
    "cad63['var_exclusive_btw_lines'] = cad63['Uploaded_variation'].isin(cad63_exclusive_variations)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAD12\n",
      "['var_in_both_lines', 'var_exclusive_btw_lines']\n",
      "\n",
      "var_in_both_lines        False    1525780\n",
      "                         True     1970421\n",
      "var_exclusive_btw_lines  False    1970421\n",
      "                         True     1525780\n",
      "dtype: int64\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "CAD63\n",
      "['var_in_both_lines', 'var_exclusive_btw_lines']\n",
      "\n",
      "var_in_both_lines        False    1730593\n",
      "                         True     1970421\n",
      "var_exclusive_btw_lines  False    1970421\n",
      "                         True     1730593\n",
      "dtype: int64\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print('CAD12')\n",
    "checkColumnsValues(cad12, ['var_in_both_lines', 'var_exclusive_btw_lines'])\n",
    "print()\n",
    "print()\n",
    "print()\n",
    "print('CAD63')\n",
    "checkColumnsValues(cad63, ['var_in_both_lines', 'var_exclusive_btw_lines'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**qc note : same counts  as above**\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "809 743 28137\n"
     ]
    }
   ],
   "source": [
    "# create list of unique 'SYMBOL' in both dataframes\n",
    "cad12_genes = cad12['SYMBOL'].unique()\n",
    "cad63_genes = cad63['SYMBOL'].unique()\n",
    "\n",
    "# creat list for SYMBOL  both cad12 and cad63\n",
    "common_genes = list(set(cad12_genes) & set(cad63_genes))\n",
    "\n",
    "# create list for SYMBOL   exclusively in cad12\n",
    "cad12_exclusive = list(set(cad12_genes) - set(cad63_genes))\n",
    "cad12['gene_in_both_lines'] = cad12['SYMBOL'].isin(common_genes)\n",
    "cad12['gene_exclusive_btw_lines'] = cad12['SYMBOL'].isin(cad12_exclusive)\n",
    "\n",
    "# create list for SYMBOL   exclusively in cad63\n",
    "cad63_exclusive = list(set(cad63_genes) - set(cad12_genes))\n",
    "cad63['gene_in_both_lines'] = cad63['SYMBOL'].isin(common_genes)\n",
    "cad63['gene_exclusive_btw_lines'] = cad63['SYMBOL'].isin(cad63_exclusive)\n",
    "\n",
    "\n",
    "print( len(cad12_exclusive), len(cad63_exclusive), len(common_genes)) \n",
    "#809 743 28137"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result.Exclusive and Shared SYMBOL : \n",
    "\n",
    "###  CAD12 Exclusive  =     809\n",
    "###  CAD63 mExclusive   =   743\n",
    "### Shared =                 28,137\n",
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cf118c35-fce5-4ab0-a2f9-786b5bf97c08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 89)\n",
      "(3701014, 89)\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(cad12.shape)      # (3496201, 89)\n",
    "print(cad63.shape)      #(3701014, 89)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result.Exclusive and Shared SYMBOL *(REDUNDANT - counts are unique to variant level NOT gene level and below is variant level counts)*\n",
    "\n",
    "```python\n",
    "checkColumnsValues(cad12, ['gene_in_both_lines', 'gene_exclusive_btw_lines'])\n",
    "print()\n",
    "print()\n",
    "print()\n",
    "print()\n",
    "checkColumnsValues(cad63, ['gene_in_both_lines', 'gene_exclusive_btw_lines'])\n",
    "```\n",
    "\n",
    "```\n",
    "---\n",
    "### CAD12 Shared =                3,491,756\n",
    "###  CAD12 Exclusive  =           4,445\n",
    "### CAD12 TOTAL shape =             3,496,201\n",
    "---\n",
    "### CAD63 Shared =                3,697,611\n",
    "###  CAD63 Exclusive   =          3,403\n",
    "### CAD63 TOTAL shape =             3,701,014\n",
    "```\n",
    "\n",
    "--- \n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "fd8a5abb-4757-46ec-bde4-f13e6b8bd43d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# fix column names by replacing _ with .\n",
    "\n",
    "cad12.columns = cad12.columns.str.replace('_', '.')\n",
    "cad63.columns = cad63.columns.str.replace('_', '.')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "---\n",
    "\n",
    "# [2] label  gnomad MAF thresholds\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "20823b15-b5ec-4325-87f2-6beaeaf95667",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add MAF column 3 groups :  if MAX.AF is < 0.01 label Rare, if less than 0.1 label less common, if greater than or equal  to 0.1 label common, else label empty\n",
    "\n",
    "cad12['MAF'] = np.select([cad12['MAX.AF'] < 0.01, \n",
    "                          (cad12['MAX.AF'] >= 0.01) & (cad12['MAX.AF'] < 0.1),\n",
    "                            cad12['MAX.AF'] >= 0.1], \n",
    "                            ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], \n",
    "                            default = np.nan)\n",
    "\n",
    "cad63['MAF'] = np.select([cad63['MAX.AF'] < 0.01, \n",
    "                          (cad63['MAX.AF'] >= 0.01) & (cad63['MAX.AF'] < 0.1), \n",
    "                          cad63['MAX.AF'] >= 0.1], \n",
    "                          ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], \n",
    "                          default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "23b1e0ee-86db-470d-bcf2-ee71e84a6714",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add MAF2 column 2 groups:  if MAX.AF is < 0.05 label Rare,  if greater than or equal  to 0.05 label  Common, else label empty\n",
    "\n",
    "cad12['MAF2'] = np.select([cad12['MAX.AF'] < 0.05, \n",
    "                           cad12['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'],\n",
    "                            default=np.nan)\n",
    "\n",
    "cad63['MAF2'] = np.select([cad63['MAX.AF'] < 0.05, \n",
    "                           cad63['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'], \n",
    "                           default=np.nan)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. gnomad MAF label counts :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CAD12\n",
      "['MAF', 'MAF2']\n",
      "\n",
      "MAF   Common (>=10%)        3253032.0\n",
      "      Less common (<10%)     152860.0\n",
      "      Rare (<1%)              54530.0\n",
      "      nan                     35779.0\n",
      "MAF2  Common (>=5%)         3326323.0\n",
      "      Rare (<5%)             134099.0\n",
      "      nan                     35779.0\n",
      "dtype: float64\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "CAD63\n",
      "['MAF', 'MAF2']\n",
      "\n",
      "MAF   Common (>=10%)        3462469.0\n",
      "      Less common (<10%)     157624.0\n",
      "      Rare (<1%)              54790.0\n",
      "      nan                     26131.0\n",
      "MAF2  Common (>=5%)         3542705.0\n",
      "      Rare (<5%)             132178.0\n",
      "      nan                     26131.0\n",
      "dtype: float64\n",
      "\n"
     ]
    }
   ],
   "source": [
    "print(\"CAD12\")\n",
    "checkColumnsValues(cad12, ['MAF', 'MAF2'])\n",
    "print()\n",
    "print()\n",
    "print()\n",
    "print()\n",
    "print(\"CAD63\")\n",
    "checkColumnsValues(cad63, ['MAF', 'MAF2'])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "#  [3] label PATHO, BENIGN, VUS if any word found in CLIN.SIG & label ClinVar final based on hierarchy\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "2c3e0494-4a2d-4421-aae3-d8d901cf6d36",
   "metadata": {},
   "outputs": [],
   "source": [
    "# call add  ClinVar label functions\n",
    "\n",
    "def label_clin_sig(df):\n",
    "    # function creates 3 new cols - PATHO VUS and BENIGN by checking CLIN.SIG  for i.e.  'pathogenic' or 'likely pathogenic'   new column  'PATHO' label boolean\n",
    "    df['PATHO'] = df['CLIN.SIG'].str.contains('pathogenic|likely pathogenic', case=False)\n",
    "    df['VUS'] = df['CLIN.SIG'].str.contains('uncertain_significance', case=False)\n",
    "    df['BENIGN'] = df['CLIN.SIG'].str.contains('benign|likely_benign', case=False)\n",
    "    return df\n",
    "\n",
    "def label_clin_sig2(df):\n",
    "    # function creates one column ClinVar label final based on  heirarchy;\n",
    "         # ## v1 label_clin_sig2() commented out:        over-estimated number of pathogenic because if-else statement ordered by PATHO check first then VUS then BENIGN\n",
    "        # ## v2 label_clin_sig2() current version:      high Confidence number of pathogenic because if-else statement ordered by VUS first then BENIGN then PATHO\n",
    "    newcol = []\n",
    "    for row in df.itertuples():\n",
    "        if row.VUS:\n",
    "            newcol.append('VUS')\n",
    "        elif row.BENIGN:\n",
    "            newcol.append('BENIGN')\n",
    "        elif row.PATHO:\n",
    "            newcol.append('PATHO')\n",
    "        else:\n",
    "            newcol.append(np.nan) # append nan value instead of '' emptry string\n",
    "\n",
    "    df['ClinVar'] = newcol\n",
    "    return df\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# done TODO  new column should  contain 'VUS' and 'BENIGN' labels if the row contain 'benign' for 'BENIGN' and 'uncertain significance' for 'VUS'.\n",
    "\n",
    "cad12 = label_clin_sig(cad12)\n",
    "cad12 = label_clin_sig2(cad12)\n",
    "cad63 = label_clin_sig(cad63)\n",
    "cad63 = label_clin_sig2(cad63)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. CAD12 ClinVar Label (if patho checked FIRST) :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31835\n",
       "2     VUS      309\n",
       "3   PATHO       15"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')      # QC v1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>31835</td>\n",
       "      <td>3464042</td>\n",
       "      <td>15</td>\n",
       "      <td>309</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>31835</td>\n",
       "      <td>3464042</td>\n",
       "      <td>15</td>\n",
       "      <td>309</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P732912   31835  3464042     15  309  3496201\n",
       "Total     31835  3464042     15  309  3496201"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "crosstabit(cad12, 'IND', 'ClinVar', True)  # QC v1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. CAD12 ClinVar Label (if patho checked LAST) : "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31835\n",
       "2     VUS      309\n",
       "3   PATHO       15"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>31835</td>\n",
       "      <td>3464042</td>\n",
       "      <td>15</td>\n",
       "      <td>309</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>31835</td>\n",
       "      <td>3464042</td>\n",
       "      <td>15</td>\n",
       "      <td>309</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P732912   31835  3464042     15  309  3496201\n",
       "Total     31835  3464042     15  309  3496201"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad12, 'IND', 'ClinVar', True) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho12 = cad12[cad12['ClinVar'] == 'PATHO']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>PATHO</th>\n",
       "      <th>True</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "PATHO    True  Total\n",
       "IND                 \n",
       "P732912    15     15\n",
       "Total      15     15"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho12, 'IND', 'PATHO', True) # qc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>BENIGN</th>\n",
       "      <th>False</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "BENIGN   False  Total\n",
       "IND                  \n",
       "P732912     15     15\n",
       "Total       15     15"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho12, 'IND', 'BENIGN', True) # qc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>VUS</th>\n",
       "      <th>False</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "VUS      False  Total\n",
       "IND                  \n",
       "P732912     15     15\n",
       "Total       15     15"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho12, 'IND', 'VUS', True) # qc "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. CAD63 ClinVar Label (if patho labeled FIRST)  :"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33116\n",
       "2     VUS      346\n",
       "3   PATHO       17"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>33116</td>\n",
       "      <td>3667535</td>\n",
       "      <td>17</td>\n",
       "      <td>346</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>33116</td>\n",
       "      <td>3667535</td>\n",
       "      <td>17</td>\n",
       "      <td>346</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P931263   33116  3667535     17  346  3701014\n",
       "Total     33116  3667535     17  346  3701014"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad63, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. CAD63 ClinVar Label (if patho labeled LAST)  : "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33116\n",
       "2     VUS      346\n",
       "3   PATHO       17"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>33116</td>\n",
       "      <td>3667535</td>\n",
       "      <td>17</td>\n",
       "      <td>346</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>33116</td>\n",
       "      <td>3667535</td>\n",
       "      <td>17</td>\n",
       "      <td>346</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P931263   33116  3667535     17  346  3701014\n",
       "Total     33116  3667535     17  346  3701014"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad63, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "patho63 = cad63[cad63['ClinVar'] == 'PATHO']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>PATHO</th>\n",
       "      <th>True</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "PATHO    True  Total\n",
       "IND                 \n",
       "P931263    17     17\n",
       "Total      17     17"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho63, 'IND', 'PATHO', True)   # qc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>BENIGN</th>\n",
       "      <th>False</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "BENIGN   False  Total\n",
       "IND                  \n",
       "P931263     17     17\n",
       "Total       17     17"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho63, 'IND', 'BENIGN', True)  # qc "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>VUS</th>\n",
       "      <th>False</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "VUS      False  Total\n",
       "IND                  \n",
       "P931263     17     17\n",
       "Total       17     17"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(patho63, 'IND', 'VUS', True) # qc "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "keepme = ['IND','Uploaded.variation',    # 'SYMBOL.SOURCE',\n",
    "             'SYMBOL' ,  'gene.in.both.lines',   'ZYG',  \n",
    "\n",
    "          'VARIANT.CLASS', 'Consequence',   'Amino.acids', #'BIOTYPE',\n",
    "\n",
    "           'var.in.both.lines',\n",
    "          \n",
    "             'ClinVar',   'MAF', 'MAF2', \n",
    "           \n",
    "             'CLIN.SIG',  #'PATHO', 'VUS', 'BENIGN',\n",
    "        \n",
    "           'MAX.AF', 'MAX.AF.POPS', \n",
    "   \n",
    "           #'SOMATIC', 'PHENO', 'PUBMED',\n",
    "\n",
    "            'SIFT', 'PolyPhen',  \n",
    "              'SIFTprediction', #'SIFTscore',\n",
    "                'PolyPhenprediction', #'PolyPhenscore', \n",
    "\n",
    "            'TRANSCRIPTION.FACTORS',  'miRNA',  'DOMAINS', 'MOTIF.NAME', 'MOTIF.POS', 'MOTIF.SCORE.CHANGE', 'HIGH.INF.POS'  ]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Result. total ClinVar per CAD12 and CAD63 : "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31835\n",
       "2     VUS      309\n",
       "3   PATHO       15"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(15, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1636481</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr10_6080046_T/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.394600</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81986</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr1_75761116_T/G</td>\n",
       "      <td>ACADM</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>splice_region_variant,splice_polypyrimidine_tr...</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.000654</td>\n",
       "      <td>gnomADe_SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2499653</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr17_63491327_G/A</td>\n",
       "      <td>ACE</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>R/Q</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.012300</td>\n",
       "      <td>SAS</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.153)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1o86.A,PDB-ENSP_mappings:1o8...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139169</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr1_161223893_G/A</td>\n",
       "      <td>APOA2</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.908800</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269741</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr9_27573524_GCCCCG/GCCCCGGCCCCGGCCCCGGCCCCGG...</td>\n",
       "      <td>C9orf72</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.133000</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2760537</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr20_62837227_G/A</td>\n",
       "      <td>COL9A3</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>R/H</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.001000</td>\n",
       "      <td>EAS</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>probably_damaging(0.994)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-Q14050-F1.A,Pfam:PF01391...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59969</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr1_53906216_G/A</td>\n",
       "      <td>DIO1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>M/I</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.024500</td>\n",
       "      <td>SAS</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>probably_damaging(0.978)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:3.40.30.10,Pfam:PF00837,PIRSF:PIRSF0013...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1333456</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr7_150742750_T/C</td>\n",
       "      <td>GIMAP5</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>L/P</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>0.003175</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.992)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:3.40.50.300,PDB-ENSP_mappings:6z3e.A,AF...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1311251</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr7_128938247_T/G</td>\n",
       "      <td>IRF5</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic,risk_factor</td>\n",
       "      <td>0.749000</td>\n",
       "      <td>EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2626072</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr19_15187315_G/A</td>\n",
       "      <td>NOTCH3</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>R/C</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>pathogenic,pathogenic/likely_pathogenic,likely...</td>\n",
       "      <td>0.004140</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>tolerated(0.05)</td>\n",
       "      <td>possibly_damaging(0.457)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:2.10.25.10,AFDB-ENSP_mappings:AF-Q9UM47...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446035</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr17_30237267_GGGCTGCAGGGGGGATGCTGGGGGTGCAGGG...</td>\n",
       "      <td>SLC6A4</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>deletion</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.594900</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971059</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_140557240_G/C</td>\n",
       "      <td>SRA1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>Q/E</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.019880</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>-</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PANTHER:PTHR18834,MobiDB_lite:mobidb-lite</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2486221</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr17_47731462_T/C</td>\n",
       "      <td>TBX21</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>risk_factor,pathogenic</td>\n",
       "      <td>0.417200</td>\n",
       "      <td>SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>451901</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr2_233758936_A/C</td>\n",
       "      <td>UGT1A4</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.761000</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163796</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_83104943_C/-</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>deletion</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>I/X</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>0.003165</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1fu1.A,PDB-ENSP_mappings:1fu...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND                                 Uploaded.variation   SYMBOL  gene.in.both.lines  ZYG        VARIANT.CLASS                                        Consequence Amino.acids  var.in.both.lines ClinVar                 MAF           MAF2                                           CLIN.SIG    MAX.AF  MAX.AF.POPS               SIFT                  PolyPhen SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "1636481  P732912                                  chr10_6080046_T/A        -                True  HOM                  SNV                                 intergenic_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.394600  gnomADg_ASJ                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "81986    P732912                                  chr1_75761116_T/G    ACADM                True  HET                  SNV  splice_region_variant,splice_polypyrimidine_tr...           -              False   PATHO          Rare (<1%)     Rare (<5%)                                         pathogenic  0.000654  gnomADe_SAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2499653  P732912                                 chr17_63491327_G/A      ACE                True  HET                  SNV                                   missense_variant         R/Q              False   PATHO  Less common (<10%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.012300          SAS     deleterious(0)             benign(0.153)    deleterious             benign                     -     -  PDB-ENSP_mappings:1o86.A,PDB-ENSP_mappings:1o8...          -         -                  -            -\n",
       "139169   P732912                                 chr1_161223893_G/A    APOA2                True  HET                  SNV                              upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.908800  gnomADg_EAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "269741   P732912  chr9_27573524_GCCCCG/GCCCCGGCCCCGGCCCCGGCCCCGG...  C9orf72                True  HET  sequence_alteration                              upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.133000  gnomADg_EAS                  -                         -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "2760537  P732912                                 chr20_62837227_G/A   COL9A3                True  HET                  SNV                                   missense_variant         R/H              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.001000          EAS  deleterious(0.02)  probably_damaging(0.994)    deleterious  probably_damaging                     -     -  AFDB-ENSP_mappings:AF-Q14050-F1.A,Pfam:PF01391...          -         -                  -            -\n",
       "59969    P732912                                  chr1_53906216_G/A     DIO1                True  HET                  SNV                                   missense_variant         M/I              False   PATHO  Less common (<10%)     Rare (<5%)                                         pathogenic  0.024500          SAS  deleterious(0.02)  probably_damaging(0.978)    deleterious  probably_damaging                     -     -  Gene3D:3.40.30.10,Pfam:PF00837,PIRSF:PIRSF0013...          -         -                  -            -\n",
       "1333456  P732912                                 chr7_150742750_T/C   GIMAP5                True  HET                  SNV                                   missense_variant         L/P              False   PATHO          Rare (<1%)     Rare (<5%)                       likely_pathogenic,pathogenic  0.003175  gnomADg_NFE     deleterious(0)  probably_damaging(0.992)    deleterious  probably_damaging                     -     -  Gene3D:3.40.50.300,PDB-ENSP_mappings:6z3e.A,AF...          -         -                  -            -\n",
       "1311251  P732912                                 chr7_128938247_T/G     IRF5                True  HET                  SNV                                     intron_variant           -               True   PATHO      Common (>=10%)  Common (>=5%)                             pathogenic,risk_factor  0.749000          EAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2626072  P732912                                 chr19_15187315_G/A   NOTCH3                True  HET                  SNV                                   missense_variant         R/C              False   PATHO          Rare (<1%)     Rare (<5%)  pathogenic,pathogenic/likely_pathogenic,likely...  0.004140  gnomADe_EAS    tolerated(0.05)  possibly_damaging(0.457)      tolerated  possibly_damaging                     -     -  Gene3D:2.10.25.10,AFDB-ENSP_mappings:AF-Q9UM47...          -         -                  -            -\n",
       "446035   P732912  chr17_30237267_GGGCTGCAGGGGGGATGCTGGGGGTGCAGGG...   SLC6A4                True  HET             deletion                              upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.594900  gnomADg_EAS                  -                         -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "971059   P732912                                 chr5_140557240_G/C     SRA1                True  HET                  SNV                                   missense_variant         Q/E              False   PATHO  Less common (<10%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.019880  gnomADg_ASJ     deleterious(0)                         -    deleterious                  -                     -     -          PANTHER:PTHR18834,MobiDB_lite:mobidb-lite          -         -                  -            -\n",
       "2486221  P732912                                 chr17_47731462_T/C    TBX21                True  HET                  SNV                              upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                             risk_factor,pathogenic  0.417200          SAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "451901   P732912                                 chr2_233758936_A/C   UGT1A4                True  HET                  SNV                                     intron_variant           -               True   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.761000          AFR                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "163796   P732912                                  chr5_83104943_C/-    XRCC4                True  HET             deletion                                 frameshift_variant         I/X              False   PATHO          Rare (<1%)     Rare (<5%)                       likely_pathogenic,pathogenic  0.003165  gnomADg_MID                  -                         -            NaN                NaN                     -     -  PDB-ENSP_mappings:1fu1.A,PDB-ENSP_mappings:1fu...          -         -                  -            -"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho12 = cad12[cad12['ClinVar'] == 'PATHO']\n",
    "print(patho12.shape)\n",
    "patho12[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(31835, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2430310</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr16_86184895_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6793</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1258708</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr7_67771556_T/C</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.7357</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1258714</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr7_67772928_A/C</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.7133</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2378574</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr16_30067171_G/C</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6180</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691521</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr4_26450075_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6931</td>\n",
       "      <td>AMR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160281</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_61494146_GG/-</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>deletion</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4555</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>904452</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_61539194_T/C</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>160287</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_61543228_-/T</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4484</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>904449</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr5_61535754_T/G</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4594</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2442051</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr17_4044350_T/C</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>I/V</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.8192</td>\n",
       "      <td>AFR</td>\n",
       "      <td>tolerated_low_confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>tolerated_low_confidence</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-O43149-F5.A,AFDB-ENSP_ma...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>31835 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND  Uploaded.variation  SYMBOL  gene.in.both.lines  ZYG VARIANT.CLASS            Consequence Amino.acids  var.in.both.lines ClinVar             MAF           MAF2 CLIN.SIG  MAX.AF  MAX.AF.POPS                         SIFT   PolyPhen            SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "2430310  P732912  chr16_86184895_G/A       -                True  HOM           SNV     intergenic_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6793          AFR                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1258708  P732912   chr7_67771556_T/C       -                True  HOM           SNV  upstream_gene_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.7357  gnomADg_SAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1258714  P732912   chr7_67772928_A/C       -                True  HOM           SNV  upstream_gene_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.7133  gnomADg_SAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2378574  P732912  chr16_30067171_G/C       -                True  HOM           SNV         intron_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6180          AFR                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "691521   P732912   chr4_26450075_G/A       -                True  HET           SNV     intergenic_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6931          AMR                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "...          ...                 ...     ...                 ...  ...           ...                    ...         ...                ...     ...             ...            ...      ...     ...          ...                          ...        ...                       ...                ...                   ...   ...                                                ...        ...       ...                ...          ...\n",
       "160281   P732912  chr5_61494146_GG/-  ZSWIM6                True  HET      deletion         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4555  gnomADg_AMI                            -          -                       NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "904452   P732912   chr5_61539194_T/C  ZSWIM6                True  HET           SNV         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4632          SAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "160287   P732912   chr5_61543228_-/T  ZSWIM6                True  HET     insertion         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4484  gnomADg_AMI                            -          -                       NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "904449   P732912   chr5_61535754_T/G  ZSWIM6                True  HET           SNV         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4594  gnomADg_AMI                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2442051  P732912   chr17_4044350_T/C   ZZEF1                True  HET           SNV       missense_variant         I/V               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.8192          AFR  tolerated_low_confidence(1)  benign(0)  tolerated_low_confidence             benign                     -     -  AFDB-ENSP_mappings:AF-O43149-F5.A,AFDB-ENSP_ma...          -         -                  -            -\n",
       "\n",
       "[31835 rows x 26 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# patho12 = cad12[cad12['ClinVar'] == 'PATHO']\n",
    "# print(patho12.shape)\n",
    "# patho12[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "benign12 = cad12[cad12['ClinVar'] == 'BENIGN']\n",
    "print(benign12.shape)\n",
    "benign12[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(309, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2571266</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr18_50872497_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.54180</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2571255</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr18_50866180_C/T</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.54070</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2571244</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr18_50861586_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.54060</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2364932</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr16_16075709_C/T</td>\n",
       "      <td>ABCC1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.76370</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1804022</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr11_17395957_A/G</td>\n",
       "      <td>ABCC8</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance,benign,likely_benign</td>\n",
       "      <td>0.78740</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89329</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr3_10152061_AAAA/-/AAA</td>\n",
       "      <td>VHL</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.33950</td>\n",
       "      <td>gnomADg_AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1935186</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr12_6063036_T/C</td>\n",
       "      <td>VWF</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>H/R</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>0.74030</td>\n",
       "      <td>SAS</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.026)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:7kwo.V,PDB-ENSP_mappings:7pm...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>542466</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chrX_123911478_-/A</td>\n",
       "      <td>XIAP</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance,benign</td>\n",
       "      <td>0.52440</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>524906</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr22_40929193_-/T</td>\n",
       "      <td>XPNPEP3</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.02912</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1581349</th>\n",
       "      <td>P732912</td>\n",
       "      <td>chr9_96819061_C/T</td>\n",
       "      <td>ZNF782</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>R/H</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.01899</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PANTHER:PTHR24376:SF54,PANTHER:PTHR24376,Gene3...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>309 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND        Uploaded.variation   SYMBOL  gene.in.both.lines  ZYG        VARIANT.CLASS                Consequence Amino.acids  var.in.both.lines ClinVar                 MAF           MAF2                                     CLIN.SIG   MAX.AF  MAX.AF.POPS          SIFT       PolyPhen SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "2571266  P732912        chr18_50872497_G/A        -                True  HET                  SNV         intergenic_variant           -               True     VUS      Common (>=10%)  Common (>=5%)                       uncertain_significance  0.54180  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2571255  P732912        chr18_50866180_C/T        -                True  HET                  SNV  regulatory_region_variant           -               True     VUS      Common (>=10%)  Common (>=5%)                       uncertain_significance  0.54070  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2571244  P732912        chr18_50861586_G/A        -                True  HET                  SNV  regulatory_region_variant           -               True     VUS      Common (>=10%)  Common (>=5%)                       uncertain_significance  0.54060  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2364932  P732912        chr16_16075709_C/T    ABCC1                True  HET                  SNV             intron_variant           -               True     VUS      Common (>=10%)  Common (>=5%)                       uncertain_significance  0.76370  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1804022  P732912        chr11_17395957_A/G    ABCC8                True  HET                  SNV             intron_variant           -               True     VUS      Common (>=10%)  Common (>=5%)  uncertain_significance,benign,likely_benign  0.78740          AFR             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "...          ...                       ...      ...                 ...  ...                  ...                        ...         ...                ...     ...                 ...            ...                                          ...      ...          ...           ...            ...            ...                ...                   ...   ...                                                ...        ...       ...                ...          ...\n",
       "89329    P732912  chr3_10152061_AAAA/-/AAA      VHL                True  HET  sequence_alteration        3_prime_UTR_variant           -              False     VUS      Common (>=10%)  Common (>=5%)                       uncertain_significance  0.33950  gnomADg_AFR             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "1935186  P732912         chr12_6063036_T/C      VWF                True  HOM                  SNV           missense_variant         H/R               True     VUS      Common (>=10%)  Common (>=5%)  benign,uncertain_significance,likely_benign  0.74030          SAS  tolerated(1)  benign(0.026)      tolerated             benign                     -     -  PDB-ENSP_mappings:7kwo.V,PDB-ENSP_mappings:7pm...          -         -                  -            -\n",
       "542466   P732912        chrX_123911478_-/A     XIAP                True  HET            insertion        3_prime_UTR_variant           -               True     VUS      Common (>=10%)  Common (>=5%)                uncertain_significance,benign  0.52440          AFR             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "524906   P732912        chr22_40929193_-/T  XPNPEP3                True  HET            insertion        3_prime_UTR_variant           -              False     VUS  Less common (<10%)     Rare (<5%)                       uncertain_significance  0.02912  gnomADg_ASJ             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "1581349  P732912         chr9_96819061_C/T   ZNF782                True  HET                  SNV           missense_variant         R/H              False     VUS  Less common (<10%)     Rare (<5%)                       uncertain_significance  0.01899  gnomADg_MID  tolerated(1)      benign(0)      tolerated             benign                     -     -  PANTHER:PTHR24376:SF54,PANTHER:PTHR24376,Gene3...          -         -                  -            -\n",
       "\n",
       "[309 rows x 26 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# patho12 = cad12[cad12['ClinVar'] == 'PATHO']\n",
    "# print(patho12.shape)\n",
    "# patho12[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "# benign12 = cad12[cad12['ClinVar'] == 'BENIGN']\n",
    "# print(benign12.shape)\n",
    "# benign12[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "vus12 = cad12[cad12['ClinVar'] == 'VUS']\n",
    "print(vus12.shape)\n",
    "vus12[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33116\n",
       "2     VUS      346\n",
       "3   PATHO       17"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "e0fc9763-5568-4919-886b-9b1d098ee844",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(17, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>23723</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr1_18872954_G/T</td>\n",
       "      <td>ALDH4A1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T/N</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.042630</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.97)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:3v9g.A,PDB-ENSP_mappings:3v9...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284152</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr9_27573524_GCCCCG/-</td>\n",
       "      <td>C9orf72</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>deletion</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.657700</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2408364</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr14_91297376_G/A</td>\n",
       "      <td>CCDC88C</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>R/C</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.007261</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.973)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PANTHER:PTHR18947,PANTHER:PTHR18947:SF31,Coile...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2653430</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr17_34252769_A/G</td>\n",
       "      <td>CCL2</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>risk_factor,pathogenic</td>\n",
       "      <td>0.555700</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543951</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr3_39265671_G/A</td>\n",
       "      <td>CX3CR1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>T/M</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>risk_factor,pathogenic,protective</td>\n",
       "      <td>0.186600</td>\n",
       "      <td>gnomADe_ASJ</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>probably_damaging(0.978)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:1.20.1070.10,PDB-ENSP_mappings:7xbw.R,P...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>543952</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr3_39265765_C/T</td>\n",
       "      <td>CX3CR1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>V/I</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic,risk_factor,protective</td>\n",
       "      <td>0.285300</td>\n",
       "      <td>EUR,gnomADg_MID</td>\n",
       "      <td>tolerated(0.61)</td>\n",
       "      <td>benign(0.012)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:1.20.1070.10,PDB-ENSP_mappings:7xbw.R,P...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>484803</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr2_232486499_A/G</td>\n",
       "      <td>ECEL1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>L/P</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.007882</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>tolerated(0.05)</td>\n",
       "      <td>possibly_damaging(0.472)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PANTHER:PTHR11733:SF195,PANTHER:PTHR11733,AFDB...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>315308</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr2_62904596_G/A</td>\n",
       "      <td>EHBP1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.185900</td>\n",
       "      <td>AMR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1142271</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr6_26090951_C/G</td>\n",
       "      <td>HFE</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>H/D</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>risk_factor,not_provided,conflicting_interpret...</td>\n",
       "      <td>0.172000</td>\n",
       "      <td>EUR</td>\n",
       "      <td>tolerated(0.09)</td>\n",
       "      <td>possibly_damaging(0.704)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>possibly_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1a6z.A,PDB-ENSP_mappings:1a6...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2550959</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr16_27344882_A/G</td>\n",
       "      <td>IL4R</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>I/V</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>protective,pathogenic</td>\n",
       "      <td>0.524900</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.001)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1iar.B,Gene3D:2.60.40.10,PDB...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1425195</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr7_128938247_T/G</td>\n",
       "      <td>IRF5</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic,risk_factor</td>\n",
       "      <td>0.749000</td>\n",
       "      <td>EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>873217</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr4_150265713_C/A</td>\n",
       "      <td>LRBA</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.007000</td>\n",
       "      <td>EUR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2808393</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr19_15189109_C/A</td>\n",
       "      <td>NOTCH3</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>G/C</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>pathogenic,likely_pathogenic</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>tolerated(0.1)</td>\n",
       "      <td>probably_damaging(0.985)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Gene3D:2.10.25.10,AFDB-ENSP_mappings:AF-Q9UM47...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2578011</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr16_69711242_G/A</td>\n",
       "      <td>NQO1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>P/S</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>risk_factor,not_provided,pathogenic</td>\n",
       "      <td>0.455300</td>\n",
       "      <td>gnomADe_EAS</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>benign(0.173)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1d4a.A,PDB-ENSP_mappings:1d4...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2637021</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr17_17796637_C/A</td>\n",
       "      <td>RAI1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>A/E</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.000088</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>benign(0.162)</td>\n",
       "      <td>deleterious</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-Q7Z5J4-F1.A,PANTHER:PTHR...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>486447</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr2_233758936_A/C</td>\n",
       "      <td>UGT1A4</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>0.761000</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2393610</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr14_77453324_G/C</td>\n",
       "      <td>VIPAS39</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>V</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>conflicting_interpretations_of_pathogenicity</td>\n",
       "      <td>0.000422</td>\n",
       "      <td>gnomADe_NFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PANTHER:PTHR13364,AFDB-ENSP_mappings:AF-Q9H9C1...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND      Uploaded.variation   SYMBOL  gene.in.both.lines  ZYG VARIANT.CLASS            Consequence Amino.acids  var.in.both.lines ClinVar                 MAF           MAF2                                           CLIN.SIG    MAX.AF      MAX.AF.POPS               SIFT                  PolyPhen SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "23723    P931263       chr1_18872954_G/T  ALDH4A1                True  HET           SNV       missense_variant         T/N              False   PATHO  Less common (<10%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.042630      gnomADg_FIN     deleterious(0)   probably_damaging(0.97)    deleterious  probably_damaging                     -     -  PDB-ENSP_mappings:3v9g.A,PDB-ENSP_mappings:3v9...          -         -                  -            -\n",
       "284152   P931263  chr9_27573524_GCCCCG/-  C9orf72                True  HOM      deletion  upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.657700      gnomADg_ASJ                  -                         -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "2408364  P931263      chr14_91297376_G/A  CCDC88C                True  HET           SNV       missense_variant         R/C              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.007261      gnomADg_NFE     deleterious(0)  probably_damaging(0.973)    deleterious  probably_damaging                     -     -  PANTHER:PTHR18947,PANTHER:PTHR18947:SF31,Coile...          -         -                  -            -\n",
       "2653430  P931263      chr17_34252769_A/G     CCL2                True  HET           SNV  upstream_gene_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                             risk_factor,pathogenic  0.555700      gnomADg_EAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "543951   P931263       chr3_39265671_G/A   CX3CR1                True  HET           SNV       missense_variant         T/M              False   PATHO      Common (>=10%)  Common (>=5%)                  risk_factor,pathogenic,protective  0.186600      gnomADe_ASJ  deleterious(0.01)  probably_damaging(0.978)    deleterious  probably_damaging                     -     -  Gene3D:1.20.1070.10,PDB-ENSP_mappings:7xbw.R,P...          -         -                  -            -\n",
       "543952   P931263       chr3_39265765_C/T   CX3CR1                True  HET           SNV       missense_variant         V/I              False   PATHO      Common (>=10%)  Common (>=5%)                  pathogenic,risk_factor,protective  0.285300  EUR,gnomADg_MID    tolerated(0.61)             benign(0.012)      tolerated             benign                     -     -  Gene3D:1.20.1070.10,PDB-ENSP_mappings:7xbw.R,P...          -         -                  -            -\n",
       "484803   P931263      chr2_232486499_A/G    ECEL1                True  HET           SNV       missense_variant         L/P              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.007882      gnomADg_FIN    tolerated(0.05)  possibly_damaging(0.472)      tolerated  possibly_damaging                     -     -  PANTHER:PTHR11733:SF195,PANTHER:PTHR11733,AFDB...          -         -                  -            -\n",
       "315308   P931263       chr2_62904596_G/A    EHBP1                True  HET           SNV         intron_variant           -              False   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.185900              AMR                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1142271  P931263       chr6_26090951_C/G      HFE                True  HET           SNV       missense_variant         H/D              False   PATHO      Common (>=10%)  Common (>=5%)  risk_factor,not_provided,conflicting_interpret...  0.172000              EUR    tolerated(0.09)  possibly_damaging(0.704)      tolerated  possibly_damaging                     -     -  PDB-ENSP_mappings:1a6z.A,PDB-ENSP_mappings:1a6...          -         -                  -            -\n",
       "2550959  P931263      chr16_27344882_A/G     IL4R                True  HET           SNV       missense_variant         I/V              False   PATHO      Common (>=10%)  Common (>=5%)                              protective,pathogenic  0.524900      gnomADg_EAS       tolerated(1)             benign(0.001)      tolerated             benign                     -     -  PDB-ENSP_mappings:1iar.B,Gene3D:2.60.40.10,PDB...          -         -                  -            -\n",
       "1425195  P931263      chr7_128938247_T/G     IRF5                True  HOM           SNV         intron_variant           -               True   PATHO      Common (>=10%)  Common (>=5%)                             pathogenic,risk_factor  0.749000              EAS                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "873217   P931263      chr4_150265713_C/A     LRBA                True  HET           SNV    3_prime_UTR_variant           -              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.007000              EUR                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2808393  P931263      chr19_15189109_C/A   NOTCH3                True  HET           SNV       missense_variant         G/C              False   PATHO          Rare (<1%)     Rare (<5%)                       pathogenic,likely_pathogenic  0.000015      gnomADg_NFE     tolerated(0.1)  probably_damaging(0.985)      tolerated  probably_damaging                     -     -  Gene3D:2.10.25.10,AFDB-ENSP_mappings:AF-Q9UM47...          -         -                  -            -\n",
       "2578011  P931263      chr16_69711242_G/A     NQO1                True  HET           SNV       missense_variant         P/S              False   PATHO      Common (>=10%)  Common (>=5%)                risk_factor,not_provided,pathogenic  0.455300      gnomADe_EAS     deleterious(0)             benign(0.173)    deleterious             benign                     -     -  PDB-ENSP_mappings:1d4a.A,PDB-ENSP_mappings:1d4...          -         -                  -            -\n",
       "2637021  P931263      chr17_17796637_C/A     RAI1                True  HET           SNV       missense_variant         A/E              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.000088      gnomADg_NFE  deleterious(0.02)             benign(0.162)    deleterious             benign                     -     -  AFDB-ENSP_mappings:AF-Q7Z5J4-F1.A,PANTHER:PTHR...          -         -                  -            -\n",
       "486447   P931263      chr2_233758936_A/C   UGT1A4                True  HET           SNV         intron_variant           -               True   PATHO      Common (>=10%)  Common (>=5%)                                         pathogenic  0.761000              AFR                  -                         -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2393610  P931263      chr14_77453324_G/C  VIPAS39                True  HET           SNV     synonymous_variant           V              False   PATHO          Rare (<1%)     Rare (<5%)       conflicting_interpretations_of_pathogenicity  0.000422      gnomADe_NFE                  -                         -              -                  -                     -     -  PANTHER:PTHR13364,AFDB-ENSP_mappings:AF-Q9H9C1...          -         -                  -            -"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "patho63 = cad63[cad63['ClinVar'] == 'PATHO']\n",
    "print(patho63.shape)\n",
    "patho63[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(33116, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>257253</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr2_11741072_A/C</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6846</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1678588</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr9_69099777_C/T</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant,NMD_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6136</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1678587</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr9_69099647_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant,NMD_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.6136</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2222330</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr13_20366042_T/C</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.7751</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2222331</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr13_20366194_C/G</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.7749</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989614</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr5_61531912_T/G</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4593</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989615</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr5_61535306_C/T</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.2719</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989616</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr5_61535754_T/G</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4594</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>989618</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr5_61539194_T/C</td>\n",
       "      <td>ZSWIM6</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.4632</td>\n",
       "      <td>SAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2618574</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr17_4044350_T/C</td>\n",
       "      <td>ZZEF1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>I/V</td>\n",
       "      <td>True</td>\n",
       "      <td>BENIGN</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign</td>\n",
       "      <td>0.8192</td>\n",
       "      <td>AFR</td>\n",
       "      <td>tolerated_low_confidence(1)</td>\n",
       "      <td>benign(0)</td>\n",
       "      <td>tolerated_low_confidence</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-O43149-F5.A,AFDB-ENSP_ma...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>33116 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND  Uploaded.variation  SYMBOL  gene.in.both.lines  ZYG VARIANT.CLASS                            Consequence Amino.acids  var.in.both.lines ClinVar             MAF           MAF2 CLIN.SIG  MAX.AF  MAX.AF.POPS                         SIFT   PolyPhen            SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "257253   P931263   chr2_11741072_A/C       -                True  HET           SNV     non_coding_transcript_exon_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6846          AFR                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1678588  P931263   chr9_69099777_C/T       -                True  HET           SNV  intron_variant,NMD_transcript_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6136  gnomADg_EAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1678587  P931263   chr9_69099647_G/A       -                True  HET           SNV  intron_variant,NMD_transcript_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.6136  gnomADg_EAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2222330  P931263  chr13_20366042_T/C       -                True  HOM           SNV              regulatory_region_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.7751  gnomADg_FIN                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2222331  P931263  chr13_20366194_C/G       -                True  HOM           SNV              regulatory_region_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.7749  gnomADg_FIN                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "...          ...                 ...     ...                 ...  ...           ...                                    ...         ...                ...     ...             ...            ...      ...     ...          ...                          ...        ...                       ...                ...                   ...   ...                                                ...        ...       ...                ...          ...\n",
       "989614   P931263   chr5_61531912_T/G  ZSWIM6                True  HET           SNV                         intron_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4593  gnomADg_AMI                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "989615   P931263   chr5_61535306_C/T  ZSWIM6                True  HET           SNV                         intron_variant           -              False  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.2719  gnomADg_AMI                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "989616   P931263   chr5_61535754_T/G  ZSWIM6                True  HET           SNV                         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4594  gnomADg_AMI                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "989618   P931263   chr5_61539194_T/C  ZSWIM6                True  HET           SNV                         intron_variant           -               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.4632          SAS                            -          -                         -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2618574  P931263   chr17_4044350_T/C   ZZEF1                True  HET           SNV                       missense_variant         I/V               True  BENIGN  Common (>=10%)  Common (>=5%)   benign  0.8192          AFR  tolerated_low_confidence(1)  benign(0)  tolerated_low_confidence             benign                     -     -  AFDB-ENSP_mappings:AF-O43149-F5.A,AFDB-ENSP_ma...          -         -                  -            -\n",
       "\n",
       "[33116 rows x 26 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# patho63 = cad63[cad63['ClinVar'] == 'PATHO']\n",
    "# print(patho63.shape)\n",
    "# patho63[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "benign63 = cad63[cad63['ClinVar'] == 'BENIGN']\n",
    "print(benign63.shape)\n",
    "benign63[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(346, 95)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>IND</th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2752162</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr18_50866180_C/T</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.5407</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2752175</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr18_50872497_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intergenic_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.5418</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2752151</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr18_50861586_G/A</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.5406</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1259993</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr6_151630085_A/G</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.3125</td>\n",
       "      <td>EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2541083</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr16_16075709_C/T</td>\n",
       "      <td>ABCC1</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.7637</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446768</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr16_46659710_-/A</td>\n",
       "      <td>VPS35</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance,likely_benign</td>\n",
       "      <td>0.3242</td>\n",
       "      <td>AMR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2079362</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr12_5985618_T/C</td>\n",
       "      <td>VWF</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>SNV</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>T</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>0.1969</td>\n",
       "      <td>gnomADe_AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Pfam:PF00093,PIRSF:PIRSF002495,SMART:SM00214,P...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2079587</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr12_6063036_T/C</td>\n",
       "      <td>VWF</td>\n",
       "      <td>True</td>\n",
       "      <td>HOM</td>\n",
       "      <td>SNV</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>H/R</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>benign,uncertain_significance,likely_benign</td>\n",
       "      <td>0.7403</td>\n",
       "      <td>SAS</td>\n",
       "      <td>tolerated(1)</td>\n",
       "      <td>benign(0.026)</td>\n",
       "      <td>tolerated</td>\n",
       "      <td>benign</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:7kwo.V,PDB-ENSP_mappings:7pm...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>558471</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chrX_123911478_-/A</td>\n",
       "      <td>XIAP</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>True</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance,benign</td>\n",
       "      <td>0.5244</td>\n",
       "      <td>AFR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>366379</th>\n",
       "      <td>P931263</td>\n",
       "      <td>chr12_32744714_-/A</td>\n",
       "      <td>YARS2</td>\n",
       "      <td>True</td>\n",
       "      <td>HET</td>\n",
       "      <td>insertion</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>False</td>\n",
       "      <td>VUS</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>uncertain_significance</td>\n",
       "      <td>0.3474</td>\n",
       "      <td>gnomADe_AMR</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>346 rows × 26 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             IND  Uploaded.variation SYMBOL  gene.in.both.lines  ZYG VARIANT.CLASS                Consequence Amino.acids  var.in.both.lines ClinVar             MAF           MAF2                                     CLIN.SIG  MAX.AF  MAX.AF.POPS          SIFT       PolyPhen SIFTprediction PolyPhenprediction TRANSCRIPTION.FACTORS miRNA                                            DOMAINS MOTIF.NAME MOTIF.POS MOTIF.SCORE.CHANGE HIGH.INF.POS\n",
       "2752162  P931263  chr18_50866180_C/T      -                True  HET           SNV  regulatory_region_variant           -               True     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.5407  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2752175  P931263  chr18_50872497_G/A      -                True  HET           SNV         intergenic_variant           -               True     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.5418  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2752151  P931263  chr18_50861586_G/A      -                True  HET           SNV  regulatory_region_variant           -               True     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.5406  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "1259993  P931263  chr6_151630085_A/G      -                True  HET           SNV  regulatory_region_variant           -              False     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.3125          EAS             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "2541083  P931263  chr16_16075709_C/T  ABCC1                True  HET           SNV             intron_variant           -               True     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.7637  gnomADg_AMI             -              -              -                  -                     -     -                                                  -          -         -                  -            -\n",
       "...          ...                 ...    ...                 ...  ...           ...                        ...         ...                ...     ...             ...            ...                                          ...     ...          ...           ...            ...            ...                ...                   ...   ...                                                ...        ...       ...                ...          ...\n",
       "446768   P931263  chr16_46659710_-/A  VPS35                True  HET     insertion        3_prime_UTR_variant           -              False     VUS  Common (>=10%)  Common (>=5%)         uncertain_significance,likely_benign  0.3242          AMR             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "2079362  P931263   chr12_5985618_T/C    VWF                True  HET           SNV         synonymous_variant           T              False     VUS  Common (>=10%)  Common (>=5%)  benign,uncertain_significance,likely_benign  0.1969  gnomADe_AFR             -              -              -                  -                     -     -  Pfam:PF00093,PIRSF:PIRSF002495,SMART:SM00214,P...          -         -                  -            -\n",
       "2079587  P931263   chr12_6063036_T/C    VWF                True  HOM           SNV           missense_variant         H/R               True     VUS  Common (>=10%)  Common (>=5%)  benign,uncertain_significance,likely_benign  0.7403          SAS  tolerated(1)  benign(0.026)      tolerated             benign                     -     -  PDB-ENSP_mappings:7kwo.V,PDB-ENSP_mappings:7pm...          -         -                  -            -\n",
       "558471   P931263  chrX_123911478_-/A   XIAP                True  HET     insertion        3_prime_UTR_variant           -               True     VUS  Common (>=10%)  Common (>=5%)                uncertain_significance,benign  0.5244          AFR             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "366379   P931263  chr12_32744714_-/A  YARS2                True  HET     insertion    downstream_gene_variant           -              False     VUS  Common (>=10%)  Common (>=5%)                       uncertain_significance  0.3474  gnomADe_AMR             -              -            NaN                NaN                     -     -                                                  -          -         -                  -            -\n",
       "\n",
       "[346 rows x 26 columns]"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# patho63 = cad63[cad63['ClinVar'] == 'PATHO']\n",
    "# print(patho63.shape)\n",
    "# patho63[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "# benign63 = cad63[cad63['ClinVar'] == 'BENIGN']\n",
    "# print(benign63.shape)\n",
    "# benign63[keepme].sort_values(['SYMBOL'])\n",
    "\n",
    "vus63 = cad63[cad63['ClinVar'] == 'VUS']\n",
    "print(vus63.shape)\n",
    "vus63[keepme].sort_values(['SYMBOL'])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "# [4] label Database - indicates if variant is found in ClinVar and Gnomad, or only ClinVar, or only Gnomad\n",
    " \n",
    "---\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "4e6a8e9b-7f39-4917-9c85-e990436f6cf2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# create a new column 'Database' in both dataframes, if MAX.AF is not NA and ClinVar is NA label 'gnomAD', if ClinVar is not NA and MAX.AF is NA label 'ClinVar', if both are not NA label 'gnomAD & ClinVar', else label 'No DB source'\n",
    "cad12['Database'] = np.select([cad12['ClinVar'].notna() & cad12['MAX.AF'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].notna(), \n",
    "                                  cad12['MAX.AF'].isna() & cad12['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "\n",
    "cad63['Database'] = np.select([cad63['ClinVar'].notna() & cad63['MAX.AF'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].notna(), \n",
    "                                  cad63['MAX.AF'].isna() & cad63['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3428264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>35778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>32158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ClinVar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3428264\n",
       "1      No DB source    35778\n",
       "2  gnomAD & ClinVar    32158\n",
       "3           ClinVar        1"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'Database')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "c6c4c702-3708-4851-844a-4734c184cfe6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3641404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>33479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3641404\n",
       "1  gnomAD & ClinVar    33479\n",
       "2      No DB source    26131"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "checkColumnValue(cad63, 'Database')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "6d981587-2c8f-4260-9d79-d74c58e149d2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Common (&gt;=10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3221872</td>\n",
       "      <td>31160</td>\n",
       "      <td>3253032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Less common (&lt;10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>152004</td>\n",
       "      <td>856</td>\n",
       "      <td>152860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rare (&lt;1%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>54388</td>\n",
       "      <td>142</td>\n",
       "      <td>54530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nan</th>\n",
       "      <td>1</td>\n",
       "      <td>35778</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>1</td>\n",
       "      <td>35778</td>\n",
       "      <td>3428264</td>\n",
       "      <td>32158</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database            ClinVar  No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "MAF                                                                          \n",
       "Common (>=10%)            0             0  3221872             31160  3253032\n",
       "Less common (<10%)        0             0   152004               856   152860\n",
       "Rare (<1%)                0             0    54388               142    54530\n",
       "nan                       1         35778        0                 0    35779\n",
       "Total                     1         35778  3428264             32158  3496201"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad12, 'MAF', 'Database', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Common (&gt;=10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>3430130</td>\n",
       "      <td>32339</td>\n",
       "      <td>3462469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Less common (&lt;10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>156632</td>\n",
       "      <td>992</td>\n",
       "      <td>157624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rare (&lt;1%)</th>\n",
       "      <td>0</td>\n",
       "      <td>54642</td>\n",
       "      <td>148</td>\n",
       "      <td>54790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nan</th>\n",
       "      <td>26131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>26131</td>\n",
       "      <td>3641404</td>\n",
       "      <td>33479</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database            No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "MAF                                                                 \n",
       "Common (>=10%)                 0  3430130             32339  3462469\n",
       "Less common (<10%)             0   156632               992   157624\n",
       "Rare (<1%)                     0    54642               148    54790\n",
       "nan                        26131        0                 0    26131\n",
       "Total                      26131  3641404             33479  3701014"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad63, 'MAF', 'Database', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinVar</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BENIGN</th>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>31834</td>\n",
       "      <td>31835</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA</th>\n",
       "      <td>0</td>\n",
       "      <td>35778</td>\n",
       "      <td>3428264</td>\n",
       "      <td>0</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PATHO</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VUS</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>309</td>\n",
       "      <td>309</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>1</td>\n",
       "      <td>35778</td>\n",
       "      <td>3428264</td>\n",
       "      <td>32158</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database  ClinVar  No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "ClinVar                                                            \n",
       "BENIGN          1             0        0             31834    31835\n",
       "NA              0         35778  3428264                 0  3464042\n",
       "PATHO           0             0        0                15       15\n",
       "VUS             0             0        0               309      309\n",
       "Total           1         35778  3428264             32158  3496201"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad12, 'ClinVar', 'Database', True)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ClinVar</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BENIGN</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>33116</td>\n",
       "      <td>33116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>NA</th>\n",
       "      <td>26131</td>\n",
       "      <td>3641404</td>\n",
       "      <td>0</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>PATHO</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>VUS</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>346</td>\n",
       "      <td>346</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>26131</td>\n",
       "      <td>3641404</td>\n",
       "      <td>33479</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database  No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "ClinVar                                                   \n",
       "BENIGN               0        0             33116    33116\n",
       "NA               26131  3641404                 0  3667535\n",
       "PATHO                0        0                17       17\n",
       "VUS                  0        0               346      346\n",
       "Total            26131  3641404             33479  3701014"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad63, 'ClinVar', 'Database', True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "# SAVE individual dfs\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 96)\n",
      "(3701014, 96)\n",
      "Index(['Uploaded.variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature.type', 'Consequence', 'cDNA.position', 'CDS.position', 'Protein.position', 'Amino.acids', 'Codons', 'Existing.variation', 'IND', 'ZYG', 'REF.ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT.CLASS', 'SYMBOL', 'SYMBOL.SOURCE', 'HGNC.ID', 'BIOTYPE', 'CANONICAL', 'MANE.SELECT', 'MANE.PLUS.CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT.ISOFORM', 'REFSEQ.MATCH', 'SOURCE', 'REFSEQ.OFFSET', 'GIVEN.REF', 'USED.REF', 'BAM.EDIT', 'GENE.PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS.OFFSET', 'gnomADg.AF', 'gnomADg.AFR.AF', 'gnomADg.AMI.AF', 'gnomADg.AMR.AF', 'gnomADg.ASJ.AF', 'gnomADg.EAS.AF', 'gnomADg.FIN.AF', 'gnomADg.MID.AF', 'gnomADg.NFE.AF', 'gnomADg.OTH.AF', 'gnomADg.SAS.AF', 'MAX.AF', 'MAX.AF.POPS', 'CLIN.SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF.NAME', 'MOTIF.POS', 'HIGH.INF.POS', 'MOTIF.SCORE.CHANGE',\n",
      "       'TRANSCRIPTION.FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN.REF.MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var.in.both.lines', 'var.exclusive.btw.lines', 'gene.in.both.lines', 'gene.exclusive.btw.lines', 'MAF', 'MAF2', 'PATHO', 'VUS', 'BENIGN', 'ClinVar', 'Database'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(cad12.shape)\n",
    "print(cad63.shape)\n",
    "print(cad12.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>436902.000000</td>\n",
       "      <td>3.460261e+06</td>\n",
       "      <td>3.460259e+06</td>\n",
       "      <td>3.460165e+06</td>\n",
       "      <td>3.460258e+06</td>\n",
       "      <td>3.460242e+06</td>\n",
       "      <td>3.460250e+06</td>\n",
       "      <td>3.460254e+06</td>\n",
       "      <td>3.460100e+06</td>\n",
       "      <td>3.460261e+06</td>\n",
       "      <td>3.460238e+06</td>\n",
       "      <td>3.460252e+06</td>\n",
       "      <td>3.460422e+06</td>\n",
       "      <td>3.496201e+06</td>\n",
       "      <td>85969.000000</td>\n",
       "      <td>1.887824e+06</td>\n",
       "      <td>8742.000000</td>\n",
       "      <td>8444.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>2488.211519</td>\n",
       "      <td>4.589795e-01</td>\n",
       "      <td>4.377915e-01</td>\n",
       "      <td>4.657630e-01</td>\n",
       "      <td>4.650005e-01</td>\n",
       "      <td>4.667297e-01</td>\n",
       "      <td>4.721054e-01</td>\n",
       "      <td>4.663075e-01</td>\n",
       "      <td>4.652989e-01</td>\n",
       "      <td>4.663965e-01</td>\n",
       "      <td>4.609162e-01</td>\n",
       "      <td>4.762419e-01</td>\n",
       "      <td>6.084945e-01</td>\n",
       "      <td>7.623309e+07</td>\n",
       "      <td>6.972944</td>\n",
       "      <td>5.257583e+00</td>\n",
       "      <td>0.460385</td>\n",
       "      <td>0.152510</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1451.121397</td>\n",
       "      <td>2.896428e-01</td>\n",
       "      <td>3.017765e-01</td>\n",
       "      <td>3.143898e-01</td>\n",
       "      <td>2.971199e-01</td>\n",
       "      <td>3.031336e-01</td>\n",
       "      <td>3.280300e-01</td>\n",
       "      <td>3.066453e-01</td>\n",
       "      <td>2.999533e-01</td>\n",
       "      <td>3.037308e-01</td>\n",
       "      <td>2.917207e-01</td>\n",
       "      <td>2.960810e-01</td>\n",
       "      <td>2.951332e-01</td>\n",
       "      <td>5.659873e+07</td>\n",
       "      <td>9.018434</td>\n",
       "      <td>7.827981e+00</td>\n",
       "      <td>0.400754</td>\n",
       "      <td>0.301893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.018500e+04</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1226.000000</td>\n",
       "      <td>2.150000e-01</td>\n",
       "      <td>1.696000e-01</td>\n",
       "      <td>1.930000e-01</td>\n",
       "      <td>2.125000e-01</td>\n",
       "      <td>2.075000e-01</td>\n",
       "      <td>1.786000e-01</td>\n",
       "      <td>2.022000e-01</td>\n",
       "      <td>2.092000e-01</td>\n",
       "      <td>2.064000e-01</td>\n",
       "      <td>2.148000e-01</td>\n",
       "      <td>2.275000e-01</td>\n",
       "      <td>3.797000e-01</td>\n",
       "      <td>2.953120e+07</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.070000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2483.000000</td>\n",
       "      <td>4.385000e-01</td>\n",
       "      <td>4.056000e-01</td>\n",
       "      <td>4.458000e-01</td>\n",
       "      <td>4.459000e-01</td>\n",
       "      <td>4.472000e-01</td>\n",
       "      <td>4.533000e-01</td>\n",
       "      <td>4.461000e-01</td>\n",
       "      <td>4.459000e-01</td>\n",
       "      <td>4.464000e-01</td>\n",
       "      <td>4.407000e-01</td>\n",
       "      <td>4.572000e-01</td>\n",
       "      <td>6.359000e-01</td>\n",
       "      <td>6.551204e+07</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>2.000000e+00</td>\n",
       "      <td>0.340000</td>\n",
       "      <td>0.001000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>3746.000000</td>\n",
       "      <td>6.884000e-01</td>\n",
       "      <td>6.928000e-01</td>\n",
       "      <td>7.248000e-01</td>\n",
       "      <td>7.022000e-01</td>\n",
       "      <td>7.100000e-01</td>\n",
       "      <td>7.547000e-01</td>\n",
       "      <td>7.154000e-01</td>\n",
       "      <td>7.057000e-01</td>\n",
       "      <td>7.100000e-01</td>\n",
       "      <td>6.921000e-01</td>\n",
       "      <td>7.127000e-01</td>\n",
       "      <td>8.767000e-01</td>\n",
       "      <td>1.131298e+08</td>\n",
       "      <td>9.000000</td>\n",
       "      <td>6.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.084000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>5000.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>2.489456e+08</td>\n",
       "      <td>358.000000</td>\n",
       "      <td>3.470000e+02</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            DISTANCE    gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF        MAX.AF  chrCoordinate    EXONnumber  INTRONnumber    SIFTscore  PolyPhenscore\n",
       "count  436902.000000  3.460261e+06    3.460259e+06    3.460165e+06    3.460258e+06    3.460242e+06    3.460250e+06    3.460254e+06    3.460100e+06    3.460261e+06    3.460238e+06    3.460252e+06  3.460422e+06   3.496201e+06  85969.000000  1.887824e+06  8742.000000    8444.000000\n",
       "mean     2488.211519  4.589795e-01    4.377915e-01    4.657630e-01    4.650005e-01    4.667297e-01    4.721054e-01    4.663075e-01    4.652989e-01    4.663965e-01    4.609162e-01    4.762419e-01  6.084945e-01   7.623309e+07      6.972944  5.257583e+00     0.460385       0.152510\n",
       "std      1451.121397  2.896428e-01    3.017765e-01    3.143898e-01    2.971199e-01    3.031336e-01    3.280300e-01    3.066453e-01    2.999533e-01    3.037308e-01    2.917207e-01    2.960810e-01  2.951332e-01   5.659873e+07      9.018434  7.827981e+00     0.400754       0.301893\n",
       "min         0.000000  0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00  0.000000e+00   1.018500e+04      1.000000  1.000000e+00     0.000000       0.000000\n",
       "25%      1226.000000  2.150000e-01    1.696000e-01    1.930000e-01    2.125000e-01    2.075000e-01    1.786000e-01    2.022000e-01    2.092000e-01    2.064000e-01    2.148000e-01    2.275000e-01  3.797000e-01   2.953120e+07      2.000000  1.000000e+00     0.070000       0.000000\n",
       "50%      2483.000000  4.385000e-01    4.056000e-01    4.458000e-01    4.459000e-01    4.472000e-01    4.533000e-01    4.461000e-01    4.459000e-01    4.464000e-01    4.407000e-01    4.572000e-01  6.359000e-01   6.551204e+07      4.000000  2.000000e+00     0.340000       0.001000\n",
       "75%      3746.000000  6.884000e-01    6.928000e-01    7.248000e-01    7.022000e-01    7.100000e-01    7.547000e-01    7.154000e-01    7.057000e-01    7.100000e-01    6.921000e-01    7.127000e-01  8.767000e-01   1.131298e+08      9.000000  6.000000e+00     1.000000       0.084000\n",
       "max      5000.000000  1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00  1.000000e+00   2.489456e+08    358.000000  3.470000e+02     1.000000       1.000000"
      ]
     },
     "execution_count": 47,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>444320.000000</td>\n",
       "      <td>3.674811e+06</td>\n",
       "      <td>3.674809e+06</td>\n",
       "      <td>3.674742e+06</td>\n",
       "      <td>3.674810e+06</td>\n",
       "      <td>3.674806e+06</td>\n",
       "      <td>3.674802e+06</td>\n",
       "      <td>3.674804e+06</td>\n",
       "      <td>3.674681e+06</td>\n",
       "      <td>3.674811e+06</td>\n",
       "      <td>3.674795e+06</td>\n",
       "      <td>3.674798e+06</td>\n",
       "      <td>3.674883e+06</td>\n",
       "      <td>3.701014e+06</td>\n",
       "      <td>87596.000000</td>\n",
       "      <td>2.002513e+06</td>\n",
       "      <td>8629.000000</td>\n",
       "      <td>8311.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>2488.533665</td>\n",
       "      <td>4.718748e-01</td>\n",
       "      <td>4.426008e-01</td>\n",
       "      <td>4.853570e-01</td>\n",
       "      <td>4.760430e-01</td>\n",
       "      <td>4.827356e-01</td>\n",
       "      <td>4.682483e-01</td>\n",
       "      <td>4.841243e-01</td>\n",
       "      <td>4.789125e-01</td>\n",
       "      <td>4.857458e-01</td>\n",
       "      <td>4.744601e-01</td>\n",
       "      <td>4.761085e-01</td>\n",
       "      <td>6.144999e-01</td>\n",
       "      <td>7.697594e+07</td>\n",
       "      <td>7.139847</td>\n",
       "      <td>5.279100e+00</td>\n",
       "      <td>0.462458</td>\n",
       "      <td>0.148457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>1450.436258</td>\n",
       "      <td>2.861300e-01</td>\n",
       "      <td>3.019205e-01</td>\n",
       "      <td>3.081501e-01</td>\n",
       "      <td>2.947012e-01</td>\n",
       "      <td>2.982019e-01</td>\n",
       "      <td>3.327319e-01</td>\n",
       "      <td>3.008077e-01</td>\n",
       "      <td>2.962162e-01</td>\n",
       "      <td>2.967645e-01</td>\n",
       "      <td>2.878847e-01</td>\n",
       "      <td>3.004508e-01</td>\n",
       "      <td>2.933959e-01</td>\n",
       "      <td>5.649273e+07</td>\n",
       "      <td>9.421144</td>\n",
       "      <td>7.897302e+00</td>\n",
       "      <td>0.402552</td>\n",
       "      <td>0.296801</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>1.028600e+04</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1228.000000</td>\n",
       "      <td>2.346000e-01</td>\n",
       "      <td>1.747000e-01</td>\n",
       "      <td>2.236000e-01</td>\n",
       "      <td>2.289000e-01</td>\n",
       "      <td>2.330000e-01</td>\n",
       "      <td>1.654000e-01</td>\n",
       "      <td>2.302000e-01</td>\n",
       "      <td>2.310000e-01</td>\n",
       "      <td>2.377000e-01</td>\n",
       "      <td>2.355000e-01</td>\n",
       "      <td>2.235000e-01</td>\n",
       "      <td>3.898000e-01</td>\n",
       "      <td>3.043392e+07</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>0.070000</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2483.000000</td>\n",
       "      <td>4.546000e-01</td>\n",
       "      <td>4.114000e-01</td>\n",
       "      <td>4.704000e-01</td>\n",
       "      <td>4.589000e-01</td>\n",
       "      <td>4.667000e-01</td>\n",
       "      <td>4.495000e-01</td>\n",
       "      <td>4.679000e-01</td>\n",
       "      <td>4.620000e-01</td>\n",
       "      <td>4.699000e-01</td>\n",
       "      <td>4.569000e-01</td>\n",
       "      <td>4.585000e-01</td>\n",
       "      <td>6.432000e-01</td>\n",
       "      <td>6.700125e+07</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>3.000000e+00</td>\n",
       "      <td>0.350000</td>\n",
       "      <td>0.001000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>3746.000000</td>\n",
       "      <td>6.991000e-01</td>\n",
       "      <td>6.991000e-01</td>\n",
       "      <td>7.390000e-01</td>\n",
       "      <td>7.122000e-01</td>\n",
       "      <td>7.220000e-01</td>\n",
       "      <td>7.574000e-01</td>\n",
       "      <td>7.283000e-01</td>\n",
       "      <td>7.179000e-01</td>\n",
       "      <td>7.241000e-01</td>\n",
       "      <td>7.032000e-01</td>\n",
       "      <td>7.185000e-01</td>\n",
       "      <td>8.813000e-01</td>\n",
       "      <td>1.130781e+08</td>\n",
       "      <td>9.000000</td>\n",
       "      <td>6.000000e+00</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.078000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>5000.000000</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>2.489456e+08</td>\n",
       "      <td>358.000000</td>\n",
       "      <td>3.540000e+02</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            DISTANCE    gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF        MAX.AF  chrCoordinate    EXONnumber  INTRONnumber    SIFTscore  PolyPhenscore\n",
       "count  444320.000000  3.674811e+06    3.674809e+06    3.674742e+06    3.674810e+06    3.674806e+06    3.674802e+06    3.674804e+06    3.674681e+06    3.674811e+06    3.674795e+06    3.674798e+06  3.674883e+06   3.701014e+06  87596.000000  2.002513e+06  8629.000000    8311.000000\n",
       "mean     2488.533665  4.718748e-01    4.426008e-01    4.853570e-01    4.760430e-01    4.827356e-01    4.682483e-01    4.841243e-01    4.789125e-01    4.857458e-01    4.744601e-01    4.761085e-01  6.144999e-01   7.697594e+07      7.139847  5.279100e+00     0.462458       0.148457\n",
       "std      1450.436258  2.861300e-01    3.019205e-01    3.081501e-01    2.947012e-01    2.982019e-01    3.327319e-01    3.008077e-01    2.962162e-01    2.967645e-01    2.878847e-01    3.004508e-01  2.933959e-01   5.649273e+07      9.421144  7.897302e+00     0.402552       0.296801\n",
       "min         0.000000  0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00    0.000000e+00  0.000000e+00   1.028600e+04      1.000000  1.000000e+00     0.000000       0.000000\n",
       "25%      1228.000000  2.346000e-01    1.747000e-01    2.236000e-01    2.289000e-01    2.330000e-01    1.654000e-01    2.302000e-01    2.310000e-01    2.377000e-01    2.355000e-01    2.235000e-01  3.898000e-01   3.043392e+07      2.000000  1.000000e+00     0.070000       0.000000\n",
       "50%      2483.000000  4.546000e-01    4.114000e-01    4.704000e-01    4.589000e-01    4.667000e-01    4.495000e-01    4.679000e-01    4.620000e-01    4.699000e-01    4.569000e-01    4.585000e-01  6.432000e-01   6.700125e+07      4.000000  3.000000e+00     0.350000       0.001000\n",
       "75%      3746.000000  6.991000e-01    6.991000e-01    7.390000e-01    7.122000e-01    7.220000e-01    7.574000e-01    7.283000e-01    7.179000e-01    7.241000e-01    7.032000e-01    7.185000e-01  8.813000e-01   1.130781e+08      9.000000  6.000000e+00     1.000000       0.078000\n",
       "max      5000.000000  1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00    1.000000e+00  1.000000e+00   2.489456e+08    358.000000  3.540000e+02     1.000000       1.000000"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad63.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 96)\n",
      "Uploaded.variation           object\n",
      "Location                     object\n",
      "Allele                       object\n",
      "Gene                         object\n",
      "Feature                      object\n",
      "Feature.type                 object\n",
      "Consequence                  object\n",
      "cDNA.position                object\n",
      "CDS.position                 object\n",
      "Protein.position             object\n",
      "Amino.acids                  object\n",
      "Codons                       object\n",
      "Existing.variation           object\n",
      "IND                          object\n",
      "ZYG                          object\n",
      "REF.ALLELE                   object\n",
      "IMPACT                       object\n",
      "DISTANCE                    float64\n",
      "STRAND                       object\n",
      "FLAGS                        object\n",
      "VARIANT.CLASS                object\n",
      "SYMBOL                       object\n",
      "SYMBOL.SOURCE                object\n",
      "HGNC.ID                      object\n",
      "BIOTYPE                      object\n",
      "CANONICAL                    object\n",
      "MANE.SELECT                  object\n",
      "MANE.PLUS.CLINICAL           object\n",
      "TSL                          object\n",
      "APPRIS                       object\n",
      "CCDS                         object\n",
      "ENSP                         object\n",
      "SWISSPROT                    object\n",
      "TREMBL                       object\n",
      "UNIPARC                      object\n",
      "UNIPROT.ISOFORM              object\n",
      "REFSEQ.MATCH                 object\n",
      "SOURCE                       object\n",
      "REFSEQ.OFFSET                object\n",
      "GIVEN.REF                    object\n",
      "USED.REF                     object\n",
      "BAM.EDIT                     object\n",
      "GENE.PHENO                   object\n",
      "SIFT                         object\n",
      "PolyPhen                     object\n",
      "EXON                         object\n",
      "INTRON                       object\n",
      "DOMAINS                      object\n",
      "miRNA                        object\n",
      "HGVSc                        object\n",
      "HGVSp                        object\n",
      "HGVS.OFFSET                  object\n",
      "gnomADg.AF                  float64\n",
      "gnomADg.AFR.AF              float64\n",
      "gnomADg.AMI.AF              float64\n",
      "gnomADg.AMR.AF              float64\n",
      "gnomADg.ASJ.AF              float64\n",
      "gnomADg.EAS.AF              float64\n",
      "gnomADg.FIN.AF              float64\n",
      "gnomADg.MID.AF              float64\n",
      "gnomADg.NFE.AF              float64\n",
      "gnomADg.OTH.AF              float64\n",
      "gnomADg.SAS.AF              float64\n",
      "MAX.AF                      float64\n",
      "MAX.AF.POPS                  object\n",
      "CLIN.SIG                     object\n",
      "SOMATIC                      object\n",
      "PHENO                        object\n",
      "PUBMED                       object\n",
      "MOTIF.NAME                   object\n",
      "MOTIF.POS                    object\n",
      "HIGH.INF.POS                 object\n",
      "MOTIF.SCORE.CHANGE           object\n",
      "TRANSCRIPTION.FACTORS        object\n",
      "chromosome                   object\n",
      "chrCoordinate                 int64\n",
      "EXONnumber                  float64\n",
      "INTRONnumber                float64\n",
      "chr                          object\n",
      "GIVEN.REF.MATCH                bool\n",
      "SIFTprediction               object\n",
      "SIFTscore                   float64\n",
      "PolyPhenprediction           object\n",
      "PolyPhenscore               float64\n",
      "keyID38                      object\n",
      "var.in.both.lines              bool\n",
      "var.exclusive.btw.lines        bool\n",
      "gene.in.both.lines             bool\n",
      "gene.exclusive.btw.lines       bool\n",
      "MAF                          object\n",
      "MAF2                         object\n",
      "PATHO                          bool\n",
      "VUS                            bool\n",
      "BENIGN                         bool\n",
      "ClinVar                      object\n",
      "Database                     object\n",
      "dtype: object\n",
      "Uploaded.variation                0\n",
      "Location                          0\n",
      "Allele                            0\n",
      "Gene                              0\n",
      "Feature                           0\n",
      "Feature.type                      0\n",
      "Consequence                       0\n",
      "cDNA.position               3410222\n",
      "CDS.position                3476095\n",
      "Protein.position            3476095\n",
      "Amino.acids                       0\n",
      "Codons                            0\n",
      "Existing.variation                0\n",
      "IND                               0\n",
      "ZYG                               0\n",
      "REF.ALLELE                        0\n",
      "IMPACT                            0\n",
      "DISTANCE                    3059299\n",
      "STRAND                            0\n",
      "FLAGS                             0\n",
      "VARIANT.CLASS                     0\n",
      "SYMBOL                            0\n",
      "SYMBOL.SOURCE                     0\n",
      "HGNC.ID                           0\n",
      "BIOTYPE                           0\n",
      "CANONICAL                         0\n",
      "MANE.SELECT                       0\n",
      "MANE.PLUS.CLINICAL                0\n",
      "TSL                               0\n",
      "APPRIS                            0\n",
      "CCDS                              0\n",
      "ENSP                              0\n",
      "SWISSPROT                         0\n",
      "TREMBL                            0\n",
      "UNIPARC                           0\n",
      "UNIPROT.ISOFORM                   0\n",
      "REFSEQ.MATCH                      0\n",
      "SOURCE                            0\n",
      "REFSEQ.OFFSET                     0\n",
      "GIVEN.REF                         0\n",
      "USED.REF                          0\n",
      "BAM.EDIT                          0\n",
      "GENE.PHENO                        0\n",
      "SIFT                              0\n",
      "PolyPhen                          0\n",
      "EXON                              0\n",
      "INTRON                            0\n",
      "DOMAINS                           0\n",
      "miRNA                             0\n",
      "HGVSc                             0\n",
      "HGVSp                             0\n",
      "HGVS.OFFSET                       0\n",
      "gnomADg.AF                    35940\n",
      "gnomADg.AFR.AF                35942\n",
      "gnomADg.AMI.AF                36036\n",
      "gnomADg.AMR.AF                35943\n",
      "gnomADg.ASJ.AF                35959\n",
      "gnomADg.EAS.AF                35951\n",
      "gnomADg.FIN.AF                35947\n",
      "gnomADg.MID.AF                36101\n",
      "gnomADg.NFE.AF                35940\n",
      "gnomADg.OTH.AF                35963\n",
      "gnomADg.SAS.AF                35949\n",
      "MAX.AF                        35779\n",
      "MAX.AF.POPS                       0\n",
      "CLIN.SIG                          0\n",
      "SOMATIC                           0\n",
      "PHENO                             0\n",
      "PUBMED                            0\n",
      "MOTIF.NAME                        0\n",
      "MOTIF.POS                         0\n",
      "HIGH.INF.POS                      0\n",
      "MOTIF.SCORE.CHANGE                0\n",
      "TRANSCRIPTION.FACTORS             0\n",
      "chromosome                        0\n",
      "chrCoordinate                     0\n",
      "EXONnumber                  3410232\n",
      "INTRONnumber                1608377\n",
      "chr                               0\n",
      "GIVEN.REF.MATCH                   0\n",
      "SIFTprediction               546104\n",
      "SIFTscore                   3487459\n",
      "PolyPhenprediction           546104\n",
      "PolyPhenscore               3487757\n",
      "keyID38                      546104\n",
      "var.in.both.lines                 0\n",
      "var.exclusive.btw.lines           0\n",
      "gene.in.both.lines                0\n",
      "gene.exclusive.btw.lines          0\n",
      "MAF                               0\n",
      "MAF2                              0\n",
      "PATHO                             0\n",
      "VUS                               0\n",
      "BENIGN                            0\n",
      "ClinVar                     3464042\n",
      "Database                          0\n",
      "dtype: int64\n",
      "(3701014, 96)\n",
      "Uploaded.variation           object\n",
      "Location                     object\n",
      "Allele                       object\n",
      "Gene                         object\n",
      "Feature                      object\n",
      "Feature.type                 object\n",
      "Consequence                  object\n",
      "cDNA.position                object\n",
      "CDS.position                 object\n",
      "Protein.position             object\n",
      "Amino.acids                  object\n",
      "Codons                       object\n",
      "Existing.variation           object\n",
      "IND                          object\n",
      "ZYG                          object\n",
      "REF.ALLELE                   object\n",
      "IMPACT                       object\n",
      "DISTANCE                    float64\n",
      "STRAND                       object\n",
      "FLAGS                        object\n",
      "VARIANT.CLASS                object\n",
      "SYMBOL                       object\n",
      "SYMBOL.SOURCE                object\n",
      "HGNC.ID                      object\n",
      "BIOTYPE                      object\n",
      "CANONICAL                    object\n",
      "MANE.SELECT                  object\n",
      "MANE.PLUS.CLINICAL           object\n",
      "TSL                          object\n",
      "APPRIS                       object\n",
      "CCDS                         object\n",
      "ENSP                         object\n",
      "SWISSPROT                    object\n",
      "TREMBL                       object\n",
      "UNIPARC                      object\n",
      "UNIPROT.ISOFORM              object\n",
      "REFSEQ.MATCH                 object\n",
      "SOURCE                       object\n",
      "REFSEQ.OFFSET                object\n",
      "GIVEN.REF                    object\n",
      "USED.REF                     object\n",
      "BAM.EDIT                     object\n",
      "GENE.PHENO                   object\n",
      "SIFT                         object\n",
      "PolyPhen                     object\n",
      "EXON                         object\n",
      "INTRON                       object\n",
      "DOMAINS                      object\n",
      "miRNA                        object\n",
      "HGVSc                        object\n",
      "HGVSp                        object\n",
      "HGVS.OFFSET                  object\n",
      "gnomADg.AF                  float64\n",
      "gnomADg.AFR.AF              float64\n",
      "gnomADg.AMI.AF              float64\n",
      "gnomADg.AMR.AF              float64\n",
      "gnomADg.ASJ.AF              float64\n",
      "gnomADg.EAS.AF              float64\n",
      "gnomADg.FIN.AF              float64\n",
      "gnomADg.MID.AF              float64\n",
      "gnomADg.NFE.AF              float64\n",
      "gnomADg.OTH.AF              float64\n",
      "gnomADg.SAS.AF              float64\n",
      "MAX.AF                      float64\n",
      "MAX.AF.POPS                  object\n",
      "CLIN.SIG                     object\n",
      "SOMATIC                      object\n",
      "PHENO                        object\n",
      "PUBMED                       object\n",
      "MOTIF.NAME                   object\n",
      "MOTIF.POS                    object\n",
      "HIGH.INF.POS                 object\n",
      "MOTIF.SCORE.CHANGE           object\n",
      "TRANSCRIPTION.FACTORS        object\n",
      "chromosome                   object\n",
      "chrCoordinate                 int64\n",
      "EXONnumber                  float64\n",
      "INTRONnumber                float64\n",
      "chr                          object\n",
      "GIVEN.REF.MATCH                bool\n",
      "SIFTprediction               object\n",
      "SIFTscore                   float64\n",
      "PolyPhenprediction           object\n",
      "PolyPhenscore               float64\n",
      "keyID38                      object\n",
      "var.in.both.lines              bool\n",
      "var.exclusive.btw.lines        bool\n",
      "gene.in.both.lines             bool\n",
      "gene.exclusive.btw.lines       bool\n",
      "MAF                          object\n",
      "MAF2                         object\n",
      "PATHO                          bool\n",
      "VUS                            bool\n",
      "BENIGN                         bool\n",
      "ClinVar                      object\n",
      "Database                     object\n",
      "dtype: object\n",
      "Uploaded.variation                0\n",
      "Location                          0\n",
      "Allele                            0\n",
      "Gene                              0\n",
      "Feature                           0\n",
      "Feature.type                      0\n",
      "Consequence                       0\n",
      "cDNA.position               3613408\n",
      "CDS.position                3681427\n",
      "Protein.position            3681427\n",
      "Amino.acids                       0\n",
      "Codons                            0\n",
      "Existing.variation                0\n",
      "IND                               0\n",
      "ZYG                               0\n",
      "REF.ALLELE                        0\n",
      "IMPACT                            0\n",
      "DISTANCE                    3256694\n",
      "STRAND                            0\n",
      "FLAGS                             0\n",
      "VARIANT.CLASS                     0\n",
      "SYMBOL                            0\n",
      "SYMBOL.SOURCE                     0\n",
      "HGNC.ID                           0\n",
      "BIOTYPE                           0\n",
      "CANONICAL                         0\n",
      "MANE.SELECT                       0\n",
      "MANE.PLUS.CLINICAL                0\n",
      "TSL                               0\n",
      "APPRIS                            0\n",
      "CCDS                              0\n",
      "ENSP                              0\n",
      "SWISSPROT                         0\n",
      "TREMBL                            0\n",
      "UNIPARC                           0\n",
      "UNIPROT.ISOFORM                   0\n",
      "REFSEQ.MATCH                      0\n",
      "SOURCE                            0\n",
      "REFSEQ.OFFSET                     0\n",
      "GIVEN.REF                         0\n",
      "USED.REF                          0\n",
      "BAM.EDIT                          0\n",
      "GENE.PHENO                        0\n",
      "SIFT                              0\n",
      "PolyPhen                          0\n",
      "EXON                              0\n",
      "INTRON                            0\n",
      "DOMAINS                           0\n",
      "miRNA                             0\n",
      "HGVSc                             0\n",
      "HGVSp                             0\n",
      "HGVS.OFFSET                       0\n",
      "gnomADg.AF                    26203\n",
      "gnomADg.AFR.AF                26205\n",
      "gnomADg.AMI.AF                26272\n",
      "gnomADg.AMR.AF                26204\n",
      "gnomADg.ASJ.AF                26208\n",
      "gnomADg.EAS.AF                26212\n",
      "gnomADg.FIN.AF                26210\n",
      "gnomADg.MID.AF                26333\n",
      "gnomADg.NFE.AF                26203\n",
      "gnomADg.OTH.AF                26219\n",
      "gnomADg.SAS.AF                26216\n",
      "MAX.AF                        26131\n",
      "MAX.AF.POPS                       0\n",
      "CLIN.SIG                          0\n",
      "SOMATIC                           0\n",
      "PHENO                             0\n",
      "PUBMED                            0\n",
      "MOTIF.NAME                        0\n",
      "MOTIF.POS                         0\n",
      "HIGH.INF.POS                      0\n",
      "MOTIF.SCORE.CHANGE                0\n",
      "TRANSCRIPTION.FACTORS             0\n",
      "chromosome                        0\n",
      "chrCoordinate                     0\n",
      "EXONnumber                  3613418\n",
      "INTRONnumber                1698501\n",
      "chr                               0\n",
      "GIVEN.REF.MATCH                   0\n",
      "SIFTprediction               562353\n",
      "SIFTscore                   3692385\n",
      "PolyPhenprediction           562353\n",
      "PolyPhenscore               3692703\n",
      "keyID38                      562353\n",
      "var.in.both.lines                 0\n",
      "var.exclusive.btw.lines           0\n",
      "gene.in.both.lines                0\n",
      "gene.exclusive.btw.lines          0\n",
      "MAF                               0\n",
      "MAF2                              0\n",
      "PATHO                             0\n",
      "VUS                               0\n",
      "BENIGN                            0\n",
      "ClinVar                     3667535\n",
      "Database                          0\n",
      "dtype: int64\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<function all_funx.describeMe(df)>"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "describeMe(cad12)\n",
    "describeMe(cad63)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "cad12.to_csv('4-cad12-snv-indel-3496201.csv', index=False)\n",
    "cad63.to_csv('4-cad63-snv-indel-3701014.csv', index=False)\n",
    "\n",
    "# TODO \n",
    "# 1. (done) note the clinvar col with labels is not star reviewed, check the clin.sig column to confirm high confidence pathogenicity\n",
    "# 2. (done) added Database column here instead of in R to keep all label functions together in python"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "\n",
    "# SAVE both individuals df\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(7197215, 26)\n"
     ]
    }
   ],
   "source": [
    "simple12 = cad12[keepme].copy()\n",
    "simple63 = cad63[keepme].copy()\n",
    "\n",
    "\n",
    "both = pd.concat([simple12, simple63], axis=0)\n",
    "print(both.shape)# (7197215, 32)\n",
    "\n",
    "both.to_csv('4-keepme-combined-cad12-cad63-7197215.csv', index=False)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py39",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
