{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Restarted py39 (Python 3.9.-1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "7a4d150d-5e35-4a08-8b58-491ef1a68850",
   "metadata": {},
   "outputs": [],
   "source": [
    "from all_funx import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ea2f9ad9-2d35-4b59-8b3b-6bce1348a1c3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# import dfs from 1_clean_excel_summary.py\n",
    "path  = \"/Users/mariapalafox/Desktop/BOLDcaution/WGS/5-annotation/3-interpret/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "216ec7fd-050d-4a84-97d7-018da0cf6acf",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(546104, 80)\n",
      "(2950097, 85)\n"
     ]
    }
   ],
   "source": [
    "indel12 = pd.read_csv('1_clean_vep_output-indel-cad12-v2.csv', low_memory=False)\n",
    "snp12 = pd.read_csv('1_clean_vep_output-snp-cad12-v2.csv', low_memory=False)\n",
    "print(indel12.shape)\n",
    "print(snp12.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "3a2a72b6-0467-492c-9ee3-132410146004",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(562353, 80)\n",
      "(3138661, 85)\n"
     ]
    }
   ],
   "source": [
    "indel63 = pd.read_csv('1_clean_vep_output-indel-cad63-v2.csv', low_memory=False)\n",
    "snp63 = pd.read_csv('1_clean_vep_output-snp-cad63-v2.csv', low_memory=False)\n",
    "print(indel63.shape)\n",
    "print(snp63.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "978cedfb-f919-4b78-bbe7-d9c4a91e8218",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 85)\n",
      "(3701014, 85)\n"
     ]
    }
   ],
   "source": [
    "cad12 = pd.concat([indel12, snp12], axis=0)\n",
    "cad63 = pd.concat([indel63, snp63], axis=0)\n",
    "print(cad12.shape)\n",
    "print(cad63.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7a387389-81e3-45a5-a199-8832a3d192e0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1525780 1730593 1970421\n"
     ]
    }
   ],
   "source": [
    "# create a list of unique 'Uploaded_variation' in both dataframes\n",
    "cad12_variations = cad12['Uploaded_variation'].unique()\n",
    "cad63_variations = cad63['Uploaded_variation'].unique()\n",
    "# create a separate list for variations that are exclusively in cad12\n",
    "cad12_exclusive_variations = list(set(cad12_variations) - set(cad63_variations))\n",
    "# create a separate list for variations that are exclusively in cad63\n",
    "cad63_exclusive_variations = list(set(cad63_variations) - set(cad12_variations))\n",
    "# create a separate list for variations that are in both cad12 and cad63\n",
    "common_variations = list(set(cad12_variations) & set(cad63_variations))\n",
    "print( len(cad12_exclusive_variations), len(cad63_exclusive_variations), len(common_variations)) # 1525780 1730593 1970421"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "5c17f2ee-ba76-4812-83c4-dc185fed2e7f",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "809 743 28137\n"
     ]
    }
   ],
   "source": [
    "# Optimize the code by avoiding unnecessary function calls and repeated operations\n",
    "cad12['var_in_both_lines'] = cad12['Uploaded_variation'].isin(common_variations)\n",
    "cad12['var_exclusive_btw_lines'] = cad12['Uploaded_variation'].isin(cad12_exclusive_variations)\n",
    "cad63['var_in_both_lines'] = cad63['Uploaded_variation'].isin(common_variations)\n",
    "cad63['var_exclusive_btw_lines'] = cad63['Uploaded_variation'].isin(cad63_exclusive_variations)\n",
    "\n",
    "\n",
    "# create a list of unique 'SYMBOL' in both dataframes\n",
    "cad12_genes = cad12['SYMBOL'].unique()\n",
    "cad63_genes = cad63['SYMBOL'].unique()\n",
    "# create a separate list for SYMBOL that are exclusively in cad12\n",
    "cad12_exclusive = list(set(cad12_genes) - set(cad63_genes))\n",
    "# create a separate list for SYMBOL that are exclusively in cad63\n",
    "cad63_exclusive = list(set(cad63_genes) - set(cad12_genes))\n",
    "# create a separate list for SYMBOL that are in both cad12 and cad63\n",
    "common_genes = list(set(cad12_genes) & set(cad63_genes))\n",
    "cad12['gene_in_both_lines'] = cad12['SYMBOL'].isin(common_genes)\n",
    "cad12['gene_exclusive_btw_lines'] = cad12['SYMBOL'].isin(cad12_exclusive)\n",
    "cad63['gene_in_both_lines'] = cad63['SYMBOL'].isin(common_genes)\n",
    "cad63['gene_exclusive_btw_lines'] = cad63['SYMBOL'].isin(cad63_exclusive)\n",
    "print( len(cad12_exclusive), len(cad63_exclusive), len(common_genes)) #809 743 28137"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "c2ba16ce-8688-4c1f-8987-b2cb037e8972",
   "metadata": {},
   "outputs": [],
   "source": [
    "save_describe(cad12, '4-cad12')\n",
    "save_describe(cad63, '4-cad63')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "cf118c35-fce5-4ab0-a2f9-786b5bf97c08",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
      "       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN_REF_MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'],\n",
      "      dtype='object')\n",
      "(3496201, 89)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"\\n['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\\n       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN_REF_MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'],\\n      dtype='object')\\n(3496201, 89)\\n\""
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(cad12.columns)\n",
    "print(cad12.shape)\n",
    "\"\"\"\n",
    "['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
    "       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN_REF_MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'],\n",
    "      dtype='object')\n",
    "(3496201, 89)\n",
    "\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e90ddf79-fec4-4967-a374-2aa63181aa18",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
      "       'TRANSCRIPTION_FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN_REF_MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var_in_both_lines', 'var_exclusive_btw_lines', 'gene_in_both_lines', 'gene_exclusive_btw_lines'],\n",
      "      dtype='object')\n",
      "(3701014, 89)\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'(3701014, 89)'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(cad63.columns)\n",
    "print(cad63.shape)\n",
    "\"\"\"(3701014, 89)\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "fd8a5abb-4757-46ec-bde4-f13e6b8bd43d",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['Uploaded.variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature.type', 'Consequence', 'cDNA.position', 'CDS.position', 'Protein.position', 'Amino.acids', 'Codons', 'Existing.variation', 'IND', 'ZYG', 'REF.ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT.CLASS', 'SYMBOL', 'SYMBOL.SOURCE', 'HGNC.ID', 'BIOTYPE', 'CANONICAL', 'MANE.SELECT', 'MANE.PLUS.CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT.ISOFORM', 'REFSEQ.MATCH', 'SOURCE', 'REFSEQ.OFFSET', 'GIVEN.REF', 'USED.REF', 'BAM.EDIT', 'GENE.PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS.OFFSET', 'gnomADg.AF', 'gnomADg.AFR.AF', 'gnomADg.AMI.AF', 'gnomADg.AMR.AF', 'gnomADg.ASJ.AF', 'gnomADg.EAS.AF', 'gnomADg.FIN.AF', 'gnomADg.MID.AF', 'gnomADg.NFE.AF', 'gnomADg.OTH.AF', 'gnomADg.SAS.AF', 'MAX.AF', 'MAX.AF.POPS', 'CLIN.SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF.NAME', 'MOTIF.POS', 'HIGH.INF.POS', 'MOTIF.SCORE.CHANGE',\n",
      "       'TRANSCRIPTION.FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN.REF.MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var.in.both.lines', 'var.exclusive.btw.lines', 'gene.in.both.lines', 'gene.exclusive.btw.lines'],\n",
      "      dtype='object')\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "\"'Uploaded.variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature.type', 'Consequence', 'cDNA.position', 'CDS.position', 'Protein.position', 'Amino.acids', 'Codons', 'Existing.variation', 'IND', 'ZYG', 'REF.ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT.CLASS', 'SYMBOL', 'SYMBOL.SOURCE', 'HGNC.ID', 'BIOTYPE', 'CANONICAL', 'MANE.SELECT', 'MANE.PLUS.CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT.ISOFORM', 'REFSEQ.MATCH', 'SOURCE', 'REFSEQ.OFFSET', 'GIVEN.REF', 'USED.REF', 'BAM.EDIT', 'GENE.PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS.OFFSET', 'gnomADg.AF', 'gnomADg.AFR.AF', 'gnomADg.AMI.AF', 'gnomADg.AMR.AF', 'gnomADg.ASJ.AF', 'gnomADg.EAS.AF', 'gnomADg.FIN.AF', 'gnomADg.MID.AF', 'gnomADg.NFE.AF', 'gnomADg.OTH.AF', 'gnomADg.SAS.AF', 'MAX.AF', 'MAX.AF.POPS', 'CLIN.SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF.NAME', 'MOTIF.POS', 'HIGH.INF.POS', 'MOTIF.SCORE.CHANGE',\\n       'TRANSCRIPTION.FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN.REF.MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var.in.both.lines', 'var.exclusive.btw.lines', 'gene.in.both.lines', 'gene.exclusive.btw.lines\""
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# fix column names by replacing _ with .\n",
    "cad12.columns = cad12.columns.str.replace('_', '.')\n",
    "cad63.columns = cad63.columns.str.replace('_', '.')\n",
    "\n",
    "\n",
    "print(cad63.columns)\n",
    "\"\"\"'Uploaded.variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature.type', 'Consequence', 'cDNA.position', 'CDS.position', 'Protein.position', 'Amino.acids', 'Codons', 'Existing.variation', 'IND', 'ZYG', 'REF.ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT.CLASS', 'SYMBOL', 'SYMBOL.SOURCE', 'HGNC.ID', 'BIOTYPE', 'CANONICAL', 'MANE.SELECT', 'MANE.PLUS.CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT.ISOFORM', 'REFSEQ.MATCH', 'SOURCE', 'REFSEQ.OFFSET', 'GIVEN.REF', 'USED.REF', 'BAM.EDIT', 'GENE.PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS.OFFSET', 'gnomADg.AF', 'gnomADg.AFR.AF', 'gnomADg.AMI.AF', 'gnomADg.AMR.AF', 'gnomADg.ASJ.AF', 'gnomADg.EAS.AF', 'gnomADg.FIN.AF', 'gnomADg.MID.AF', 'gnomADg.NFE.AF', 'gnomADg.OTH.AF', 'gnomADg.SAS.AF', 'MAX.AF', 'MAX.AF.POPS', 'CLIN.SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF.NAME', 'MOTIF.POS', 'HIGH.INF.POS', 'MOTIF.SCORE.CHANGE',\n",
    "       'TRANSCRIPTION.FACTORS', 'chromosome', 'chrCoordinate', 'EXONnumber', 'INTRONnumber', 'chr', 'GIVEN.REF.MATCH', 'SIFTprediction', 'SIFTscore', 'PolyPhenprediction', 'PolyPhenscore', 'keyID38', 'var.in.both.lines', 'var.exclusive.btw.lines', 'gene.in.both.lines', 'gene.exclusive.btw.lines\"\"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "20823b15-b5ec-4325-87f2-6beaeaf95667",
   "metadata": {},
   "outputs": [],
   "source": [
    "# add MAF column to  dataframes, if MAX.AF is < 0.01 label Rare, if its less than 0.1 label less common, if its greater than or equal  to 0.1 label common, else label empty string\n",
    "cad12['MAF'] = np.select([cad12['MAX.AF'] < 0.01, \n",
    "                          (cad12['MAX.AF'] >= 0.01) & (cad12['MAX.AF'] < 0.1),\n",
    "                            cad12['MAX.AF'] >= 0.1], \n",
    "                            ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], \n",
    "                            default = np.nan)\n",
    "\n",
    "cad63['MAF'] = np.select([cad63['MAX.AF'] < 0.01, \n",
    "                          (cad63['MAX.AF'] >= 0.01) & (cad63['MAX.AF'] < 0.1), \n",
    "                          cad63['MAX.AF'] >= 0.1], \n",
    "                          ['Rare (<1%)', 'Less common (<10%)', 'Common (>=10%)'], \n",
    "                          default = np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "23b1e0ee-86db-470d-bcf2-ee71e84a6714",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert the R code into python\n",
    "cad12['MAF2'] = np.select([cad12['MAX.AF'] < 0.05, \n",
    "                           cad12['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'],\n",
    "                            default=np.nan)\n",
    "\n",
    "cad63['MAF2'] = np.select([cad63['MAX.AF'] < 0.05, \n",
    "                           cad63['MAX.AF'] >= 0.05], \n",
    "                           ['Rare (<5%)', 'Common (>=5%)'], \n",
    "                           default=np.nan)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "2c3e0494-4a2d-4421-aae3-d8d901cf6d36",
   "metadata": {},
   "outputs": [],
   "source": [
    "# functionn that checks CLIN.SIG column for rows containing 'pathogenic' or 'likely pathogenic' and creates new column with 'PATHO' label. the new column should also  contain 'VUS' and 'BENIGN' labels if the row contain 'benign' for 'BENIGN' and 'uncertain significance' for 'VUS'.\n",
    "\n",
    "def label_clin_sig(df):\n",
    "    df['PATHO'] = df['CLIN.SIG'].str.contains('pathogenic|likely pathogenic', case=False)\n",
    "    df['VUS'] = df['CLIN.SIG'].str.contains('uncertain_significance', case=False)\n",
    "    df['BENIGN'] = df['CLIN.SIG'].str.contains('benign|likely_benign', case=False)\n",
    "    return df\n",
    "\n",
    "\n",
    "# new function creates one column with new labels based on rule heirarchy; if pathogenic or likely pathogenic then labeled PATHO else if uncertain_significance then VUS else if benign or likely benign then BENIGN\n",
    "def label_clin_sig2(df):\n",
    "    newcol = []\n",
    "    for row in df.itertuples():\n",
    "        if row.PATHO:\n",
    "            newcol.append('PATHO')\n",
    "        elif row.VUS:\n",
    "            newcol.append('VUS')\n",
    "        elif row.BENIGN:\n",
    "            newcol.append('BENIGN')\n",
    "        else:\n",
    "            newcol.append(np.nan) # append nan value instead of '' emptry string\n",
    "    df['ClinVar'] = newcol\n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "905fae81-c738-4e9e-ae07-c729d7085baa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3464042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>31769</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>302</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3464042\n",
       "1  BENIGN    31769\n",
       "2     VUS      302\n",
       "3   PATHO       88"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad12 = label_clin_sig(cad12)\n",
    "cad12 = label_clin_sig2(cad12)\n",
    "checkColumnValue(cad12, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "c6cd0f0b-d4ce-4025-be86-c6c8b342a86b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>3667535</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BENIGN</td>\n",
       "      <td>33038</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>VUS</td>\n",
       "      <td>330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PATHO</td>\n",
       "      <td>111</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Count    count\n",
       "0     NaN  3667535\n",
       "1  BENIGN    33038\n",
       "2     VUS      330\n",
       "3   PATHO      111"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# label ClinVar using not star reviewed system, over-estimates number of pathogenic variants\n",
    "cad63 = label_clin_sig(cad63)\n",
    "cad63 = label_clin_sig2(cad63)\n",
    "checkColumnValue(cad63, 'ClinVar')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "e41d4ab0-2622-4230-b9f0-7a05a0c2d0c2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P732912</th>\n",
       "      <td>31769</td>\n",
       "      <td>3464042</td>\n",
       "      <td>88</td>\n",
       "      <td>302</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>31769</td>\n",
       "      <td>3464042</td>\n",
       "      <td>88</td>\n",
       "      <td>302</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P732912   31769  3464042     88  302  3496201\n",
       "Total     31769  3464042     88  302  3496201"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho12 = cad12[cad12['ClinVar'] == 'PATHO']\n",
    "crosstabit(cad12, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "2587e33f-07a4-40ec-b3d8-6dbccba32f5b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>ClinVar</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>NA</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>IND</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P931263</th>\n",
       "      <td>33038</td>\n",
       "      <td>3667535</td>\n",
       "      <td>111</td>\n",
       "      <td>330</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>33038</td>\n",
       "      <td>3667535</td>\n",
       "      <td>111</td>\n",
       "      <td>330</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "ClinVar  BENIGN       NA  PATHO  VUS    Total\n",
       "IND                                          \n",
       "P931263   33038  3667535    111  330  3701014\n",
       "Total     33038  3667535    111  330  3701014"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho63 = cad63[cad63['ClinVar'] == 'PATHO']\n",
    "crosstabit(cad63, 'IND', 'ClinVar', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "baf0e47c-81e6-4341-a17f-c04d1ef010b6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature.type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA.position</th>\n",
       "      <th>CDS.position</th>\n",
       "      <th>Protein.position</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing.variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>REF.ALLELE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL.SOURCE</th>\n",
       "      <th>HGNC.ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE.SELECT</th>\n",
       "      <th>MANE.PLUS.CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT.ISOFORM</th>\n",
       "      <th>REFSEQ.MATCH</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>REFSEQ.OFFSET</th>\n",
       "      <th>GIVEN.REF</th>\n",
       "      <th>USED.REF</th>\n",
       "      <th>BAM.EDIT</th>\n",
       "      <th>GENE.PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVS.OFFSET</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>chr</th>\n",
       "      <th>GIVEN.REF.MATCH</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "      <th>keyID38</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>var.exclusive.btw.lines</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>gene.exclusive.btw.lines</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>ClinVar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2118</th>\n",
       "      <td>chr1_7961851_GTGCTGGACGGTGTCCCT/-</td>\n",
       "      <td>chr1:7961851-7961868</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000116288</td>\n",
       "      <td>ENST00000338639</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs200968609</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>GTGCTGGACGGTGTCCCT</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>PARK7</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:16369</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_007262.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS93.1</td>\n",
       "      <td>ENSP00000340278</td>\n",
       "      <td>Q99497.214</td>\n",
       "      <td>V9HWC2.55</td>\n",
       "      <td>UPI0000072F6F</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>GTGCTGGACGGTGTCCCT</td>\n",
       "      <td>GTGCTGGACGGTGTCCCT</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/6</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000338639.10:c.-24+75_-24+92del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.139600</td>\n",
       "      <td>0.046980</td>\n",
       "      <td>0.045050</td>\n",
       "      <td>0.1133</td>\n",
       "      <td>0.11430</td>\n",
       "      <td>0.06402</td>\n",
       "      <td>0.315800</td>\n",
       "      <td>0.095540</td>\n",
       "      <td>0.186000</td>\n",
       "      <td>0.12900</td>\n",
       "      <td>0.102000</td>\n",
       "      <td>0.315800</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>benign,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>16240358,28873462,30863299,28932933,34306325</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>7961851</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>163796</th>\n",
       "      <td>chr5_83104943_C/-</td>\n",
       "      <td>chr5:83104943</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000152422</td>\n",
       "      <td>ENST00000396027</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>103</td>\n",
       "      <td>24</td>\n",
       "      <td>8</td>\n",
       "      <td>I/X</td>\n",
       "      <td>atC/at</td>\n",
       "      <td>rs869320677</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>C</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>XRCC4</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:12831</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_003401.5</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>A1</td>\n",
       "      <td>CCDS4058.1</td>\n",
       "      <td>ENSP00000379344</td>\n",
       "      <td>Q13426.199</td>\n",
       "      <td>A0A024RAL0.63</td>\n",
       "      <td>UPI0000072B0A</td>\n",
       "      <td>Q13426-2</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2/8</td>\n",
       "      <td>-</td>\n",
       "      <td>PDB-ENSP_mappings:1fu1.A,PDB-ENSP_mappings:1fu...</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000396027.9:c.25del</td>\n",
       "      <td>ENSP00000379344.4:p.His9ThrfsTer8</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000394</td>\n",
       "      <td>0.000072</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000283</td>\n",
       "      <td>0.003165</td>\n",
       "      <td>0.000765</td>\n",
       "      <td>0.00000</td>\n",
       "      <td>0.000207</td>\n",
       "      <td>0.003165</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>likely_pathogenic,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>26822949,25728776,25839420</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr5</td>\n",
       "      <td>83104943</td>\n",
       "      <td>2.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Rare (&lt;1%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269741</th>\n",
       "      <td>chr9_27573524_GCCCCG/GCCCCGGCCCCGGCCCCGGCCCCGG...</td>\n",
       "      <td>chr9:27573524-27573529</td>\n",
       "      <td>GCCCCGGCCCCGGCCCCGGCCCCGGCCCCG</td>\n",
       "      <td>ENSG00000147894</td>\n",
       "      <td>ENST00000380003</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs143561967</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>43.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>C9orf72</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:28337</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_018325.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS6522.1</td>\n",
       "      <td>ENSP00000369339</td>\n",
       "      <td>Q96LT7.150</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI00001D3FEF</td>\n",
       "      <td>Q96LT7-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.046780</td>\n",
       "      <td>0.071110</td>\n",
       "      <td>0.004728</td>\n",
       "      <td>0.1178</td>\n",
       "      <td>0.01102</td>\n",
       "      <td>0.13300</td>\n",
       "      <td>0.009990</td>\n",
       "      <td>0.010000</td>\n",
       "      <td>0.017550</td>\n",
       "      <td>0.03858</td>\n",
       "      <td>0.040460</td>\n",
       "      <td>0.133000</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>31836585,32471232</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr9</td>\n",
       "      <td>27573524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>446035</th>\n",
       "      <td>chr17_30237267_GGGCTGCAGGGGGGATGCTGGGGGTGCAGGG...</td>\n",
       "      <td>chr17:30237267-30237309</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000108576</td>\n",
       "      <td>ENST00000650711</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs774676466</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1570.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>SLC6A4</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:11050</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001045.6</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS11256.1</td>\n",
       "      <td>ENSP00000498537</td>\n",
       "      <td>P31645.208</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI0000135493</td>\n",
       "      <td>P31645-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG</td>\n",
       "      <td>GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.283100</td>\n",
       "      <td>0.145300</td>\n",
       "      <td>0.199200</td>\n",
       "      <td>0.4085</td>\n",
       "      <td>0.42210</td>\n",
       "      <td>0.59490</td>\n",
       "      <td>0.277000</td>\n",
       "      <td>0.288500</td>\n",
       "      <td>0.304000</td>\n",
       "      <td>0.30200</td>\n",
       "      <td>0.466100</td>\n",
       "      <td>0.594900</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>11772685,12966525,12915525,15691525,11335745,1...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr17</td>\n",
       "      <td>30237267</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>17</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13321</th>\n",
       "      <td>chr1_10337509_A/G</td>\n",
       "      <td>chr1:10337509</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000054523</td>\n",
       "      <td>ENST00000676179</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3786.0</td>\n",
       "      <td>3398.0</td>\n",
       "      <td>1133.0</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>tAt/tGt</td>\n",
       "      <td>rs2297881,COSV55805285</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>A</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>KIF1B</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:16636</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001365951.3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000502065</td>\n",
       "      <td>O60333.222</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000003424C</td>\n",
       "      <td>O60333-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>deleterious_low_confidence(0)</td>\n",
       "      <td>probably_damaging(0.973)</td>\n",
       "      <td>31/49</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-O60333-F1.A,PANTHER:PTHR...</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000676179.1:c.3398A&gt;G</td>\n",
       "      <td>ENSP00000502065.1:p.Tyr1133Cys</td>\n",
       "      <td>-</td>\n",
       "      <td>0.034550</td>\n",
       "      <td>0.046030</td>\n",
       "      <td>0.042760</td>\n",
       "      <td>0.0312</td>\n",
       "      <td>0.05213</td>\n",
       "      <td>0.07377</td>\n",
       "      <td>0.050400</td>\n",
       "      <td>0.126600</td>\n",
       "      <td>0.021860</td>\n",
       "      <td>0.03820</td>\n",
       "      <td>0.026490</td>\n",
       "      <td>0.126600</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>benign,pathogenic</td>\n",
       "      <td>0,1</td>\n",
       "      <td>1,1</td>\n",
       "      <td>30126838</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>10337509</td>\n",
       "      <td>31.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>deleterious_low_confidence</td>\n",
       "      <td>0.0</td>\n",
       "      <td>probably_damaging</td>\n",
       "      <td>0.973</td>\n",
       "      <td>1_010337509_A_G</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                       Uploaded.variation                 Location                          Allele             Gene          Feature Feature.type            Consequence cDNA.position CDS.position Protein.position Amino.acids   Codons      Existing.variation      IND  ZYG                                   REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS        VARIANT.CLASS   SYMBOL SYMBOL.SOURCE     HGNC.ID         BIOTYPE CANONICAL     MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS         CCDS             ENSP   SWISSPROT         TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET                                    GIVEN.REF                                     USED.REF BAM.EDIT GENE.PHENO                           SIFT                  PolyPhen   EXON INTRON                                            DOMAINS miRNA                                  HGVSc                              HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF    MAX.AF  MAX.AF.POPS                      CLIN.SIG SOMATIC PHENO                                             PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH              SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines             MAF           MAF2  PATHO    VUS  BENIGN ClinVar\n",
       "2118                    chr1_7961851_GTGCTGGACGGTGTCCCT/-     chr1:7961851-7961868                               -  ENSG00000116288  ENST00000338639   Transcript         intron_variant           NaN          NaN              NaN           -        -             rs200968609  P732912  HET                           GTGCTGGACGGTGTCCCT  MODIFIER       NaN      1     -             deletion    PARK7          HGNC  HGNC:16369  protein_coding       YES     NM_007262.5                  -   1     P1     CCDS93.1  ENSP00000340278  Q99497.214      V9HWC2.55  UPI0000072F6F               -            -  Ensembl             -                           GTGCTGGACGGTGTCCCT                           GTGCTGGACGGTGTCCCT        -          1                              -                         -      -    1/6                                                  -     -  ENST00000338639.10:c.-24+75_-24+92del                                  -           -    0.139600        0.046980        0.045050          0.1133         0.11430         0.06402        0.315800        0.095540        0.186000         0.12900        0.102000  0.315800  gnomADg_FIN             benign,pathogenic       -     1       16240358,28873462,30863299,28932933,34306325          -         -            -                  -                     -       chr1        7961851         NaN           1.0   1             True                         NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Common (>=10%)  Common (>=5%)   True  False    True   PATHO\n",
       "163796                                  chr5_83104943_C/-            chr5:83104943                               -  ENSG00000152422  ENST00000396027   Transcript     frameshift_variant           103           24                8         I/X   atC/at             rs869320677  P732912  HET                                            C      HIGH       NaN      1     -             deletion    XRCC4          HGNC  HGNC:12831  protein_coding       YES     NM_003401.5                  -   5     A1   CCDS4058.1  ENSP00000379344  Q13426.199  A0A024RAL0.63  UPI0000072B0A        Q13426-2            -  Ensembl             -                                            C                                            C        -          1                              -                         -    2/8      -  PDB-ENSP_mappings:1fu1.A,PDB-ENSP_mappings:1fu...     -              ENST00000396027.9:c.25del  ENSP00000379344.4:p.His9ThrfsTer8           1    0.000394        0.000072        0.000000          0.0000         0.00000         0.00000        0.000283        0.003165        0.000765         0.00000        0.000207  0.003165  gnomADg_MID  likely_pathogenic,pathogenic       -     1                         26822949,25728776,25839420          -         -            -                  -                     -       chr5       83104943         2.0           NaN   5             True                         NaN        NaN                NaN            NaN              NaN              False                     True                True                     False      Rare (<1%)     Rare (<5%)   True  False   False   PATHO\n",
       "269741  chr9_27573524_GCCCCG/GCCCCGGCCCCGGCCCCGGCCCCGG...   chr9:27573524-27573529  GCCCCGGCCCCGGCCCCGGCCCCGGCCCCG  ENSG00000147894  ENST00000380003   Transcript  upstream_gene_variant           NaN          NaN              NaN           -        -             rs143561967  P732912  HET                                       GCCCCG  MODIFIER      43.0     -1     -  sequence_alteration  C9orf72          HGNC  HGNC:28337  protein_coding       YES     NM_018325.5                  -   1     P1   CCDS6522.1  ENSP00000369339  Q96LT7.150              -  UPI00001D3FEF        Q96LT7-1            -  Ensembl             -                                       GCCCCG                                       GCCCCG        -          1                              -                         -      -      -                                                  -     -                                      -                                  -           -    0.046780        0.071110        0.004728          0.1178         0.01102         0.13300        0.009990        0.010000        0.017550         0.03858        0.040460  0.133000  gnomADg_EAS                    pathogenic       -     1                                  31836585,32471232          -         -            -                  -                     -       chr9       27573524         NaN           NaN   9             True                         NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Common (>=10%)  Common (>=5%)   True  False   False   PATHO\n",
       "446035  chr17_30237267_GGGCTGCAGGGGGGATGCTGGGGGTGCAGGG...  chr17:30237267-30237309                               -  ENSG00000108576  ENST00000650711   Transcript  upstream_gene_variant           NaN          NaN              NaN           -        -             rs774676466  P732912  HET  GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG  MODIFIER    1570.0     -1     -             deletion   SLC6A4          HGNC  HGNC:11050  protein_coding       YES     NM_001045.6                  -   -     P1  CCDS11256.1  ENSP00000498537  P31645.208              -  UPI0000135493        P31645-1            -  Ensembl             -  GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG  GGGCTGCAGGGGGGATGCTGGGGGTGCAGGGGAGATGCTGGGG        -          1                              -                         -      -      -                                                  -     -                                      -                                  -           -    0.283100        0.145300        0.199200          0.4085         0.42210         0.59490        0.277000        0.288500        0.304000         0.30200        0.466100  0.594900  gnomADg_EAS                    pathogenic       -     1  11772685,12966525,12915525,15691525,11335745,1...          -         -            -                  -                     -      chr17       30237267         NaN           NaN  17             True                         NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Common (>=10%)  Common (>=5%)   True  False   False   PATHO\n",
       "13321                                   chr1_10337509_A/G            chr1:10337509                               G  ENSG00000054523  ENST00000676179   Transcript       missense_variant        3786.0       3398.0           1133.0         Y/C  tAt/tGt  rs2297881,COSV55805285  P732912  HET                                            A  MODERATE       NaN      1     -                  SNV    KIF1B          HGNC  HGNC:16636  protein_coding       YES  NM_001365951.3                  -   -     P1            -  ENSP00000502065  O60333.222              -  UPI000003424C        O60333-1            -  Ensembl             -                                            A                                            A        -          1  deleterious_low_confidence(0)  probably_damaging(0.973)  31/49      -  AFDB-ENSP_mappings:AF-O60333-F1.A,PANTHER:PTHR...     -            ENST00000676179.1:c.3398A>G     ENSP00000502065.1:p.Tyr1133Cys           -    0.034550        0.046030        0.042760          0.0312         0.05213         0.07377        0.050400        0.126600        0.021860         0.03820        0.026490  0.126600  gnomADg_MID             benign,pathogenic     0,1   1,1                                           30126838          -         -            -                  -                     -       chr1       10337509        31.0           NaN   1             True  deleterious_low_confidence        0.0  probably_damaging          0.973  1_010337509_A_G              False                     True                True                     False  Common (>=10%)  Common (>=5%)   True  False    True   PATHO"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho12.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "e0fc9763-5568-4919-886b-9b1d098ee844",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded.variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature.type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA.position</th>\n",
       "      <th>CDS.position</th>\n",
       "      <th>Protein.position</th>\n",
       "      <th>Amino.acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing.variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>REF.ALLELE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT.CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL.SOURCE</th>\n",
       "      <th>HGNC.ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE.SELECT</th>\n",
       "      <th>MANE.PLUS.CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT.ISOFORM</th>\n",
       "      <th>REFSEQ.MATCH</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>REFSEQ.OFFSET</th>\n",
       "      <th>GIVEN.REF</th>\n",
       "      <th>USED.REF</th>\n",
       "      <th>BAM.EDIT</th>\n",
       "      <th>GENE.PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVS.OFFSET</th>\n",
       "      <th>gnomADg.AF</th>\n",
       "      <th>gnomADg.AFR.AF</th>\n",
       "      <th>gnomADg.AMI.AF</th>\n",
       "      <th>gnomADg.AMR.AF</th>\n",
       "      <th>gnomADg.ASJ.AF</th>\n",
       "      <th>gnomADg.EAS.AF</th>\n",
       "      <th>gnomADg.FIN.AF</th>\n",
       "      <th>gnomADg.MID.AF</th>\n",
       "      <th>gnomADg.NFE.AF</th>\n",
       "      <th>gnomADg.OTH.AF</th>\n",
       "      <th>gnomADg.SAS.AF</th>\n",
       "      <th>MAX.AF</th>\n",
       "      <th>MAX.AF.POPS</th>\n",
       "      <th>CLIN.SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF.NAME</th>\n",
       "      <th>MOTIF.POS</th>\n",
       "      <th>HIGH.INF.POS</th>\n",
       "      <th>MOTIF.SCORE.CHANGE</th>\n",
       "      <th>TRANSCRIPTION.FACTORS</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chrCoordinate</th>\n",
       "      <th>EXONnumber</th>\n",
       "      <th>INTRONnumber</th>\n",
       "      <th>chr</th>\n",
       "      <th>GIVEN.REF.MATCH</th>\n",
       "      <th>SIFTprediction</th>\n",
       "      <th>SIFTscore</th>\n",
       "      <th>PolyPhenprediction</th>\n",
       "      <th>PolyPhenscore</th>\n",
       "      <th>keyID38</th>\n",
       "      <th>var.in.both.lines</th>\n",
       "      <th>var.exclusive.btw.lines</th>\n",
       "      <th>gene.in.both.lines</th>\n",
       "      <th>gene.exclusive.btw.lines</th>\n",
       "      <th>MAF</th>\n",
       "      <th>MAF2</th>\n",
       "      <th>PATHO</th>\n",
       "      <th>VUS</th>\n",
       "      <th>BENIGN</th>\n",
       "      <th>ClinVar</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>56517</th>\n",
       "      <td>chr2_48755617_-/GCTGCA</td>\n",
       "      <td>chr2:48755616-48755617</td>\n",
       "      <td>GCTGCA</td>\n",
       "      <td>ENSG00000138039</td>\n",
       "      <td>ENST00000294954</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>inframe_insertion</td>\n",
       "      <td>108-109</td>\n",
       "      <td>55-56</td>\n",
       "      <td>19</td>\n",
       "      <td>P/LQP</td>\n",
       "      <td>ccg/cTGCAGCcg</td>\n",
       "      <td>rs71245621</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HOM</td>\n",
       "      <td>-</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>LHCGR</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:6585</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_000233.4</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P2</td>\n",
       "      <td>CCDS1842.1</td>\n",
       "      <td>ENSP00000294954</td>\n",
       "      <td>P22888.225</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000013E1E1</td>\n",
       "      <td>P22888-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/11</td>\n",
       "      <td>-</td>\n",
       "      <td>AFDB-ENSP_mappings:AF-P22888-F1.A,PANTHER:PTHR...</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000294954.12:c.50_55dup</td>\n",
       "      <td>ENSP00000294954.6:p.Leu17_Gln18dup</td>\n",
       "      <td>-</td>\n",
       "      <td>0.219200</td>\n",
       "      <td>0.187100</td>\n",
       "      <td>0.358100</td>\n",
       "      <td>0.185500</td>\n",
       "      <td>0.300500</td>\n",
       "      <td>0.011390</td>\n",
       "      <td>0.188100</td>\n",
       "      <td>0.2930</td>\n",
       "      <td>0.267200</td>\n",
       "      <td>0.240700</td>\n",
       "      <td>0.118600</td>\n",
       "      <td>0.35810</td>\n",
       "      <td>gnomADg_AMI</td>\n",
       "      <td>benign,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>12679452,9851790,7556872,2244890</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr2</td>\n",
       "      <td>48755616</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>284152</th>\n",
       "      <td>chr9_27573524_GCCCCG/-</td>\n",
       "      <td>chr9:27573524-27573529</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000147894</td>\n",
       "      <td>ENST00000380003</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs143561967</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HOM</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>43.0</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>C9orf72</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:28337</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_018325.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS6522.1</td>\n",
       "      <td>ENSP00000369339</td>\n",
       "      <td>Q96LT7.150</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI00001D3FEF</td>\n",
       "      <td>Q96LT7-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>GCCCCG</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.532600</td>\n",
       "      <td>0.441400</td>\n",
       "      <td>0.503600</td>\n",
       "      <td>0.547400</td>\n",
       "      <td>0.657700</td>\n",
       "      <td>0.610300</td>\n",
       "      <td>0.635300</td>\n",
       "      <td>0.4933</td>\n",
       "      <td>0.557600</td>\n",
       "      <td>0.544700</td>\n",
       "      <td>0.556100</td>\n",
       "      <td>0.65770</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>31836585,32471232</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr9</td>\n",
       "      <td>27573524</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>9</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345567</th>\n",
       "      <td>chr11_68774782_A/-</td>\n",
       "      <td>chr11:68774782</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000110090</td>\n",
       "      <td>ENST00000265641</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1169875761</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>CPT1A</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:2328</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001876.4</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS8185.1</td>\n",
       "      <td>ENSP00000265641</td>\n",
       "      <td>P50416.203</td>\n",
       "      <td>A0A024R5F4.59</td>\n",
       "      <td>UPI000013D658</td>\n",
       "      <td>P50416-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>13/18</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000265641.10:c.1575+534del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.002580</td>\n",
       "      <td>0.000886</td>\n",
       "      <td>0.002732</td>\n",
       "      <td>0.003820</td>\n",
       "      <td>0.001892</td>\n",
       "      <td>0.000991</td>\n",
       "      <td>0.013790</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.002541</td>\n",
       "      <td>0.001449</td>\n",
       "      <td>0.002844</td>\n",
       "      <td>0.01379</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>uncertain_significance,pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>12189492</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr11</td>\n",
       "      <td>68774782</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13.0</td>\n",
       "      <td>11</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>377988</th>\n",
       "      <td>chr12_101765329_-/T</td>\n",
       "      <td>chr12:101765328-101765329</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000111670</td>\n",
       "      <td>ENST00000299314</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs546802775</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>GNPTAB</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:29670</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_024312.5</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>P1</td>\n",
       "      <td>CCDS9088.1</td>\n",
       "      <td>ENSP00000299314</td>\n",
       "      <td>Q3T906.151</td>\n",
       "      <td>-</td>\n",
       "      <td>UPI000004CC0C</td>\n",
       "      <td>Q3T906-1</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>12/20</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000299314.12:c.1613-25dup</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.002679</td>\n",
       "      <td>0.000708</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.000663</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>0.001548</td>\n",
       "      <td>0.006169</td>\n",
       "      <td>0.0000</td>\n",
       "      <td>0.004177</td>\n",
       "      <td>0.001461</td>\n",
       "      <td>0.001681</td>\n",
       "      <td>0.01207</td>\n",
       "      <td>gnomADe_FIN</td>\n",
       "      <td>likely_benign,likely_pathogenic</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr12</td>\n",
       "      <td>101765328</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12.0</td>\n",
       "      <td>12</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Less common (&lt;10%)</td>\n",
       "      <td>Rare (&lt;5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13912</th>\n",
       "      <td>chr1_11794419_T/G</td>\n",
       "      <td>chr1:11794419</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000215910</td>\n",
       "      <td>ENST00000688073</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1801131,CM981315,COSV57171183</td>\n",
       "      <td>P931263</td>\n",
       "      <td>HET</td>\n",
       "      <td>T</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>4834.0</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>SNV</td>\n",
       "      <td>C1orf167</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:25262</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>NM_001010881.2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P2</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000510540</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A8I5KXP5.2</td>\n",
       "      <td>UPI000387BF29</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.263400</td>\n",
       "      <td>0.167000</td>\n",
       "      <td>0.219300</td>\n",
       "      <td>0.207000</td>\n",
       "      <td>0.297000</td>\n",
       "      <td>0.220500</td>\n",
       "      <td>0.327800</td>\n",
       "      <td>0.3662</td>\n",
       "      <td>0.315900</td>\n",
       "      <td>0.270900</td>\n",
       "      <td>0.406800</td>\n",
       "      <td>0.41720</td>\n",
       "      <td>SAS</td>\n",
       "      <td>risk_factor,conflicting_interpretations_of_pat...</td>\n",
       "      <td>0,0,1</td>\n",
       "      <td>1,1,1</td>\n",
       "      <td>21467728,18603647,21146954,21533139,21919968,2...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>chr1</td>\n",
       "      <td>11794419</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>True</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1_011794419_T_G</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>True</td>\n",
       "      <td>False</td>\n",
       "      <td>Common (&gt;=10%)</td>\n",
       "      <td>Common (&gt;=5%)</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>PATHO</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Uploaded.variation                   Location  Allele             Gene          Feature Feature.type              Consequence cDNA.position CDS.position Protein.position Amino.acids         Codons               Existing.variation      IND  ZYG REF.ALLELE    IMPACT  DISTANCE STRAND FLAGS VARIANT.CLASS    SYMBOL SYMBOL.SOURCE     HGNC.ID         BIOTYPE CANONICAL     MANE.SELECT MANE.PLUS.CLINICAL TSL APPRIS        CCDS             ENSP   SWISSPROT         TREMBL        UNIPARC UNIPROT.ISOFORM REFSEQ.MATCH   SOURCE REFSEQ.OFFSET GIVEN.REF USED.REF BAM.EDIT GENE.PHENO SIFT PolyPhen  EXON INTRON                                            DOMAINS miRNA                             HGVSc                               HGVSp HGVS.OFFSET  gnomADg.AF  gnomADg.AFR.AF  gnomADg.AMI.AF  gnomADg.AMR.AF  gnomADg.ASJ.AF  gnomADg.EAS.AF  gnomADg.FIN.AF  gnomADg.MID.AF  gnomADg.NFE.AF  gnomADg.OTH.AF  gnomADg.SAS.AF   MAX.AF  MAX.AF.POPS                                           CLIN.SIG SOMATIC  PHENO                                             PUBMED MOTIF.NAME MOTIF.POS HIGH.INF.POS MOTIF.SCORE.CHANGE TRANSCRIPTION.FACTORS chromosome  chrCoordinate  EXONnumber  INTRONnumber chr  GIVEN.REF.MATCH SIFTprediction  SIFTscore PolyPhenprediction  PolyPhenscore          keyID38  var.in.both.lines  var.exclusive.btw.lines  gene.in.both.lines  gene.exclusive.btw.lines                 MAF           MAF2  PATHO    VUS  BENIGN ClinVar\n",
       "56517   chr2_48755617_-/GCTGCA     chr2:48755616-48755617  GCTGCA  ENSG00000138039  ENST00000294954   Transcript        inframe_insertion       108-109        55-56               19       P/LQP  ccg/cTGCAGCcg                       rs71245621  P931263  HOM          -  MODERATE       NaN     -1     -     insertion     LHCGR          HGNC   HGNC:6585  protein_coding       YES     NM_000233.4                  -   1     P2  CCDS1842.1  ENSP00000294954  P22888.225              -  UPI000013E1E1        P22888-1            -  Ensembl             -         -        -        -          1    -        -  1/11      -  AFDB-ENSP_mappings:AF-P22888-F1.A,PANTHER:PTHR...     -     ENST00000294954.12:c.50_55dup  ENSP00000294954.6:p.Leu17_Gln18dup           -    0.219200        0.187100        0.358100        0.185500        0.300500        0.011390        0.188100          0.2930        0.267200        0.240700        0.118600  0.35810  gnomADg_AMI                                  benign,pathogenic       -      1                   12679452,9851790,7556872,2244890          -         -            -                  -                     -       chr2       48755616         1.0           NaN   2             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False      Common (>=10%)  Common (>=5%)   True  False    True   PATHO\n",
       "284152  chr9_27573524_GCCCCG/-     chr9:27573524-27573529       -  ENSG00000147894  ENST00000380003   Transcript    upstream_gene_variant           NaN          NaN              NaN           -              -                      rs143561967  P931263  HOM     GCCCCG  MODIFIER      43.0     -1     -      deletion   C9orf72          HGNC  HGNC:28337  protein_coding       YES     NM_018325.5                  -   1     P1  CCDS6522.1  ENSP00000369339  Q96LT7.150              -  UPI00001D3FEF        Q96LT7-1            -  Ensembl             -    GCCCCG   GCCCCG        -          1    -        -     -      -                                                  -     -                                 -                                   -           -    0.532600        0.441400        0.503600        0.547400        0.657700        0.610300        0.635300          0.4933        0.557600        0.544700        0.556100  0.65770  gnomADg_ASJ                                         pathogenic       -      1                                  31836585,32471232          -         -            -                  -                     -       chr9       27573524         NaN           NaN   9             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False      Common (>=10%)  Common (>=5%)   True  False   False   PATHO\n",
       "345567      chr11_68774782_A/-             chr11:68774782       -  ENSG00000110090  ENST00000265641   Transcript           intron_variant           NaN          NaN              NaN           -              -                     rs1169875761  P931263  HET          A  MODIFIER       NaN     -1     -      deletion     CPT1A          HGNC   HGNC:2328  protein_coding       YES     NM_001876.4                  -   1     P1  CCDS8185.1  ENSP00000265641  P50416.203  A0A024R5F4.59  UPI000013D658        P50416-1            -  Ensembl             -         A        A        -          1    -        -     -  13/18                                                  -     -  ENST00000265641.10:c.1575+534del                                   -           -    0.002580        0.000886        0.002732        0.003820        0.001892        0.000991        0.013790          0.0000        0.002541        0.001449        0.002844  0.01379  gnomADg_FIN                  uncertain_significance,pathogenic       -      1                                           12189492          -         -            -                  -                     -      chr11       68774782         NaN          13.0  11             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Less common (<10%)     Rare (<5%)   True   True   False   PATHO\n",
       "377988     chr12_101765329_-/T  chr12:101765328-101765329       T  ENSG00000111670  ENST00000299314   Transcript           intron_variant           NaN          NaN              NaN           -              -                      rs546802775  P931263  HET          -  MODIFIER       NaN     -1     -     insertion    GNPTAB          HGNC  HGNC:29670  protein_coding       YES     NM_024312.5                  -   1     P1  CCDS9088.1  ENSP00000299314  Q3T906.151              -  UPI000004CC0C        Q3T906-1            -  Ensembl             -         -        -        -          1    -        -     -  12/20                                                  -     -   ENST00000299314.12:c.1613-25dup                                   -           -    0.002679        0.000708        0.000000        0.000663        0.000000        0.001548        0.006169          0.0000        0.004177        0.001461        0.001681  0.01207  gnomADe_FIN                    likely_benign,likely_pathogenic       -      1                                                  -          -         -            -                  -                     -      chr12      101765328         NaN          12.0  12             True            NaN        NaN                NaN            NaN              NaN              False                     True                True                     False  Less common (<10%)     Rare (<5%)   True  False    True   PATHO\n",
       "13912        chr1_11794419_T/G              chr1:11794419       G  ENSG00000215910  ENST00000688073   Transcript  downstream_gene_variant           NaN          NaN              NaN           -              -  rs1801131,CM981315,COSV57171183  P931263  HET          T  MODIFIER    4834.0      1     -           SNV  C1orf167          HGNC  HGNC:25262  protein_coding       YES  NM_001010881.2                  -   -     P2           -  ENSP00000510540           -   A0A8I5KXP5.2  UPI000387BF29               -            -  Ensembl             -         T        T        -          -    -        -     -      -                                                  -     -                                 -                                   -           -    0.263400        0.167000        0.219300        0.207000        0.297000        0.220500        0.327800          0.3662        0.315900        0.270900        0.406800  0.41720          SAS  risk_factor,conflicting_interpretations_of_pat...   0,0,1  1,1,1  21467728,18603647,21146954,21533139,21919968,2...          -         -            -                  -                     -       chr1       11794419         NaN           NaN   1             True              -        NaN                  -            NaN  1_011794419_T_G               True                    False                True                     False      Common (>=10%)  Common (>=5%)   True   True    True   PATHO"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "patho63.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "4e6a8e9b-7f39-4917-9c85-e990436f6cf2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3428264</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>35778</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>32158</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ClinVar</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3428264\n",
       "1      No DB source    35778\n",
       "2  gnomAD & ClinVar    32158\n",
       "3           ClinVar        1"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# create a new column 'Database' in both dataframes, if MAX.AF is not NA and ClinVar is NA label 'gnomAD', if ClinVar is not NA and MAX.AF is NA label 'ClinVar', if both are not NA label 'gnomAD & ClinVar', else label 'No DB source'\n",
    "cad12['Database'] = np.select([cad12['ClinVar'].notna() & cad12['MAX.AF'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].isna(), \n",
    "                                  cad12['MAX.AF'].notna() & cad12['ClinVar'].notna(), \n",
    "                                  cad12['MAX.AF'].isna() & cad12['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "checkColumnValue(cad12, 'Database')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "c6c4c702-3708-4851-844a-4734c184cfe6",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Count</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>gnomAD</td>\n",
       "      <td>3641404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>gnomAD &amp; ClinVar</td>\n",
       "      <td>33479</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>No DB source</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "              Count    count\n",
       "0            gnomAD  3641404\n",
       "1  gnomAD & ClinVar    33479\n",
       "2      No DB source    26131"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cad63['Database'] = np.select([cad63['ClinVar'].notna() & cad63['MAX.AF'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].isna(), \n",
    "                                  cad63['MAX.AF'].notna() & cad63['ClinVar'].notna(), \n",
    "                                  cad63['MAX.AF'].isna() & cad63['ClinVar'].isna()], \n",
    "                                  ['ClinVar', 'gnomAD', 'gnomAD & ClinVar', 'No DB source'], \n",
    "                                  default = np.nan)\n",
    "checkColumnValue(cad63, 'Database')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "6d981587-2c8f-4260-9d79-d74c58e149d2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(3496201, 96)\n",
      "(3701014, 96)\n"
     ]
    }
   ],
   "source": [
    "print(cad12.shape)\n",
    "print(cad63.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "1bbb6b7b-b4ff-4388-b8fb-116da0b3eb04",
   "metadata": {},
   "outputs": [],
   "source": [
    "cad12.to_csv('4-cad12-snv-indel-3496201.csv', index=False)\n",
    "cad63.to_csv('4-cad63-snv-indel-3701014.csv', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "5aff0a90-b5f3-41ba-8b42-87d4f8339642",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>ClinVar</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Common (&gt;=10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3221872</td>\n",
       "      <td>31160</td>\n",
       "      <td>3253032</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Less common (&lt;10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>152004</td>\n",
       "      <td>856</td>\n",
       "      <td>152860</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rare (&lt;1%)</th>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>54388</td>\n",
       "      <td>142</td>\n",
       "      <td>54530</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nan</th>\n",
       "      <td>1</td>\n",
       "      <td>35778</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>35779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>1</td>\n",
       "      <td>35778</td>\n",
       "      <td>3428264</td>\n",
       "      <td>32158</td>\n",
       "      <td>3496201</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database            ClinVar  No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "MAF                                                                          \n",
       "Common (>=10%)            0             0  3221872             31160  3253032\n",
       "Less common (<10%)        0             0   152004               856   152860\n",
       "Rare (<1%)                0             0    54388               142    54530\n",
       "nan                       1         35778        0                 0    35779\n",
       "Total                     1         35778  3428264             32158  3496201"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad12, 'MAF', 'Database', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "2fdbf9db-5507-4ad4-ab00-c1f38706a240",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>Database</th>\n",
       "      <th>No DB source</th>\n",
       "      <th>gnomAD</th>\n",
       "      <th>gnomAD &amp; ClinVar</th>\n",
       "      <th>Total</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>MAF</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Common (&gt;=10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>3430130</td>\n",
       "      <td>32339</td>\n",
       "      <td>3462469</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Less common (&lt;10%)</th>\n",
       "      <td>0</td>\n",
       "      <td>156632</td>\n",
       "      <td>992</td>\n",
       "      <td>157624</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Rare (&lt;1%)</th>\n",
       "      <td>0</td>\n",
       "      <td>54642</td>\n",
       "      <td>148</td>\n",
       "      <td>54790</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>nan</th>\n",
       "      <td>26131</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>26131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Total</th>\n",
       "      <td>26131</td>\n",
       "      <td>3641404</td>\n",
       "      <td>33479</td>\n",
       "      <td>3701014</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Database            No DB source   gnomAD  gnomAD & ClinVar    Total\n",
       "MAF                                                                 \n",
       "Common (>=10%)                 0  3430130             32339  3462469\n",
       "Less common (<10%)             0   156632               992   157624\n",
       "Rare (<1%)                     0    54642               148    54790\n",
       "nan                        26131        0                 0    26131\n",
       "Total                      26131  3641404             33479  3701014"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "crosstabit(cad63, 'MAF', 'Database', True)\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "# TODO \n",
    "# 1. note the clinvar col with labels is not star reviewed, check the clin.sig column to confirm high confidence pathogenicity\n",
    "# 2. added Database column here instead of in R to keep all label functions together in python"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'checkColumnValue' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[1], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m \u001b[43mcheckColumnValue\u001b[49m(cad12, \u001b[38;5;124m'\u001b[39m\u001b[38;5;124mvar.in.both.lines\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "\u001b[0;31mNameError\u001b[0m: name 'checkColumnValue' is not defined"
     ]
    }
   ],
   "source": [
    "checkColumnValue(cad12, 'var.in.both.lines')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py39",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.19"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
