note: indented columns were dropped at the start of script
4-combine-label-line-variants.py to reduce the file size
but many are of interest for closer investigation.
i should check overlap and change column names and save
then drop column in final step before R


'Uploaded_variation'
       'Location'
       'Allele'
       'Gene'
       'Feature'
 'Feature_type'
 'Consequence'
       'cDNA_position'
       'CDS_position'
 'Protein_position'
 'Amino_acids'
       'Codons'
 'Existing_variation'
 'IND'
 'ZYG'
       'REF_ALLELE'
 'IMPACT'
 'DISTANCE' 
 'STRAND'
       'FLAGS'
 'VARIANT_CLASS'
 'SYMBOL'
       'SYMBOL_SOURCE'
       'HGNC_ID'
 'BIOTYPE'
       'CANONICAL'
       'MANE_SELECT'
       'MANE_PLUS_CLINICAL'
        'TSL'
       'APPRIS'
        'CCDS'
       'ENSP'
 'SWISSPROT'
       'TREMBL'
       'UNIPARC'
       'UNIPROT_ISOFORM'
        'REFSEQ_MATCH'
        'SOURCE'
       'REFSEQ_OFFSET'
       'GIVEN_REF'
       'USED_REF'
        'BAM_EDIT'
 'GENE_PHENO'
       'SIFT'
       'PolyPhen'
 'EXON'
 'INTRON'
 'DOMAINS'
 'miRNA'
       'HGVSc'
       'HGVSp'
       'HGVS_OFFSET'
 'gnomADg_AF'
 'gnomADg_AFR_AF'
 'gnomADg_AMI_AF'
 'gnomADg_AMR_AF'
 'gnomADg_ASJ_AF'
 'gnomADg_EAS_AF'
 'gnomADg_FIN_AF'
 'gnomADg_MID_AF'
 'gnomADg_NFE_AF'
 'gnomADg_OTH_AF'
 'gnomADg_SAS_AF'
 'MAX_AF'
 'MAX_AF_POPS'
 'CLIN_SIG'
       'SOMATIC'
        'PHENO'
 'PUBMED'
 'MOTIF_NAME'
       'MOTIF_POS'
 'HIGH_INF_POS'
 'MOTIF_SCORE_CHANGE'
'TRANSCRIPTION_FACTORS'
        'chromosome'
        'chrCoordinate'
'SIFTprediction'
'SIFTscore'
'PolyPhenprediction'
'PolyPhenscore'
'EXONnumber'
'INTRONnumber'
        'chr'
        'keyID38'
       'GIVEN_REF_MATCH'