
"""
> str(wgs)
'data.frame':   7197215 obs. of  36 variables:
 $ Uploaded.variation      : chr  "chr1_14398_TGT/-" "chr1_20317_A/-" "chr1_104161_-/ACAC/ACACACAC" "chr1_133707_AGA/-" ...
 $ Feature.type            : chr  "Transcript" "Transcript" "Transcript" "Transcript" ...
 $ chr                     : chr  "1" "1" "1" "1" ...
 $ SYMBOL                  : chr  "DDX11L2" "WASH7P" "-" "-" ...
 $ HGVSc                   : chr  "ENST00000456328.2:n.1648_1650del" "NR_024540.1:n.205-1951del" "ENST00000477740.5:n.480+8536_480+8539dup" "-" ...
 $ HGVSp                   : chr  "-" "-" "-" "-" ...
 $ var.in.both.lines       : chr  "False" "False" "False" "False" ...
 $ var.exclusive.btw.lines : chr  "True" "True" "True" "True" ...
 $ gene.in.both.lines      : chr  "True" "False" "True" "True" ...
 $ gene.exclusive.btw.lines: chr  "False" "True" "False" "False" ...
 $ IMPACT                  : chr  "MODIFIER" "MODIFIER" "MODIFIER" "MODIFIER" ...
 $ VARIANT.CLASS           : chr  "deletion" "deletion" "insertion" "deletion" ...
 $ SIFTprediction          : chr  NA NA NA NA ...
 $ SIFTscore               : num  NA NA NA NA NA NA NA NA NA NA ...
 $ PolyPhenprediction      : chr  NA NA NA NA ...
 $ PolyPhenscore           : num  NA NA NA NA NA NA NA NA NA NA ...
 $ MAX.AF                  : num  0.0364 0.3694 0.4661 0.0109 0.0357 ...
 $ MAX.AF.POPS             : chr  "gnomADg_MID" "gnomADg_FIN" "gnomADg_EAS" "gnomADg_ASJ" ...
 $ DOMAINS                 : chr  "-" "-" "-" "-" ...
 $ miRNA                   : chr  "-" "-" "-" "-" ...
 $ Consequence             : chr  "non_coding_transcript_exon_variant" "intron_variant,non_coding_transcript_variant" "intron_variant,non_coding_transcript_variant" "upstream_gene_variant" ...
 $ Amino.acids             : chr  "-" "-" "-" "-" ...
 $ CLIN.SIG                : chr  "-" "-" "-" "-" ...
 $ Existing.variation      : chr  "rs756427959" "rs1320311790" "rs372078516" "rs1354754438" ...
 $ SOMATIC                 : chr  "-" "-" "-" "-" ...
 $ PHENO                   : chr  "-" "-" "-" "-" ...
 $ PUBMED                  : chr  "-" "-" "-" "-" ...
 $ IND                     : chr  "P732912" "P732912" "P732912" "P732912" ...
 $ ZYG                     : chr  "HET" "HET" "HET" "HET" ...
 $ BIOTYPE                 : chr  "lncRNA" "transcribed_pseudogene" "lncRNA" "lncRNA" ...
 $ EXON                    : chr  "3/3" "-" "-" "-" ...
 $ TRANSCRIPTION.FACTORS   : chr  "-" "-" "-" "-" ...
 $ PATHO                   : chr  "False" "False" "False" "False" ...
 $ VUS                     : chr  "False" "False" "False" "False" ...
 $ BENIGN                  : chr  "False" "False" "False" "False" ...
 $ ClinVar                 : chr  NA NA NA NA ...
>
> names(wgs)
 [1] "Uploaded.variation"       "Feature.type"             "chr"                      "SYMBOL"
 [5] "HGVSc"                    "HGVSp"                    "var.in.both.lines"        "var.exclusive.btw.lines"
 [9] "gene.in.both.lines"       "gene.exclusive.btw.lines" "IMPACT"                   "VARIANT.CLASS"
[13] "SIFTprediction"           "SIFTscore"                "PolyPhenprediction"       "PolyPhenscore"
[17] "MAX.AF"                   "MAX.AF.POPS"              "DOMAINS"                  "miRNA"
[21] "Consequence"              "Amino.acids"              "CLIN.SIG"                 "Existing.variation"
[25] "SOMATIC"                  "PHENO"                    "PUBMED"                   "IND"
[29] "ZYG"                      "BIOTYPE"                  "EXON"                     "TRANSCRIPTION.FACTORS"
[33] "PATHO"                    "VUS"                      "BENIGN"                   "ClinVar"
>
"""
