{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "35d259ab-815c-4e9c-ae80-f1fa193efad8",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Index(['#Uploaded_variation', 'Location', 'Allele', 'Gene', 'Feature', 'Feature_type', 'Consequence', 'cDNA_position', 'CDS_position', 'Protein_position', 'Amino_acids', 'Codons', 'Existing_variation', 'IND', 'ZYG', 'REF_ALLELE', 'IMPACT', 'DISTANCE', 'STRAND', 'FLAGS', 'VARIANT_CLASS', 'SYMBOL', 'SYMBOL_SOURCE', 'HGNC_ID', 'BIOTYPE', 'CANONICAL', 'MANE_SELECT', 'MANE_PLUS_CLINICAL', 'TSL', 'APPRIS', 'CCDS', 'ENSP', 'SWISSPROT', 'TREMBL', 'UNIPARC', 'UNIPROT_ISOFORM', 'REFSEQ_MATCH', 'SOURCE', 'REFSEQ_OFFSET', 'GIVEN_REF', 'USED_REF', 'BAM_EDIT', 'GENE_PHENO', 'SIFT', 'PolyPhen', 'EXON', 'INTRON', 'DOMAINS', 'miRNA', 'HGVSc', 'HGVSp', 'HGVS_OFFSET', 'gnomADg_AF', 'gnomADg_AFR_AF', 'gnomADg_AMI_AF', 'gnomADg_AMR_AF', 'gnomADg_ASJ_AF', 'gnomADg_EAS_AF', 'gnomADg_FIN_AF', 'gnomADg_MID_AF', 'gnomADg_NFE_AF', 'gnomADg_OTH_AF', 'gnomADg_SAS_AF', 'MAX_AF', 'MAX_AF_POPS', 'CLIN_SIG', 'SOMATIC', 'PHENO', 'PUBMED', 'MOTIF_NAME', 'MOTIF_POS', 'HIGH_INF_POS', 'MOTIF_SCORE_CHANGE',\n",
      "       'TRANSCRIPTION_FACTORS'],\n",
      "      dtype='object')\n"
     ]
    }
   ],
   "source": [
    "from all_funx import *\n",
    "\n",
    "\n",
    "# input: individual WGS results for indels\n",
    "# output: vep html equivalent plots for report\n",
    "\n",
    "indel12 = \"annotated_indels_CAD12_autoX.txt\"\n",
    "snp12 = \"annotated_snps_CAD12_autoX.txt\"\n",
    "\n",
    "indel63 = \"annotated_indels_CAD63_autoX.txt\"\n",
    "snp63 = \"annotated_snps_CAD63_autoX.txt\"\n",
    "\n",
    "indel12 = pd.read_csv(indel12, delimiter=\"\\t\", engine='python',\\\n",
    "                 comment='##')\n",
    "# snp12 = pd.read_csv(snp12, delimiter=\"\\t\", engine='python',\\\n",
    "#                  comment='##')\n",
    "# indel63 = pd.read_csv(indel63, delimiter=\"\\t\", engine='python',\\\n",
    "#                       comment='##')\n",
    "# snp63 = pd.read_csv(snp63, delimiter=\"\\t\", engine='python',\\\n",
    "#                         comment='##')\n",
    "\n",
    "print(indel12.columns)\n",
    "\n",
    "# uniqueCount(indel12,)\n",
    "\n",
    "# checkColumnValues(indel12, \"Consequence\")\n",
    "# checkColumnValues(indel12, \"IMPACT\")\n",
    "# checkColumnValues(indel12, \"ZYG\")\n",
    "\n",
    "\n",
    "# #df = pd.read_csv(path, delimiter=\"\\t\", low_memory=False)\n",
    "# print(df.head())\n",
    "# df.describe()\n",
    "\n",
    "\n",
    "#crosstabit(df,\"Consequence\", \"ZYG\", True)\n",
    "# crosstabit(df,\"Consequence\", \"ZYG\", True)\n",
    "# crosstabit(snp12,\"Consequence\", \"IMPACT\", True)\n",
    "# crosstabit(snp63,\"Consequence\", \"IMPACT\", True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>#Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>IND</th>\n",
       "      <th>ZYG</th>\n",
       "      <th>REF_ALLELE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND</th>\n",
       "      <th>FLAGS</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>MANE_SELECT</th>\n",
       "      <th>MANE_PLUS_CLINICAL</th>\n",
       "      <th>TSL</th>\n",
       "      <th>APPRIS</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>UNIPROT_ISOFORM</th>\n",
       "      <th>REFSEQ_MATCH</th>\n",
       "      <th>SOURCE</th>\n",
       "      <th>REFSEQ_OFFSET</th>\n",
       "      <th>GIVEN_REF</th>\n",
       "      <th>USED_REF</th>\n",
       "      <th>BAM_EDIT</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>miRNA</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>gnomADg_AF</th>\n",
       "      <th>gnomADg_AFR_AF</th>\n",
       "      <th>gnomADg_AMI_AF</th>\n",
       "      <th>gnomADg_AMR_AF</th>\n",
       "      <th>gnomADg_ASJ_AF</th>\n",
       "      <th>gnomADg_EAS_AF</th>\n",
       "      <th>gnomADg_FIN_AF</th>\n",
       "      <th>gnomADg_MID_AF</th>\n",
       "      <th>gnomADg_NFE_AF</th>\n",
       "      <th>gnomADg_OTH_AF</th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>TRANSCRIPTION_FACTORS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr1_14398_TGT/-</td>\n",
       "      <td>chr1:14398-14400</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000290825</td>\n",
       "      <td>ENST00000456328</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>non_coding_transcript_exon_variant</td>\n",
       "      <td>1646-1648</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs756427959</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>TGT</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>DDX11L2</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>TGT</td>\n",
       "      <td>TGT</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000456328.2:n.1648_1650del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.01817</td>\n",
       "      <td>0.002934</td>\n",
       "      <td>0.01613</td>\n",
       "      <td>0.02403</td>\n",
       "      <td>0.03436</td>\n",
       "      <td>0.002087</td>\n",
       "      <td>0.02536</td>\n",
       "      <td>0.03636</td>\n",
       "      <td>0.03001</td>\n",
       "      <td>0.01686</td>\n",
       "      <td>0.01924</td>\n",
       "      <td>0.03636</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr1_20317_A/-</td>\n",
       "      <td>chr1:20317</td>\n",
       "      <td>-</td>\n",
       "      <td>653635</td>\n",
       "      <td>NR_024540.1</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1320311790</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH7P</td>\n",
       "      <td>EntrezGene</td>\n",
       "      <td>HGNC:38034</td>\n",
       "      <td>transcribed_pseudogene</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>OK</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2/10</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>NR_024540.1:n.205-1951del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.2744</td>\n",
       "      <td>0.1077</td>\n",
       "      <td>0.2891</td>\n",
       "      <td>0.3073</td>\n",
       "      <td>0.3022</td>\n",
       "      <td>0.347</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>0.3142</td>\n",
       "      <td>0.3461</td>\n",
       "      <td>0.2865</td>\n",
       "      <td>0.2727</td>\n",
       "      <td>0.3694</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr1_104161_-/ACAC/ACACACAC</td>\n",
       "      <td>chr1:104160-104161</td>\n",
       "      <td>ACAC</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENST00000477740</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant,non_coding_transcript_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs372078516</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3/3</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000477740.5:n.480+8536_480+8539dup</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.3246</td>\n",
       "      <td>0.2267</td>\n",
       "      <td>0.297</td>\n",
       "      <td>0.3429</td>\n",
       "      <td>0.3627</td>\n",
       "      <td>0.4661</td>\n",
       "      <td>0.2832</td>\n",
       "      <td>0.4492</td>\n",
       "      <td>0.3618</td>\n",
       "      <td>0.3337</td>\n",
       "      <td>0.4206</td>\n",
       "      <td>0.4661</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr1_133707_AGA/-</td>\n",
       "      <td>chr1:133707-133709</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000238009</td>\n",
       "      <td>ENST00000477740</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1354754438</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>AGA</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>4490</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>5</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>AGA</td>\n",
       "      <td>AGA</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.00245</td>\n",
       "      <td>0.0004914</td>\n",
       "      <td>0</td>\n",
       "      <td>0.001503</td>\n",
       "      <td>0.01094</td>\n",
       "      <td>0.01074</td>\n",
       "      <td>9.868e-05</td>\n",
       "      <td>0.003401</td>\n",
       "      <td>0.002703</td>\n",
       "      <td>0.003115</td>\n",
       "      <td>0.008426</td>\n",
       "      <td>0.01094</td>\n",
       "      <td>gnomADg_ASJ</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...</td>\n",
       "      <td>chr1:136603-136604</td>\n",
       "      <td>GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGTGTGAGGCAAGGGCTCG</td>\n",
       "      <td>ENSG00000239906</td>\n",
       "      <td>ENST00000493797</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1557463817</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>-</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>3186</td>\n",
       "      <td>-1</td>\n",
       "      <td>-</td>\n",
       "      <td>insertion</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>2</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02026</td>\n",
       "      <td>0.02241</td>\n",
       "      <td>0</td>\n",
       "      <td>0.02184</td>\n",
       "      <td>0.02338</td>\n",
       "      <td>0.01248</td>\n",
       "      <td>0.01727</td>\n",
       "      <td>0.03571</td>\n",
       "      <td>0.02032</td>\n",
       "      <td>0.008929</td>\n",
       "      <td>0.0132</td>\n",
       "      <td>0.03571</td>\n",
       "      <td>gnomADg_MID</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546099</th>\n",
       "      <td>chrX_156020867_G/-</td>\n",
       "      <td>chrX:156020867</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs748173079</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>G</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>94</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02063</td>\n",
       "      <td>0.004833</td>\n",
       "      <td>0.009934</td>\n",
       "      <td>0.02908</td>\n",
       "      <td>0.01323</td>\n",
       "      <td>0.03532</td>\n",
       "      <td>0.01851</td>\n",
       "      <td>0.02193</td>\n",
       "      <td>0.02955</td>\n",
       "      <td>0.02039</td>\n",
       "      <td>0.02675</td>\n",
       "      <td>0.03532</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546100</th>\n",
       "      <td>chrX_156021625_AGG/-</td>\n",
       "      <td>chrX:156021625-156021627</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs780697342</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>AGG</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>-</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>AGG</td>\n",
       "      <td>AGG</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>1/9</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>ENST00000359512.8:c.189-53_189-51del</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.3355</td>\n",
       "      <td>0.3148</td>\n",
       "      <td>0.2969</td>\n",
       "      <td>0.3306</td>\n",
       "      <td>0.2883</td>\n",
       "      <td>0.07769</td>\n",
       "      <td>0.3503</td>\n",
       "      <td>0.2731</td>\n",
       "      <td>0.3796</td>\n",
       "      <td>0.3183</td>\n",
       "      <td>0.2122</td>\n",
       "      <td>0.3796</td>\n",
       "      <td>gnomADg_NFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546101</th>\n",
       "      <td>chrX_156026900_CTC/-</td>\n",
       "      <td>chrX:156026900-156026902</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs771561228</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>CTC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>1526</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>CTC</td>\n",
       "      <td>CTC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.02029</td>\n",
       "      <td>0.01182</td>\n",
       "      <td>0.01969</td>\n",
       "      <td>0.008842</td>\n",
       "      <td>0.01583</td>\n",
       "      <td>0</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>0.0119</td>\n",
       "      <td>0.02768</td>\n",
       "      <td>0.01651</td>\n",
       "      <td>0.002979</td>\n",
       "      <td>0.0461</td>\n",
       "      <td>gnomADg_FIN</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546102</th>\n",
       "      <td>chrX_156028977_CCCCCGCACC/-</td>\n",
       "      <td>chrX:156028977-156028986</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs1433659066</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HET</td>\n",
       "      <td>CCCCCGCACC</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>3603</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>CCCCCGCACC</td>\n",
       "      <td>CCCCCGCACC</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.1572</td>\n",
       "      <td>0.04482</td>\n",
       "      <td>0.1468</td>\n",
       "      <td>0.1844</td>\n",
       "      <td>0.1297</td>\n",
       "      <td>0.3677</td>\n",
       "      <td>0.1693</td>\n",
       "      <td>0.1136</td>\n",
       "      <td>0.1745</td>\n",
       "      <td>0.1614</td>\n",
       "      <td>0.2222</td>\n",
       "      <td>0.3677</td>\n",
       "      <td>gnomADg_EAS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>546103</th>\n",
       "      <td>chrX_156029657_A/-</td>\n",
       "      <td>chrX:156029657</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000182484</td>\n",
       "      <td>ENST00000359512</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs368859346</td>\n",
       "      <td>P732912</td>\n",
       "      <td>HOM</td>\n",
       "      <td>A</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>4283</td>\n",
       "      <td>1</td>\n",
       "      <td>-</td>\n",
       "      <td>deletion</td>\n",
       "      <td>WASH6P</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>HGNC:31685</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>P1</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSP00000504557</td>\n",
       "      <td>-</td>\n",
       "      <td>A0A7I2YQU6.6</td>\n",
       "      <td>UPI000059DBFE</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>Ensembl</td>\n",
       "      <td>-</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>0.9999</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9995</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9989</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9998</td>\n",
       "      <td>1</td>\n",
       "      <td>0.9993</td>\n",
       "      <td>1</td>\n",
       "      <td>gnomADg_AMI,gnomADg_ASJ,gnomADg_FIN,gnomADg_MI...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>546104 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      #Uploaded_variation                  Location                                            Allele             Gene          Feature Feature_type                                   Consequence cDNA_position CDS_position Protein_position Amino_acids Codons Existing_variation      IND  ZYG  REF_ALLELE    IMPACT DISTANCE STRAND FLAGS VARIANT_CLASS   SYMBOL SYMBOL_SOURCE     HGNC_ID                 BIOTYPE CANONICAL MANE_SELECT MANE_PLUS_CLINICAL TSL APPRIS CCDS             ENSP SWISSPROT        TREMBL        UNIPARC UNIPROT_ISOFORM REFSEQ_MATCH   SOURCE REFSEQ_OFFSET   GIVEN_REF    USED_REF BAM_EDIT GENE_PHENO SIFT PolyPhen EXON INTRON DOMAINS miRNA                                     HGVSc HGVSp HGVS_OFFSET gnomADg_AF gnomADg_AFR_AF gnomADg_AMI_AF gnomADg_AMR_AF gnomADg_ASJ_AF gnomADg_EAS_AF gnomADg_FIN_AF gnomADg_MID_AF gnomADg_NFE_AF gnomADg_OTH_AF gnomADg_SAS_AF   MAX_AF                                        MAX_AF_POPS CLIN_SIG SOMATIC PHENO PUBMED MOTIF_NAME MOTIF_POS HIGH_INF_POS MOTIF_SCORE_CHANGE TRANSCRIPTION_FACTORS\n",
       "0                                        chr1_14398_TGT/-          chr1:14398-14400                                                 -  ENSG00000290825  ENST00000456328   Transcript            non_coding_transcript_exon_variant     1646-1648            -                -           -      -        rs756427959  P732912  HET         TGT  MODIFIER        -      1     -      deletion  DDX11L2    EntrezGene           -                  lncRNA       YES           -                  -   1      -    -                -         -             -              -               -            -  Ensembl             -         TGT         TGT        -          -    -        -  3/3      -       -     -          ENST00000456328.2:n.1648_1650del     -           -    0.01817       0.002934        0.01613        0.02403        0.03436       0.002087        0.02536        0.03636        0.03001        0.01686        0.01924  0.03636                                        gnomADg_MID        -       -     -      -          -         -            -                  -                     -\n",
       "1                                          chr1_20317_A/-                chr1:20317                                                 -           653635      NR_024540.1   Transcript  intron_variant,non_coding_transcript_variant             -            -                -           -      -       rs1320311790  P732912  HET           A  MODIFIER        -     -1     -      deletion   WASH7P    EntrezGene  HGNC:38034  transcribed_pseudogene       YES           -                  -   -      -    -                -         -             -              -               -            -   RefSeq             -           A           A       OK          -    -        -    -   2/10       -     -                 NR_024540.1:n.205-1951del     -           -     0.2744         0.1077         0.2891         0.3073         0.3022          0.347         0.3694         0.3142         0.3461         0.2865         0.2727   0.3694                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -\n",
       "2                             chr1_104161_-/ACAC/ACACACAC        chr1:104160-104161                                              ACAC  ENSG00000238009  ENST00000477740   Transcript  intron_variant,non_coding_transcript_variant             -            -                -           -      -        rs372078516  P732912  HET           -  MODIFIER        -     -1     -     insertion        -             -           -                  lncRNA       YES           -                  -   5      -    -                -         -             -              -               -            -  Ensembl             -           -           -        -          -    -        -    -    3/3       -     -  ENST00000477740.5:n.480+8536_480+8539dup     -           -     0.3246         0.2267          0.297         0.3429         0.3627         0.4661         0.2832         0.4492         0.3618         0.3337         0.4206   0.4661                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -\n",
       "3                                       chr1_133707_AGA/-        chr1:133707-133709                                                 -  ENSG00000238009  ENST00000477740   Transcript                         upstream_gene_variant             -            -                -           -      -       rs1354754438  P732912  HET         AGA  MODIFIER     4490     -1     -      deletion        -             -           -                  lncRNA       YES           -                  -   5      -    -                -         -             -              -               -            -  Ensembl             -         AGA         AGA        -          -    -        -    -      -       -     -                                         -     -           -    0.00245      0.0004914              0       0.001503        0.01094        0.01074      9.868e-05       0.003401       0.002703       0.003115       0.008426  0.01094                                        gnomADg_ASJ        -       -     -      -          -         -            -                  -                     -\n",
       "4       chr1_136604_-/GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGT...        chr1:136603-136604  GGCTGACCTCTGTCCGCGTGGGAGGGGCCGGTGTGAGGCAAGGGCTCG  ENSG00000239906  ENST00000493797   Transcript                       downstream_gene_variant             -            -                -           -      -       rs1557463817  P732912  HET           -  MODIFIER     3186     -1     -     insertion        -             -           -                  lncRNA       YES           -                  -   2      -    -                -         -             -              -               -            -  Ensembl             -           -           -        -          -    -        -    -      -       -     -                                         -     -           -    0.02026        0.02241              0        0.02184        0.02338        0.01248        0.01727        0.03571        0.02032       0.008929         0.0132  0.03571                                        gnomADg_MID        -       -     -      -          -         -            -                  -                     -\n",
       "...                                                   ...                       ...                                               ...              ...              ...          ...                                           ...           ...          ...              ...         ...    ...                ...      ...  ...         ...       ...      ...    ...   ...           ...      ...           ...         ...                     ...       ...         ...                ...  ..    ...  ...              ...       ...           ...            ...             ...          ...      ...           ...         ...         ...      ...        ...  ...      ...  ...    ...     ...   ...                                       ...   ...         ...        ...            ...            ...            ...            ...            ...            ...            ...            ...            ...            ...      ...                                                ...      ...     ...   ...    ...        ...       ...          ...                ...                   ...\n",
       "546099                                 chrX_156020867_G/-            chrX:156020867                                                 -  ENSG00000182484  ENST00000359512   Transcript                         upstream_gene_variant             -            -                -           -      -        rs748173079  P732912  HET           G  MODIFIER       94      1     -      deletion   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -           G           G        -          -    -        -    -      -       -     -                                         -     -           -    0.02063       0.004833       0.009934        0.02908        0.01323        0.03532        0.01851        0.02193        0.02955        0.02039        0.02675  0.03532                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -\n",
       "546100                               chrX_156021625_AGG/-  chrX:156021625-156021627                                                 -  ENSG00000182484  ENST00000359512   Transcript                                intron_variant             -            -                -           -      -        rs780697342  P732912  HET         AGG  MODIFIER        -      1     -      deletion   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         AGG         AGG        -          -    -        -    -    1/9       -     -      ENST00000359512.8:c.189-53_189-51del     -           -     0.3355         0.3148         0.2969         0.3306         0.2883        0.07769         0.3503         0.2731         0.3796         0.3183         0.2122   0.3796                                        gnomADg_NFE        -       -     -      -          -         -            -                  -                     -\n",
       "546101                               chrX_156026900_CTC/-  chrX:156026900-156026902                                                 -  ENSG00000182484  ENST00000359512   Transcript                       downstream_gene_variant             -            -                -           -      -        rs771561228  P732912  HET         CTC  MODIFIER     1526      1     -      deletion   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -         CTC         CTC        -          -    -        -    -      -       -     -                                         -     -           -    0.02029        0.01182        0.01969       0.008842        0.01583              0         0.0461         0.0119        0.02768        0.01651       0.002979   0.0461                                        gnomADg_FIN        -       -     -      -          -         -            -                  -                     -\n",
       "546102                        chrX_156028977_CCCCCGCACC/-  chrX:156028977-156028986                                                 -  ENSG00000182484  ENST00000359512   Transcript                       downstream_gene_variant             -            -                -           -      -       rs1433659066  P732912  HET  CCCCCGCACC  MODIFIER     3603      1     -      deletion   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -  CCCCCGCACC  CCCCCGCACC        -          -    -        -    -      -       -     -                                         -     -           -     0.1572        0.04482         0.1468         0.1844         0.1297         0.3677         0.1693         0.1136         0.1745         0.1614         0.2222   0.3677                                        gnomADg_EAS        -       -     -      -          -         -            -                  -                     -\n",
       "546103                                 chrX_156029657_A/-            chrX:156029657                                                 -  ENSG00000182484  ENST00000359512   Transcript                       downstream_gene_variant             -            -                -           -      -        rs368859346  P732912  HOM           A  MODIFIER     4283      1     -      deletion   WASH6P          HGNC  HGNC:31685          protein_coding       YES           -                  -   -     P1    -  ENSP00000504557         -  A0A7I2YQU6.6  UPI000059DBFE               -            -  Ensembl             -           A           A        -          -    -        -    -      -       -     -                                         -     -           -     0.9998         0.9999              1         0.9995              1         0.9989              1              1         0.9998              1         0.9993        1  gnomADg_AMI,gnomADg_ASJ,gnomADg_FIN,gnomADg_MI...        -       -     -      -          -         -            -                  -                     -\n",
       "\n",
       "[546104 rows x 74 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indel12\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "py39",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.18"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
